<!doctype html>



  


<html class="theme-next muse use-motion">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>



<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />












  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.4.0" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.0.1" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="Hexo, NexT" />








  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.0.1" />






<meta name="description" content="自我提升和自我完善是一种终身的修行">
<meta property="og:type" content="website">
<meta property="og:title" content="lizhong's notes">
<meta property="og:url" content="http://yoursite.com/index.html">
<meta property="og:site_name" content="lizhong's notes">
<meta property="og:description" content="自我提升和自我完善是一种终身的修行">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="lizhong's notes">
<meta name="twitter:description" content="自我提升和自我完善是一种终身的修行">



<script type="text/javascript" id="hexo.configuration">
  var NexT = window.NexT || {};
  var CONFIG = {
    scheme: 'Muse',
    sidebar: {"position":"left","display":"post"},
    fancybox: true,
    motion: true,
    duoshuo: {
      userId: 0,
      author: '博主'
    }
  };
</script>




  <link rel="canonical" href="http://yoursite.com/"/>

  <title> lizhong's notes </title>
</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  










  
  
    
  

  <div class="container one-collumn sidebar-position-left 
   page-home 
 ">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-meta ">
  

  <div class="custom-logo-site-title">
    <a href="/"  class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <span class="site-title">lizhong's notes</span>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>
  <p class="site-subtitle">套路提高生产力</p>
</div>

<div class="site-nav-toggle">
  <button>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
    <span class="btn-bar"></span>
  </button>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            归档
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      

      
    </ul>
  

  
</nav>

 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            
  <section id="posts" class="posts-expand">
    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/6666/06/06/shell-Cheetsheets/" itemprop="url">
                  Shell Scripting Cheetsheets
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="6666-06-06T00:00:00+08:00" content="6666-06-06">
              6666-06-06
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="A-Bioinformatician’s-UNIX-Toolbox"><a href="#A-Bioinformatician’s-UNIX-Toolbox" class="headerlink" title="A Bioinformatician’s UNIX Toolbox"></a><a href="http://lh3lh3.users.sourceforge.net/biounix.shtml" target="_blank" rel="external">A Bioinformatician’s UNIX Toolbox</a></h2><h2 id="awk"><a href="#awk" class="headerlink" title="awk"></a>awk</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div></pre></td><td class="code"><pre><div class="line">#取某几列</div><div class="line">awk &apos;&#123;print $2,$4,$5&#125;&apos; input.txt &gt; output.txt</div><div class="line">cat *log|grep &quot;variants loaded&quot;|awk &apos;&#123;print $1&#125;&apos;|les</div><div class="line"></div><div class="line"># 查看文件有几列</div><div class="line">cat in.txt | awk  &apos;&#123;print NF&#125;&apos;</div><div class="line"></div><div class="line"># -F 修改分隔符为冒号：</div><div class="line">awk -F: &apos;&#123;print $1&#125;&apos; in.txt</div><div class="line">awk -F&quot; &quot; &apos;&#123;print $1&#125;&apos; in.txt</div><div class="line"></div><div class="line"># 正则匹配</div><div class="line">awk &apos;/^test[0-9]+/&apos; input.txt</div><div class="line">awk &apos;$1 ~ /^test[0-9]+/&apos; input.txt</div><div class="line">awk &apos;&#123;if($5==66) print $0&#125;&apos; input.txt</div><div class="line"></div><div class="line"># 三行程序，以“;”分割</div><div class="line">awk &apos;&#123;BASELINE=42; if($1&gt;BASELINE) $5=$5+100; print $0&#125;&apos; data.f</div><div class="line"></div><div class="line"># 数某一列出现次数</div><div class="line">awk &apos;&#123;l[$1]++&#125;END&#123;for (x in l) print x,l[x]&#125;&apos; input.txt</div><div class="line"></div><div class="line"># 几列之间运算 calculate the sum of column 2 and 3 and put it at the end of a row or replace the first column:</div><div class="line">awk &apos;&#123;print $0,$2+$3&#125;&apos; input.txt</div><div class="line">awk &apos;&#123;$1=$2+$3;print&#125;&apos; input.txt</div><div class="line"></div><div class="line"># 取出文件某几行，比如5-10行</div><div class="line">sed -n &apos;5,10p&apos; data.txt</div><div class="line">awk &apos;NR&gt;=5&amp;&amp;NR&lt;=10&apos; data.txt</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="tr"><a href="#tr" class="headerlink" title="tr"></a>tr</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># 把回车都替换成空格</div><div class="line">awk &apos;&#123;print $1&#125;&apos; in.txt|tr &quot;\n&quot; &quot; &quot;|less</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="纵向合并文件"><a href="#纵向合并文件" class="headerlink" title="纵向合并文件"></a>纵向合并文件</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">paste a.txt b.txt &gt; ab.txt</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="find"><a href="#find" class="headerlink" title="find"></a>find</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">find . -name *.pl</div><div class="line">find /path/dir -regextype &quot;posix-egrep&quot; -regex &quot;.*\.(pl|sh|py|r|pdf|)$&quot;</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="fasta-fastq-vcf"><a href="#fasta-fastq-vcf" class="headerlink" title="fasta/fastq/vcf"></a>fasta/fastq/vcf</h2><ul>
<li><p>取fasta中的某一段序列（代替perl里的substr效果）</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtools faidx hg19.fasta chr1:1-100</div></pre></td></tr></table></figure>
</li>
<li><p>为vcf文件建索引，GATK依赖</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">samtools/htslib/bgzip -c file.vcf &gt; file.vcf.gz</div><div class="line">samtools/htslib/tabix -p vcf file.vcf.gz</div></pre></td></tr></table></figure>
</li>
</ul>
<p><br></p>
<h2 id="Example-header-of-PBS-qsub-Command-Lines"><a href="#Example-header-of-PBS-qsub-Command-Lines" class="headerlink" title="Example header of PBS(qsub) Command Lines"></a>Example header of PBS(qsub) Command Lines</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div></pre></td><td class="code"><pre><div class="line">#$ -S /bin/sh</div><div class="line">#$ -e z.1.sh.SuperMem-1.pbs.$JOB_ID.e</div><div class="line">#$ -o z.1.sh.SuperMem-1.pbs.$JOB_ID.o</div><div class="line">#$ -l vf=1G</div><div class="line">#$ -m n</div><div class="line">#$ -cwd</div><div class="line">#$ -P st_supermem</div><div class="line">#$ -q supermem.q@supermem-0-2</div><div class="line"># 指定supermem-0-2节点</div><div class="line"></div><div class="line"># start time</div><div class="line">date1=`date &quot;+%Y-%m-%d %H:%M:%S&quot;`; date1_sys=`date -d &quot;$date1&quot; +%s`;echo &quot;start running ========= at $date1&quot;</div><div class="line"></div><div class="line">perl test.pl</div><div class="line"></div><div class="line"># end time and running time; output to z.1.sh.SuperMem-1.pbs.$JOB_ID.o</div><div class="line">date2=`date &quot;+%Y-%m-%d %H:%M:%S&quot;`; date2_sys=`date -d &quot;$date2&quot; +%s`; interval=`expr $date2_sys - $date1_sys`; hour=`expr $interval / 3600`;left_second=`expr $interval % 3600`; min=`expr $left_second / 60`; second=`expr $interval % 60`; echo &quot;done  running ========= at $date2 in $hour hour $min min $second s&quot;</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="kill"><a href="#kill" class="headerlink" title="kill"></a>kill</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"># 一次杀掉所有进程，慎用</div><div class="line">pgrep -u wanglizhong |xargs kill</div><div class="line"></div><div class="line">kill -9 $(ps -ef|grep wanglizhong|awk &apos;&#123;print $2&#125;&apos;)</div><div class="line"></div><div class="line">pkill -u wanglizhong</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="解压-压缩"><a href="#解压-压缩" class="headerlink" title="解压/压缩"></a>解压/压缩</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div><div class="line">33</div><div class="line">34</div><div class="line">35</div><div class="line">36</div><div class="line">37</div><div class="line">38</div><div class="line">39</div><div class="line">40</div><div class="line">41</div><div class="line">42</div><div class="line">43</div><div class="line">44</div><div class="line">45</div><div class="line">46</div><div class="line">47</div><div class="line">48</div><div class="line">49</div><div class="line">50</div><div class="line">51</div><div class="line">52</div><div class="line">53</div><div class="line">54</div><div class="line">55</div><div class="line">56</div><div class="line">57</div><div class="line">58</div><div class="line">59</div><div class="line">60</div><div class="line">61</div><div class="line">62</div><div class="line">63</div><div class="line">64</div><div class="line">65</div><div class="line">66</div><div class="line">67</div></pre></td><td class="code"><pre><div class="line">.tar</div><div class="line">解包：tar xvf Something.tar</div><div class="line">打包：tar cvf Something.tar DirName</div><div class="line">（注：tar是打包，不是压缩！）</div><div class="line">———————————————</div><div class="line">.gz</div><div class="line">解压1：gunzip Something.gz</div><div class="line">解压2：gzip -d Something.gz</div><div class="line">压缩：gzip Something</div><div class="line"></div><div class="line">.tar.gz 和 .tgz</div><div class="line">解压：tar zxvf Something.tar.gz</div><div class="line">压缩：tar zcvf Something.tar.gz DirName</div><div class="line">———————————————</div><div class="line">.bz2</div><div class="line">解压1：bzip2 -d Something.bz2</div><div class="line">解压2：bunzip2 Something.bz2</div><div class="line">压缩： bzip2 -z Something</div><div class="line"></div><div class="line">.tar.bz2</div><div class="line">解压：tar jxvf Something.tar.bz2</div><div class="line">压缩：tar jcvf Something.tar.bz2 DirName</div><div class="line">———————————————</div><div class="line">.bz</div><div class="line">解压1：bzip2 -d Something.bz</div><div class="line">解压2：bunzip2 Something.bz</div><div class="line">压缩：未知</div><div class="line"></div><div class="line">.tar.bz</div><div class="line">解压：tar jxvf Something.tar.bz</div><div class="line">压缩：未知</div><div class="line">———————————————</div><div class="line">.Z</div><div class="line">解压：uncompress Something.Z</div><div class="line">压缩：compress Something</div><div class="line">.tar.Z</div><div class="line"></div><div class="line">解压：tar Zxvf Something.tar.Z</div><div class="line">压缩：tar Zcvf Something.tar.Z DirName</div><div class="line">———————————————</div><div class="line">.zip</div><div class="line">解压：unzip Something.zip</div><div class="line">压缩：zip Something.zip DirName</div><div class="line">———————————————</div><div class="line">.rar</div><div class="line">解压：rar x Something.rar</div><div class="line">压缩：rar a Something.rar DirName</div><div class="line">———————————————</div><div class="line">.lha</div><div class="line">解压：lha -e Something.lha</div><div class="line">压缩：lha -a Something.lha Something</div><div class="line">———————————————</div><div class="line">.rpm</div><div class="line">解包：rpm2cpio Something.rpm | cpio -div</div><div class="line">#———————————————</div><div class="line">.deb</div><div class="line">解包：ar p Something.deb data.tar.gz | tar zxf -</div><div class="line">———————————————</div><div class="line">.tar .tgz .tar.gz .tar.Z .tar.bz .tar.bz2 .zip .cpio .rpm .deb .slp .arj .rar .ace .lha .lzh .lzx .lzs .arc .sda .sfx .lnx .zoo .cab .kar .cpt .pit .sit .sea</div><div class="line">解压：sEx x Something.*</div><div class="line">压缩：sEx a Something.* Something</div><div class="line"></div><div class="line">.tar.xz    (xz http://tukaani.org/xz/)</div><div class="line">xz -d Something.tar.xz</div><div class="line"></div><div class="line"># gzip 压缩比</div><div class="line">-num 用指定的数字 num 调整压缩的速度，-1 或 --fast 表示最快压缩方法（低压缩比），-9 或--best表示最慢压缩方法（高压缩比）。系统缺省值为 6。</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/11/09/Reference-check/" itemprop="url">
                  投稿前Reference部分的检查清单
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-11-09T17:55:09+08:00" content="2017-11-09">
              2017-11-09
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>因citation文件的下载来源不同，导入endnote之后直接使用可能存在一些不一致的格式情况，建议投稿前一一排查。</p>
<p>这些内容很细碎，主要是通过细节给编辑和审稿人体现专业精神，如果发现很多小错误会给人不专业、不仔细的印象。</p>
<p>列举一些情况：</p>
<p>1、多余的空格、标点符号<br><img src="http://ofr9vioug.bkt.clouddn.com/6C08901821A1F04ADF33CBCAFC37A2DE.jpg?imageView/2/w/500" alt="image"></p>
<p>2、期刊/作者名，首字母大写、全部大写和部分大写不统一<br><img src="http://ofr9vioug.bkt.clouddn.com/698E733A9609816A2DFDF4188B95D885.jpg?imageView/2/w/500" alt="image"><br><img src="http://ofr9vioug.bkt.clouddn.com/90A18BD5C1C9CA2CDED2D9DC71DC76FB.jpg?imageView/2/w/500" alt="image"><br><img src="http://ofr9vioug.bkt.clouddn.com/B1398DA5-039B-491F-84B3-14A1D8F0031C.png?imageView/2/w/500" alt="image"></p>
<p>3、期刊全称、缩写不统一<br><img src="http://ofr9vioug.bkt.clouddn.com/6577D6B0-71F5-450A-B57E-1010A6E4BEC0.png?imageView/2/w/500" alt="image"></p>
<p>4、作者名字全称、缩写不统一<br><img src="http://ofr9vioug.bkt.clouddn.com/78B28D51-ADFC-4278-843F-AA7A64DDBF72.png?imageView/2/w/500" alt="image"></p>
<p>5、作者名字，「先姓后名」和「先名后姓」不统一；多个字的名，如Lizhong，「缩写一个」（L.），与「缩写两个」（L. Z.）的不统一<br><img src="http://ofr9vioug.bkt.clouddn.com/938C727E-8D2E-4F7A-A2D0-DE35D16D9B45.png?imageView/2/w/500" alt="image"><br><img src="http://ofr9vioug.bkt.clouddn.com/04C965E7-A2E3-4AD4-8587-7D619773BA32.png?imageView/2/w/500" alt="image"></p>
<p>6、特殊词的斜体，如物种名<br><img src="http://ofr9vioug.bkt.clouddn.com/D9D06BDD-F4E6-428B-B0AB-116D680D55AB.png?imageView/2/w/500" alt="image"></p>
<p>7、检查目标期刊要求的引用文献style，下载相应的style配置文件放到endnote/Styles 文件夹内，即可选用。</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/09/21/Plot-Quantile-Quantile-plot-with-qqline-using-ggplot2/" itemprop="url">
                  Plot Quantile-Quantile plot with qqline using ggplot2
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-09-21T19:03:37+08:00" content="2017-09-21">
              2017-09-21
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170921-190651.png?imageView/2/w/500" alt="image"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">myresiduals &lt;- rnorm(100)</div><div class="line">pdf(&quot;01.f4.pdf&quot;,width=3,height=3)</div><div class="line"></div><div class="line">ggplot(data=as.data.frame(qqnorm( myresiduals , plot=F)), mapping=aes(x=x, y=y)) + geom_point() + geom_smooth(method=&quot;lm&quot;, se=FALSE)+labs(title=&quot;D(Agarli,Urial;Mouflon1,Mouflon2)&quot;,x=&quot;Expected Z-scores&quot;,y=&quot;Observed Z-scores&quot;)+ theme_classic()</div><div class="line"></div><div class="line">dev.off()</div></pre></td></tr></table></figure></p>
<p>Quantile-Quantile plot can use to analysis D-statistics like this:<br><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170921-190533@2x.png?imageView/2/w/500" alt="image"><br>(DOI: 10.1126/science.aan3842 , Fig. 2)</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/09/07/VCF-manipulation-with-GATK/" itemprop="url">
                  VCF manipulation with GATK
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-09-07T13:51:56+08:00" content="2017-09-07">
              2017-09-07
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="HaplotypeCaller-每个Sample生成GVCF文件-相当于一个SNP-calling的中间文件"><a href="#HaplotypeCaller-每个Sample生成GVCF文件-相当于一个SNP-calling的中间文件" class="headerlink" title="HaplotypeCaller, 每个Sample生成GVCF文件, 相当于一个SNP calling的中间文件"></a><code>HaplotypeCaller</code>, 每个Sample生成GVCF文件, 相当于一个SNP calling的中间文件</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.8.0_45;</div><div class="line"></div><div class="line">jre1.8.0_45/bin/java -Djava.io.tmpdir=tmpdir \</div><div class="line">-jar gatk3.6/GenomeAnalysisTK.jar \</div><div class="line">-T GenotypeGVCFs \</div><div class="line">-R ref.fa \</div><div class="line">-I sampleA.bam \</div><div class="line">-nct 15 \</div><div class="line">-ERC GVCF \ # gvcf模式</div><div class="line">-L interver.list \ # chr10:10000001-15000000</div><div class="line">[--pcr_indel_model NONE ] \ # 如果bam去除了PCR duplications可以加上这个</div><div class="line">-o sampleA.chr10_10000001_15000000.g.vcf.gz \</div><div class="line">-variant_index_type LINEAR \</div><div class="line">-variant_index_parameter 128000 \</div><div class="line">2&gt;&amp;1 | tee 01.vcfByWindow/chr10_10000001_15000000/14-5.gvcf.gz.log  # log输出</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="GenotypeGVCFs-多个sample做SNP-calling"><a href="#GenotypeGVCFs-多个sample做SNP-calling" class="headerlink" title="GenotypeGVCFs, 多个sample做SNP calling"></a><code>GenotypeGVCFs</code>, 多个sample做SNP calling</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.8.0_45;</div><div class="line"></div><div class="line">jre1.8.0_45/bin/java -Djava.io.tmpdir=tmpdir \</div><div class="line">-jar gatk3.6/GenomeAnalysisTK.jar \</div><div class="line">-T GenotypeGVCFs \</div><div class="line">-R ref.fa \</div><div class="line">-V gvcf.list \ # 多个体的gvcf列表，一行一个</div><div class="line">-stand_call_conf 30.0 \</div><div class="line">-stand_emit_conf 10.0 \</div><div class="line">-o variants.vcf.gz;</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="CatVariants，把分区域多线程call的variants合并起来，注意使用java1-7-gatk3-3-3-4"><a href="#CatVariants，把分区域多线程call的variants合并起来，注意使用java1-7-gatk3-3-3-4" class="headerlink" title="CatVariants，把分区域多线程call的variants合并起来，注意使用java1.7+gatk3.3/3.4"></a><code>CatVariants</code>，把分区域多线程call的variants合并起来，注意使用<code>java1.7</code>+<code>gatk3.3/3.4</code></h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.7.0_55;</div><div class="line"></div><div class="line">jre1.7.0_55/bin/java -cp GATK-3.4/GenomeAnalysisTK.jar org.broadinstitute.gatk.tools.CatVariants \</div><div class="line">-R ref.fa \</div><div class="line">-V 01.vcfByWindow/chr1_1_5000000/variants.vcf.gz \ # 或者-V vcf.list</div><div class="line">-V 01.vcfByWindow/chr1_5000001_10000000/variants.vcf.gz \</div><div class="line">-V 01.vcfByWindow/chr1_10000001_15000000/variants.vcf.gz \</div><div class="line">-V 01.vcfByWindow/chr1_15000001_20000000/variants.vcf.gz \</div><div class="line">-out chr1.01.vcf.gz \ # 是-out，不是-o</div><div class="line">[-assumeSorted] \ # 如果--variant按顺序排的，则加上；乱的就不加</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="SelectVariants，取overlap位点，保留set2里和set1-overlap的位点"><a href="#SelectVariants，取overlap位点，保留set2里和set1-overlap的位点" class="headerlink" title="SelectVariants，取overlap位点，保留set2里和set1 overlap的位点"></a><code>SelectVariants</code>，取overlap位点，保留set2里和set1 overlap的位点</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.8.0_45;</div><div class="line"></div><div class="line">jre1.8.0_45/bin/java -Djava.io.tmpdir=tmpdir \</div><div class="line">-jar gatk3.6/GenomeAnalysisTK.jar \</div><div class="line">-T SelectVariants \</div><div class="line">-R ref.fa \</div><div class="line">-V set2.vcf.gz \</div><div class="line">-o set2.overlap.vcf.gz \</div><div class="line">--concordance set1.vcf.gz</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="SelectVariants，去除-保留个体"><a href="#SelectVariants，去除-保留个体" class="headerlink" title="SelectVariants，去除/保留个体"></a><code>SelectVariants</code>，去除/保留个体</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.8.0_45;</div><div class="line"></div><div class="line">jre1.8.0_45/bin/java -Djava.io.tmpdir=tmpdir \</div><div class="line">-jar gatk3.6/GenomeAnalysisTK.jar \</div><div class="line">-T SelectVariants \</div><div class="line">-R ref.fa \</div><div class="line">-V set1.vcf.gz \</div><div class="line">-o set1.remove.vcf.gz \</div><div class="line"></div><div class="line"># 下面两个二选一</div><div class="line">-xl_sf remove.list \ # 去除个体</div><div class="line">-sf keep.list \      # 保留个体</div></pre></td></tr></table></figure>
<h2 id="CombineVariants-genotypeMergeOptions-UNIQUIFY，合并多个vcf，不管个体ID，相同ID的个体都当成不同的并重新命名-两个set位点可以不完全相同，没信息的genotype会自动标记成"><a href="#CombineVariants-genotypeMergeOptions-UNIQUIFY，合并多个vcf，不管个体ID，相同ID的个体都当成不同的并重新命名-两个set位点可以不完全相同，没信息的genotype会自动标记成" class="headerlink" title="CombineVariants -genotypeMergeOptions UNIQUIFY，合并多个vcf，不管个体ID，相同ID的个体都当成不同的并重新命名; 两个set位点可以不完全相同，没信息的genotype会自动标记成./."></a><code>CombineVariants -genotypeMergeOptions UNIQUIFY</code>，合并多个vcf，不管个体ID，相同ID的个体都当成不同的并重新命名; 两个set位点可以不完全相同，没信息的genotype会自动标记成<code>./.</code></h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.8.0_45;</div><div class="line"></div><div class="line">jre1.8.0_45/bin/java -Djava.io.tmpdir=tmpdir \</div><div class="line">-jar gatk3.6/GenomeAnalysisTK.jar \</div><div class="line">-T CombineVariants \</div><div class="line">-R ref.fa \</div><div class="line">-V:v1 a.vcf.gz \</div><div class="line">-V:v2 b.vcf.gz \</div><div class="line">-o ab.UNIQUIFY.vcf.gz \</div><div class="line">-genotypeMergeOptions UNIQUIFY</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="CombineVariants-genotypeMergeOptions-PRIORITIZE-priority-v1-v2，可能有相同个体的，按照优先级选择具体的genotype"><a href="#CombineVariants-genotypeMergeOptions-PRIORITIZE-priority-v1-v2，可能有相同个体的，按照优先级选择具体的genotype" class="headerlink" title="CombineVariants -genotypeMergeOptions PRIORITIZE -priority v1,v2，可能有相同个体的，按照优先级选择具体的genotype"></a><code>CombineVariants -genotypeMergeOptions PRIORITIZE -priority v1,v2</code>，可能有相同个体的，按照优先级选择具体的genotype</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=java/jre1.8.0_45;</div><div class="line"></div><div class="line">jre1.8.0_45/bin/java -Djava.io.tmpdir=tmpdir \</div><div class="line">-jar gatk3.6/GenomeAnalysisTK.jar \</div><div class="line">-T CombineVariants \</div><div class="line">-R ref.fa \</div><div class="line">-V:v1 a.vcf.gz \</div><div class="line">-V:v2 b.vcf.gz \</div><div class="line">-o ab.PRIORITIZE.vcf.gz \</div><div class="line">-genotypeMergeOptions PRIORITIZE -priority v1,v2</div></pre></td></tr></table></figure>
<p><br></p>
<p>类似：<a href="https://wlz0726.github.io/2017/04/12/vcftools/" target="_blank" rel="external">vcftools</a></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/08/18/Map-with-BWA-and-do-Indel-realignment-with-GATK/" itemprop="url">
                  Map with BWA and do Indel realignment with GATK
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-08-18T14:11:22+08:00" content="2017-08-18">
              2017-08-18
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                
                  ， 
                

              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/pipeline/" itemprop="url" rel="index">
                    <span itemprop="name">pipeline</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="01-Mapping-BWAMEM-GATKRealgn-pl"><a href="#01-Mapping-BWAMEM-GATKRealgn-pl" class="headerlink" title="01.Mapping_BWAMEM-GATKRealgn.pl"></a>01.Mapping_BWAMEM-GATKRealgn.pl</h1><p>get script<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">git clone https://github.com/wlz0726/SequenceMapping.git</div></pre></td></tr></table></figure></p>
<h2 id="step1-Mapping-with-BWA-MEM-and-sort-with-samtools-connected-with-tunnel"><a href="#step1-Mapping-with-BWA-MEM-and-sort-with-samtools-connected-with-tunnel" class="headerlink" title="step1: Mapping with BWA MEM and sort with samtools, connected with tunnel"></a>step1: Mapping with BWA MEM and sort with samtools, connected with tunnel</h2><p>NB: samtools should be v1.* or later<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">bwa mem -R ref.fa SampleID.1.fq.gz SampleID.2.fq.gz | samtools sort -O bam -o SampleID.sort.bam</div><div class="line">samtools index SampleID.sort.bam</div></pre></td></tr></table></figure></p>
<h2 id="step2-remove-PCR-duplicates-with-picard"><a href="#step2-remove-PCR-duplicates-with-picard" class="headerlink" title="step2: remove PCR duplicates with picard"></a>step2: remove PCR duplicates with <a href="http://broadinstitute.github.io/picard/" target="_blank" rel="external">picard</a></h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">java -jar picard.jar MarkDuplicates \</div><div class="line">INPUT=SampleID.sort.bam \</div><div class="line">OUTPUT=SampleID.rmdup.bam \</div><div class="line">METRICS_FILE =SampleID.dup.txt \</div><div class="line">REMOVE_DUPLICATES=true \</div><div class="line">VALIDATION_STRINGENCY=SILENT \</div><div class="line">CREATE_INDEX=true</div></pre></td></tr></table></figure>
<h2 id="step3-do-local-realignment-of-reads-to-enhance-the-alignments-in-the-vicinity-of-indel-polymorphisms"><a href="#step3-do-local-realignment-of-reads-to-enhance-the-alignments-in-the-vicinity-of-indel-polymorphisms" class="headerlink" title="step3: do local realignment of reads to enhance the alignments in the vicinity of indel polymorphisms"></a>step3: do local realignment of reads to enhance the alignments in the vicinity of indel polymorphisms</h2><p>NB: <a href="https://software.broadinstitute.org/gatk/" target="_blank" rel="external">GATK v3.6</a><br>NB: <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.php" target="_blank" rel="external">Note that indel realignment is no longer necessary for variant discovery if you plan to use a variant caller that performs a haplotype assembly step, such as HaplotypeCaller or MuTect2. However it is still required when using legacy callers such as UnifiedGenotyper or the original MuTect.</a></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div></pre></td><td class="code"><pre><div class="line"># use the RealignerTargetCreator to identify regions where realignment was needed</div><div class="line">java -jar GenomeAnalysisTK.jar -T RealignerTargetCreator \</div><div class="line">-R ref.fa \</div><div class="line">-o SampleID.intervals \</div><div class="line">-I SampleID.rmdup.bam</div><div class="line"></div><div class="line"># use IndelRealigner to realign the regions found in the RealignerTargetCreator step</div><div class="line">java -jar GenomeAnalysisTK.jar -T IndelRealigner \</div><div class="line">-R ref.fa \</div><div class="line">-targetIntervals SampleID.intervals \</div><div class="line">-I SampleID.rmdup.bam \</div><div class="line">-o SampleID.realn.bam</div></pre></td></tr></table></figure>
<hr>
<h2 id="Before-mapping-Filter-and-Trim-your-Paired-end-FastQ-file"><a href="#Before-mapping-Filter-and-Trim-your-Paired-end-FastQ-file" class="headerlink" title="Before mapping: Filter and Trim your Paired-end FastQ file"></a>Before mapping: <a href="https://wlz0726.github.io/2017/08/18/Filter-and-Trim-your-Paired-end-FastQ-file/" target="_blank" rel="external">Filter and Trim your Paired-end FastQ file</a></h2><h2 id="After-mapping-do-SNP-calling-with-GATK"><a href="#After-mapping-do-SNP-calling-with-GATK" class="headerlink" title="After mapping: do SNP calling with GATK"></a>After mapping: <a href="https://wlz0726.github.io/2017/03/23/GATK-SNP-calling/" target="_blank" rel="external">do SNP calling with GATK</a></h2>
          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/08/18/Filter-and-Trim-your-Paired-end-FastQ-file/" itemprop="url">
                  Filter and Trim your Paired-end FastQ file
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-08-18T10:46:27+08:00" content="2017-08-18">
              2017-08-18
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="FastQtrimmer-pl"><a href="#FastQtrimmer-pl" class="headerlink" title="FastQtrimmer.pl"></a>FastQtrimmer.pl</h1><h3 id="Filter-and-Trim-your-Paired-end-FastQ-file"><a href="#Filter-and-Trim-your-Paired-end-FastQ-file" class="headerlink" title="Filter and Trim your Paired-end FastQ file"></a>Filter and Trim your Paired-end FastQ file</h3><p>get script<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">git clone https://github.com/wlz0726/SequenceMapping.git</div></pre></td></tr></table></figure></p>
<p>As you get the Illumina sequencing data from a company like BGI/Novogene, first thing you (probably) need to do is to do the sequnce quality check and filter the low quality reads.</p>
<p>Here I assume you have no adaptor problem and the sequencing company have already remove the adaptor before hand over you the data.</p>
<p>FastQtrimmer.pl will trim and filtering Paired-end reads in gz format, and the following reads will be removed:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">1)  &gt;=10% unidentified nucleotides (N)</div><div class="line">2)  with a phred quality &lt;=7 for &gt;65% of the read length</div><div class="line">3)  reads are trimmed if they had three consecutive bp with a phred quality of &lt;=13,</div><div class="line">    and discarded if they were shorter than 45 bp</div></pre></td></tr></table></figure></p>
<p>Example:<br>Your have one Paired-end FastQ file named <code>SampleID.1.fq.gz</code> and  <code>SampleID.2.fq.gz</code> with phred33 quality system, the read length is 100 bp,</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">perl FastQtrimmer.pl SampleID.1.fq.gz SampleID.2.fq.gz SampleID.Filter 33 100</div></pre></td></tr></table></figure>
<p>the results will be:<br>SampleID.Filter.1.TrimTwoSide.fq.gz<br>SampleID.Filter.2.TrimTwoSide.fq.gz</p>
<hr>
<h1 id="fastq-Phred64-to-33-gz-pl"><a href="#fastq-Phred64-to-33-gz-pl" class="headerlink" title="fastq_Phred64_to_33_gz.pl"></a>fastq_Phred64_to_33_gz.pl</h1><p>If your reads are in Phred64 quality system, and you need to transform Phred64 reads to Phred33, run like this:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">perl fastq_Phred64_to_33_gz.pl input.1.fq.gz input.2.fq.gz OutputPrefix</div></pre></td></tr></table></figure></p>
<hr>
<h2 id="After-read-Quality-control-Map-with-BWA-and-do-Indel-realignment-with-GATK"><a href="#After-read-Quality-control-Map-with-BWA-and-do-Indel-realignment-with-GATK" class="headerlink" title="After read Quality control: Map with BWA and do Indel realignment with GATK"></a>After read Quality control: <a href="https://wlz0726.github.io/2017/08/18/Map-with-BWA-and-do-Indel-realignment-with-GATK/" target="_blank" rel="external">Map with BWA and do Indel realignment with GATK</a></h2>
          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/28/Data-management-of-plink-format/" itemprop="url">
                  Data management of plink format
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-28T16:33:27+08:00" content="2017-06-28">
              2017-06-28
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><a href="https://www.cog-genomics.org/plink/1.9/" target="_blank" rel="external">plink 1.9 manul</a></p>
<h1 id="取出某一些位点"><a href="#取出某一些位点" class="headerlink" title="取出某一些位点"></a>取出某一些位点</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">plink-1.9/plink --bfile input --extract postions.in --make-bed --out 03.array.v2.overlap;</div></pre></td></tr></table></figure>
<h1 id="合并两个文件"><a href="#合并两个文件" class="headerlink" title="合并两个文件"></a>合并两个文件</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"># 先转换成vcf格式</div><div class="line">plink-1.9/plink --bfile 04.array --recode vcf --out 05.array</div><div class="line">plink-1.9/plink --bfile 04.seq --recode vcf --out 05.seq</div><div class="line"># 合并</div><div class="line">samtools/bcftools/bcftools merge --merge all --no-version --threads 10 05.array.vcf.gz 05.seq.vcf.gz |bgzip -c &gt; 06.merge.vcf.gz</div><div class="line">tabix -p vcf 06.merge.vcf.gz</div></pre></td></tr></table></figure>
<h1 id="1234编码转换成ACGT编码"><a href="#1234编码转换成ACGT编码" class="headerlink" title="1234编码转换成ACGT编码"></a>1234编码转换成ACGT编码</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"># allele1234 to alleleACGT</div><div class="line">plink-1.9/plink --file data --recode --alleleACGT multichar --out data.plink</div><div class="line"># 反过来用 --allele1234</div></pre></td></tr></table></figure>
<h1 id="设染色体"><a href="#设染色体" class="headerlink" title="设染色体"></a>设染色体</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">--chr-set 90</div><div class="line">--cow</div><div class="line">--dog</div><div class="line">--horse</div><div class="line">--mouse</div><div class="line">--rice</div><div class="line">--sheep</div></pre></td></tr></table></figure>
<p>–cow = –chr-set 29 no-xy<br>–dog = –chr-set 38<br>–horse = –chr-set 31 no-xy no-mt<br>–mouse = –chr-set 19 no-xy no-mt<br>–rice5 = –chr-set -12<br>–sheep = –chr-set 26 no-xy no-mt<br>–autosome-num [value] = –chr-set [value] no-y no-xy no-mt</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/23/More-about-sequence-alignments/" itemprop="url">
                  More about sequence alignments
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-23T17:04:06+08:00" content="2017-06-23">
              2017-06-23
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="序列比对PPT"><a href="#序列比对PPT" class="headerlink" title="序列比对PPT"></a>序列比对PPT</h1><p>链接:<a href="http://pan.baidu.com/s/1i5Ia9wD" target="_blank" rel="external">http://pan.baidu.com/s/1i5Ia9wD</a>  密码:jsio</p>
<hr>
<h1 id="一、延伸材料"><a href="#一、延伸材料" class="headerlink" title="一、延伸材料"></a>一、延伸材料</h1><p>Blast使用<br><a href="https://www.ncbi.nlm.nih.gov/books/NBK1734/" target="_blank" rel="external">https://www.ncbi.nlm.nih.gov/books/NBK1734/</a><br>Needleman-Wunsch algorithmdemo<br><a href="http://experiments.mostafa.io/public/needleman-wunsch/" target="_blank" rel="external">http://experiments.mostafa.io/public/needleman-wunsch/</a><br>Dynamic Programming<br><a href="http://www.avatar.se/molbioinfo2001/dynprog/dynamic.html" target="_blank" rel="external">http://www.avatar.se/molbioinfo2001/dynprog/dynamic.html</a><br>FastQ格式<br><a href="https://en.wikipedia.org/wiki/FASTQ_format" target="_blank" rel="external">https://en.wikipedia.org/wiki/FASTQ_format</a><br>SAM/BAM格式说明<br><a href="http://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="external">http://samtools.github.io/hts-specs/SAMv1.pdf</a><br>SAM flags解释<br><a href="https://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="external">https://broadinstitute.github.io/picard/explain-flags.html</a></p>
<h1 id="二、常用比对软件及简单描述"><a href="#二、常用比对软件及简单描述" class="headerlink" title="二、常用比对软件及简单描述"></a>二、常用比对软件及简单描述</h1><ul>
<li>多序列比对<br><a href="http://www.ebi.ac.uk/Tools/msa/" target="_blank" rel="external">http://www.ebi.ac.uk/Tools/msa/</a><br>ClustalW，最广、经典<br>MUSCLE，快<br>T-coffee，相比ClustalW准确度高<br>Prank，相比ClustalW准确性高<br>MAFFT，速度准确度都好过ClustalW</li>
</ul>
<p>速度：MUSCLE &gt; MAFFT ~ Prank&gt;  ClustalW &gt;T-Coffee<br>准确性：MAFFT &gt; Prank &gt; MUSCLE &gt;  T-Coffee &gt; ClustalW</p>
<ul>
<li>短序列比对<br>BWA<br><a href="http://bio-bwa.sourceforge.net/bwa.shtml" target="_blank" rel="external">http://bio-bwa.sourceforge.net/bwa.shtml</a><br>Bowtie2<br><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank" rel="external">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a><br>SOAP2<br><a href="http://soap.genomics.org.cn/soapaligner.html" target="_blank" rel="external">http://soap.genomics.org.cn/soapaligner.html</a><br>Samtools<br><a href="http://samtools.sourceforge.net/" target="_blank" rel="external">http://samtools.sourceforge.net/</a></li>
</ul>
<p>BWA为最常用的序列比对软件，可以处理几十bp到1Mb长的reads；<br>Bowtie2常用于RNA-seq、Chip-seq等，速度比BWA快；<br>SOAP2输出的比对格式与BWA、Bowtie2不同（不是SAM/BAM格式），使用时需要注意；<br>Samtools 是比对文件（SAM/BAM）处理相关最常用的工具软件。</p>
<h1 id="三、比对相关命令（BWA）："><a href="#三、比对相关命令（BWA）：" class="headerlink" title="三、比对相关命令（BWA）："></a>三、比对相关命令（BWA）：</h1><ol>
<li><p>为参考序列Reference建库：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ bwa index –a bwtsw ref.fa</div></pre></td></tr></table></figure>
</li>
<li><p>使用BWA-MEM算法比对，保存为SAM格式</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ bwa mem ref.fa read.1.fq.gz read.2.fq.gz &gt; align.sam</div></pre></td></tr></table></figure>
</li>
<li><p>使用BWA-MEM算法比对，保存为BAM格式（压缩的SAM格式）</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">$ bwa mem ref.fa lane1.read.1.fq.gz lane1.read.2.fq.gz | samtools view -bS -o align1.bam –</div><div class="line">$ bwa mem ref.fa lane2.read.1.fq.gz lane2.read.2.fq.gz | samtools view -bS -o align2.bam -</div></pre></td></tr></table></figure>
</li>
<li><p>合并多个比对文件（比如同一个样品，分别在两条lane测序，生成两对reads）</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ samtools merge merge.bam align1.bam align2.bam</div></pre></td></tr></table></figure>
</li>
<li><p>将比对结果进行排序</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ samtools sort align.bam align.sort</div></pre></td></tr></table></figure>
</li>
<li><p>去除PCR重复</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ samtools rmdup output.sort.bam output.dedup.bam</div></pre></td></tr></table></figure>
</li>
<li><p>使用GATK套件Indel附近区域进行重新比对（Realignment）</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">$ java -jar GenomeAnalysisTK.jar -R ref.fa -T RealignerTargetCreator -o output.realn.intervals -I output.dedup.bam</div><div class="line">$ java -jar GenomeAnalysisTK.jar -R ref.fa -T IndelRealigner -targetIntervals output.realn.intervals -o output.realn.bam -I output.dedup.bam</div></pre></td></tr></table></figure>
</li>
</ol>
<h1 id="四、处理比对文件（SAM-BAM格式）"><a href="#四、处理比对文件（SAM-BAM格式）" class="headerlink" title="四、处理比对文件（SAM/BAM格式）"></a>四、处理比对文件（SAM/BAM格式）</h1><ol>
<li><p>打开SAM文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ less align.sam</div></pre></td></tr></table></figure>
</li>
<li><p>打开BAM文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">$ samtools view align.bam |less     # 查看比对信息（无Header信息）</div><div class="line">$ samtools view -h align.bam |less     # 查看Header和比对信息</div><div class="line">$ samtools view -H align.bam |less     # 查看Header信息</div></pre></td></tr></table></figure>
</li>
<li><p>查看比对深度</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">$ samtools depth align.bam | less    </div><div class="line">$ samtools depth -r chr1:1000-2000 | less      # 只查看染色体chr1上1000-2000位置</div></pre></td></tr></table></figure>
</li>
<li><p>samtools主要功能描述<br><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170623-170835.png?imageView/2/w/500" alt="image"></p>
</li>
</ol>
<hr>
<h1 id="试一试："><a href="#试一试：" class="headerlink" title="试一试："></a>试一试：</h1><ul>
<li>使用下列序列进行搜索<br>MSTAVLENPGLGRKLSDFGQETSYIEDNCNQNGAISLIFSLKEEVGALAKVLRLFEENDVNLTHIESRPSRLKKDEYEFFTHLDKRSLPALTNIIKILRHDIGATVHELSRDKKKDTVPWFPRTIQELDRFANQILSYGAELDADHPGFKDPVYRARRKQFADAYNYRHGQPIPRVEYMEEEKKTWGTVFKTLKSLYKTHACYEYNHIFPLLEKYCGFHEDNIPQLEDVSQFLQTCTGFRLRPVALLSSRDFLGGLAFRVFHCTQYIRHGSKPMYTPEPDICHELLGHVPLFSDRSFAQFSQEIGLASLGAPDEYIEKLATIYWFTVEFGLCKQGDSIKAYGAGLLSSFGELQYCLSEKPKLLPLELEKTAIQNYTVTEFQPLYYVAESFNDAKEKVRNFAATIPRPFSVRYDPYTQRIEVLDNTQQLKILADSINSEIGILCSALQKIK</li>
</ul>
<p>从结果中获得以下信息：<br>该序列在人类中的基因描述；<br>第一个匹配结果的bit score得分、E值、Identities、Gap数目；<br>上述指标第二个匹配结果如何；<br>下载前两个匹配结果，选择任一比对软件将与原序列与下载的软件一起进行多序列比对，它们之间的差异在什么地方，与blast结果比对的结果有什么异同。</p>
<ul>
<li>使用NCBI数据库获得不少于10个物种BRCA2基因氨基酸序列，并应用Clustal软件（<a href="http://www.ebi.ac.uk/Tools/msa/clustalo/" target="_blank" rel="external">http://www.ebi.ac.uk/Tools/msa/clustalo/</a>）进行序列多重比对得到系统进化树，分析其进化关系</li>
</ul>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/23/remove-unmapped-reads-of-a-BAM-file/" itemprop="url">
                  remove unmapped reads of a BAM file
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-23T12:55:34+08:00" content="2017-06-23">
              2017-06-23
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtools view -bF 4 input.bam -o output.filterUnmapped.bam</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/21/Random-subsample-from-a-BAM-file/" itemprop="url">
                  Random subsample from a BAM file
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-21T16:15:19+08:00" content="2017-06-21">
              2017-06-21
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="picard-DownsampleSam"><a href="#picard-DownsampleSam" class="headerlink" title="picard DownsampleSam"></a>picard DownsampleSam</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">java -jar picard.jar DownsampleSam P=0.25 I=QX70A.sorted.bam O=QX70A.Per0.25.bam CREATE_INDEX=true</div></pre></td></tr></table></figure>
<p><a href="https://broadinstitute.github.io/picard/command-line-overview.html#DownsampleSam" target="_blank" rel="external">picard</a></p>
<h1 id="samtools-s"><a href="#samtools-s" class="headerlink" title="samtools -s"></a>samtools -s</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtools view -h -b -s 0.25 QX70A.sorted.bam -o QX70A.Per0.25.bam</div></pre></td></tr></table></figure>
<h1 id="GATK-dfrac"><a href="#GATK-dfrac" class="headerlink" title="GATK -dfrac"></a>GATK -dfrac</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">java -jar GenomeAnalysisTK.jar -T PrintReads -R ref.fa -I input.bam -o output.bam -dfrac 0.25</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/19/tmux/" itemprop="url">
                  tmux
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-19T13:17:32+08:00" content="2017-06-19">
              2017-06-19
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>tmux安装过程报错可能是缺libevent，先安装libevent<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">wget https://github.com/libevent/libevent/releases/download/release-2.1.8-stable/libevent-2.1.8-stable.tar.gz</div><div class="line">tar zxvf libevent-2.1.8-stable.tar.gz</div><div class="line">cd libevent-2.1.8-stable</div><div class="line">./configure</div><div class="line">make</div><div class="line">make install</div></pre></td></tr></table></figure></p>
<p>安装tmux，运行tmux如果缺库，则把libevent-* 连接到lib目录。</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/19/无Root配置zsh和Oh-my-zsh/" itemprop="url">
                  无Root配置zsh和Oh-my-zsh
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-19T13:02:48+08:00" content="2017-06-19">
              2017-06-19
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>无root，系统里没有zsh的时候，无法用chsh切换zsh</p>
<h2 id="zsh"><a href="#zsh" class="headerlink" title="zsh"></a>zsh</h2><h3 id="安装zsh"><a href="#安装zsh" class="headerlink" title="安装zsh"></a>安装zsh</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">wget -O zsh.tar.gz https://sourceforge.net/projects/zsh/files/latest/download</div><div class="line">tar zxvf zsh.tar.gz</div><div class="line">cd zsh-*</div><div class="line">./configure --prefix=/home/wanglizhong/software/zsh.install/zsh-5.2/</div><div class="line">make</div><div class="line">make install</div></pre></td></tr></table></figure>
<h3 id="在-bash-profile-加一行"><a href="#在-bash-profile-加一行" class="headerlink" title="在 .bash_profile 加一行"></a>在 <code>.bash_profile</code> 加一行</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">exec /home/wanglizhong/software/zsh.install/zsh-5.2/bin/zsh</div></pre></td></tr></table></figure>
<h2 id="oh-my-zsh"><a href="#oh-my-zsh" class="headerlink" title="oh-my-zsh"></a>oh-my-zsh</h2><h3 id="安装oh-my-zsh"><a href="#安装oh-my-zsh" class="headerlink" title="安装oh-my-zsh"></a>安装oh-my-zsh</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line">git clone git://github.com/robbyrussell/oh-my-zsh.git ~/.oh-my-zsh</div><div class="line"># 备份原来的.zshrc</div><div class="line">cp ~/.zshrc ~/.zshrc.orig</div><div class="line"># 拷贝.zshrc模板</div><div class="line">cp ~/.oh-my-zsh/templates/zshrc.zsh-template ~/.zshrc</div></pre></td></tr></table></figure>
<h3 id="修改-zshrc配置-Optional"><a href="#修改-zshrc配置-Optional" class="headerlink" title="修改.zshrc配置(Optional)"></a>修改.zshrc配置(Optional)</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div></pre></td><td class="code"><pre><div class="line"># oh-my-zsh目录 Path to your oh-my-zsh installation.</div><div class="line">export ZSH=$HOME/.oh-my-zsh</div><div class="line"></div><div class="line"># oh-my-zsh主题</div><div class="line">ZSH_THEME=&quot;michelebologna&quot;</div><div class="line"></div><div class="line">DISABLE_AUTO_UPDATE=&quot;true&quot;</div><div class="line">plugins=(git)</div><div class="line">export LC_ALL=C</div><div class="line"></div><div class="line">source $ZSH/oh-my-zsh.sh</div></pre></td></tr></table></figure>
<p><a href="https://github.com/robbyrussell/oh-my-zsh/wiki/Themes" target="_blank" rel="external">更多oh-my-zsh主题</a></p>
<h3 id="zshrc其他配置规则和-bashrc几乎一样"><a href="#zshrc其他配置规则和-bashrc几乎一样" class="headerlink" title=".zshrc其他配置规则和.bashrc几乎一样"></a>.zshrc其他配置规则和.bashrc几乎一样</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div></pre></td><td class="code"><pre><div class="line">#shortcut</div><div class="line">alias em=&apos;/usr/bin/emacs&apos;</div><div class="line">alias ll=&apos;ls -l&apos;</div><div class="line">alias la=&apos;ls -a&apos;</div><div class="line">alias les=&apos;less -S -x4&apos;</div><div class="line">alias ks=&apos;ls&apos;</div><div class="line">alias ch=&apos;chmod +x&apos;</div><div class="line">alias ee=&apos;exit&apos;</div><div class="line">alias tt=&apos;top -u wanglizhong&apos;</div><div class="line">alias sourcerc=&apos;source ~/.zshrc&apos;</div><div class="line">alias head1=&apos;head -1&apos;</div><div class="line">alias githelp=&apos;cat /home/wanglizhong/helpfile/git&apos;</div><div class="line"># tmux</div><div class="line">alias tls=&apos;tmux ls&apos;</div><div class="line">alias tnew=&apos;tmux new -s&apos;</div><div class="line">alias tatt=&apos;tmux att -t&apos;</div><div class="line">alias tkill=&apos;tmux kill-session -t&apos;</div><div class="line"># help</div><div class="line">alias find-help=&apos;cat /home/wanglizhong/tools/find&apos;</div><div class="line">alias awk-help=&apos;cat /home/wanglizhong/tools/awk&apos;</div><div class="line"># qsub</div><div class="line">alias qs=&apos;qstat|sort -t&quot; &quot; -k1 -n|les -N&apos;</div><div class="line">alias qd=&apos;qdel&apos;</div><div class="line">alias qj=&apos;qstat -j&apos;</div><div class="line"></div><div class="line"># PATH</div><div class="line">export R_LIBS=/ifshk7/BC_PS/wanglizhong/R:$R_LIBS # !!!!!</div><div class="line">export PERL5LIB=/ifshk7/BC_PS/wanglizhong/perl5:/home/wanglizhong/software/vcftools/vcftools-build/lib/perl5/site_perl/5.8.8:$PERL5LIB</div><div class="line">export LD_LIBRARY_PATH=/ifshk4/BC_PUB/biosoft/pipe/bc_paap/01.Usr/lib:/ifshk7/BC_PS/wanglizhong/lib64:/ifshk7/BC_PS/wanglizhong/lib:$LD_LIBRARY_PATH</div><div class="line">export PATH=/ifshk7/BC_PS/wanglizhong/bin:$PATH</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/06/06/Do-Phasing-with-SHAPEIT/" itemprop="url">
                  Do Phasing with SHAPEIT
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-06-06T14:55:02+08:00" content="2017-06-06">
              2017-06-06
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">export JAVA_HOME=/home/wanglizhong/software/java/jre1.8.0_45;</div><div class="line"></div><div class="line"># beagle was used to improve genotype calls using genotype likelihoods</div><div class="line">java -jar beagle/4.1/beagle.27Jul16.86a.jar gl=1.vcf.gz out=1.genotype;</div><div class="line"></div><div class="line"># phase with shapeit</div><div class="line">shapeit --thread 4 --effective-size 14269 --input-vcf 1.genotype.vcf.gz -M genetic_map_b37/genetic_map_chr1_combined_b37.txt --output-max 1.shapeit.haps 1.shapeit.sample --output-log 1.shapeit.log</div><div class="line"></div><div class="line"># Convert the SHAPEIT haplotypes to vcf formats</div><div class="line">shapeit -convert --input-haps 1.shapeit --output-vcf 1.shapeit.phased.vcf --output-log 1.shapeit.convert.log</div></pre></td></tr></table></figure>
<p><br></p>
<p><a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html" target="_blank" rel="external">more about shapeit</a></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/05/27/LD-prune-with-plink/" itemprop="url">
                  LD-prune with plink
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-05-27T15:07:51+08:00" content="2017-05-27">
              2017-05-27
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">plink-1.07/plink --noweb --tfile input --indep-pairwise 50 5 0.5 --out tmp1</div><div class="line">plink-1.07/plink --noweb --tfile input --extract tmp1.prune.in --recode --transpose --out output</div></pre></td></tr></table></figure>
<p><a href="http://zzz.bwh.harvard.edu/plink/summary.shtml#prune" target="_blank" rel="external">Linkage disequilibrium based SNP pruning</a></p>
<p><code>--indep-pairwise 50 5 0.5</code><br><code>--indep-pairwise 50 5 0.2</code></p>
<p>a) consider a window of 50 SNPs<br>b) calculate LD between each pair of SNPs in the window<br>b) remove one of a pair of SNPs if the LD is greater than 0.5<br>c) shift the window 5 SNPs forward and repeat the procedure</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/05/27/Ts-Tv/" itemprop="url">
                  Ts/Tv
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-05-27T14:57:11+08:00" content="2017-05-27">
              2017-05-27
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>The stricter the calls, the bigger ts/tv value one should get.</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">vcftools/vcftools-build/bin/vcftools --gzvcf input.vcf.gz --TsTv-summary --out input.gz.TsTv</div><div class="line"></div><div class="line">zcat input.vcf.gz| vcftools/vcftools-build/bin/vcf-tstv &gt; input.vcf.gz.tstv</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/05/02/Publications/" itemprop="url">
                  Publications
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-05-02T14:22:53+08:00" content="2017-05-02">
              2017-05-02
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="Publications"><a href="#Publications" class="headerlink" title="Publications"></a>Publications</h2><ul>
<li><p>Qiu Q*, <strong>Wang L</strong>*, Wang K*, Yang Y* et al: Yak whole-genome resequencing reveals domestication signatures and prehistoric population expansions. Nature Communications 2015, 6:10283. (*equal contribution in authorship)[<a href="https://www.nature.com/articles/ncomms10283" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Wang K*, <strong>Wang L</strong>*, Lenstra JA*, Jian J* et al: The genome sequence of the wisent (Bison bonasus). Gigascience 2017. (*equal contribution in authorship) [<a href="https://academic.oup.com/gigascience/article-lookup/doi/10.1093/gigascience/gix016" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Mei C*, Wang H*, Liao Q*, <strong>Wang L</strong>* et al: Genetic architecture and selection of Chinese cattle revealed by whole genome resequencing. Molecular Biology and Evolution 2017. (Accepted, *equal contribution in authorship)</p>
</li>
<li><p><strong>Wang L</strong>, Zhou H, Han J et al: Genome-scale transcriptome analysis of the alpine “glasshouse” plant Rheum nobile (Polygonaceae) with special translucent bracts. PloS one 2014, 9(10): e110712. [<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0110712" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Yang Y*, <strong>Wang L</strong>*, Han J et al: Comparative transcriptomic analysis revealed adaptation mechanism of Phrynocephalus erythrurus, the highest altitude Lizard living in the Qinghai-Tibet Plateau. BMC Evol Biol 2015, 15:101. (*equal contribution in authorship) [<a href="https://bmcevolbiol.biomedcentral.com/articles/10.1186/s12862-015-0371-8" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Liang C*, <strong>Wang L</strong>*, Wu X et al: Genome-wide association study identifies loci for the Polled phenotype in yak. PloS one 2016, 11(7): e0158642. (*equal contribution in authorship) [<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0158642" target="_blank" rel="external">link</a>]</p>
</li>
</ul>
<p><br></p>
<ul>
<li><p>Qiu Q, Zhang G, Ma T, Qian W, Wang J, Ye Z, …, <strong>Wang L</strong>, et al: The yak genome and adaptation to life at high altitude. Nature Genetics 2012, 44(8):946-949. [<a href="http://www.nature.com/ng/journal/v44/n8/full/ng.2343.html" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Wang K, Hu Q, Ma H, <strong>Wang L</strong> et al: Genome-wide variation within and between wild and domestic yak. Mol Ecol Resour 2014, 14(4):794-801. [<a href="http://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12226/abstract" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Wang K, Yang Y, <strong>Wang L</strong> et al: Different gene expressions between cattle and yak provide insights into high-altitude adaptation. Anim Genet 2015, Nov 5. [<a href="http://onlinelibrary.wiley.com/doi/10.1111/age.12377/full" target="_blank" rel="external">link</a>]</p>
</li>
<li><p>Zhang X, Wang K, <strong>Wang L</strong> et al: Genome-wide patterns of copy number variation in the Chinese yak genome. BMC Genomics 2016, 17(1):1-12. [<a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-2702-6" target="_blank" rel="external">link</a>]</p>
</li>
</ul>
<p>up to 2017.12.6</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/28/bwa-aln/" itemprop="url">
                  bwa-aln
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-28T14:08:42+08:00" content="2017-04-28">
              2017-04-28
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div></pre></td><td class="code"><pre><div class="line"># aln pipeline</div><div class="line">mkdir tmp</div><div class="line"># step1</div><div class="line">bwa aln -m 100000 -t 4 -i 15 -q 10 -f sampleA_1.sai reference.fa sampleA_1.fq.gz</div><div class="line">bwa aln -m 100000 -t 4 -i 15 -q 10 -f sampleA_2.sai reference.fa sampleA_2.fq.gz</div><div class="line"></div><div class="line"># step2</div><div class="line"># map and sort; unmapped标记成 mapQ=0；</div><div class="line">bwa sampe -a 600 -r &quot;@RG\tID:sampleA\tPL:illumina\tPU:sampleA\tLB:sampleA\tSM:sampleA&quot; reference.fa out_1.sai out_2.sai input_1.fq.gz input_2.fq.gz|awk &apos;&#123;if (!/^@/ &amp;&amp; and($2,4) == 4) &#123;$5=0; $6=&quot;*&quot;;&#125; gsub(/ /,&quot;\t&quot;,$0); print&#125;&apos; | samtools view -bS -T reference.fa - &gt; sampleA.bam</div><div class="line"></div><div class="line"># FixMateInformation and sort</div><div class="line">java -Xmx3g -Djava.io.tmpdir=tmp -XX:MaxPermSize=512m -XX:-UseGCOverheadLimit -jar picard.jar FixMateInformation I=sampleA.bam O=sampleA.sort.bam TMP_DIR=tmp SO=coordinate VALIDATION_STRINGENCY=SILENT</div><div class="line"></div><div class="line"># index</div><div class="line">samtools1.2 index sampleA.sort.bam</div><div class="line"></div><div class="line"># calmd, recalculate MD/NM tags and &apos;=&apos; bases</div><div class="line">samtools calmd -b sampleA.bam reference.fa &gt; sampleA.sort.bam</div><div class="line"></div><div class="line"># remove PCR duplicates</div><div class="line">java -Xmx2G -Djava.io.tmpdir=tmp -XX:MaxPermSize=512m -XX:-UseGCOverheadLimit -jar picard.jar MarkDuplicates MAX_FILE_HANDLES=1000 REMOVE_DUPLICATES=true I=sampleA.sort.bam O=sampleA.rmdup.bam METRICS_FILE=sampleA.txt TMP_DIR=tmp VALIDATION_STRINGENCY=SILENT</div><div class="line"></div><div class="line"># merge</div><div class="line">java -Xmx2G -Djava.io.tmpdir=tmp -XX:MaxPermSize=512m -XX:-UseGCOverheadLimit -jar picard.jar MergeSamFiles MAX_FILE_HANDLES=1000 I=sampleA01.rmdup.bam I=sampleA02.rmdup.bam I=sampleA03.rmdup.bam O=sampleA.all.bam TMP_DIR=tmp SO=coordinate AS=true VALIDATION_STRINGENCY=SILENT</div><div class="line"></div><div class="line"># index</div><div class="line">samtools index sampleA.all.bam</div><div class="line"></div><div class="line"># realign</div><div class="line">java -Xmx4g -Djava.io.tmpdir=tmp -XX:MaxPermSize=512m -XX:-UseGCOverheadLimit -jar GenomeAnalysisTK.jar -T RealignerTargetCreator -R reference.fa -I sampleA.all.bam -o sampleA.realn.intervals -B:snps,VCF dbsnp_132.hg19.vcf -B:indels,VCF 1000G_indels_for_realignment.hg19.vcf -l INFO</div><div class="line"></div><div class="line">java -Xmx4g -Djava.io.tmpdir=tmp -XX:MaxPermSize=512m -XX:-UseGCOverheadLimit -jar GenomeAnalysisTK.jar -T IndelRealigner -R reference.fa -I sampleA.all.bam -o sampleA.realn.bam -targetIntervals sampleA.realn.intervals -maxInMemory 300000 -l INFO</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/27/CNVnator-Based-Analysis/" itemprop="url">
                  CNVnator Based Analysis
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-27T18:05:14+08:00" content="2017-04-27">
              2017-04-27
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><a href="https://github.com/abyzovlab/CNVnator" target="_blank" rel="external">CNVnator</a></p>
<p>(Other software: Genome STRiP)</p>
<h1 id="Call-CNV-using-CNVnator"><a href="#Call-CNV-using-CNVnator" class="headerlink" title="Call CNV using CNVnator"></a>Call CNV using CNVnator</h1><h2 id="Running-example"><a href="#Running-example" class="headerlink" title="Running example"></a>Running example</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">export ROOTSYS=/path/to/software/root;</div><div class="line">export LD_LIBRARY_PATH=/path/to/software/root/lib:$LD_LIBRARY_PATH;</div><div class="line">cnvnator -root AGS10A.Chr1.50.root -unique -tree AGS10A.sorted.bam -chrom Chr1 &gt;AGS10A.Chr1.50.root.l1 2&gt;AGS10A.Chr1.50.root.e1;</div><div class="line">cnvnator -root AGS10A.Chr1.50.root -d split_chr_dir -his 50 -chrom Chr1 &gt;AGS10A.Chr1.50.root.l2 2&gt;AGS10A.Chr1.50.root.e2</div><div class="line">cnvnator -root AGS10A.Chr1.50.root -stat 50 -chrom Chr1 &gt;AGS10A.Chr1.50.root.l3 2&gt;AGS10A.Chr1.50.root.e3;</div><div class="line">cnvnator -root AGS10A.Chr1.50.root -partition 50 -chrom Chr1 &gt;AGS10A.Chr1.50.root.l4 2&gt;AGS10A.Chr1.50.root.e4</div><div class="line">cnvnator -root AGS10A.Chr1.50.root -call 50 -chrom Chr1 &gt; AGS10A.Chr1.50.cnv 2&gt;AGS10A.Chr1.50.root.e5</div></pre></td></tr></table></figure>
<h2 id="Result-format"><a href="#Result-format" class="headerlink" title="Result format:"></a>Result format:</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line">CNV_type  coordinates  CNV_size  normalized_RD  e-val1  e-val2  e-val3  e-val4  q0</div><div class="line"></div><div class="line">- normalized_RD -- normalized to 1.</div><div class="line">- e-val1        -- is calculated using t-test statistics.</div><div class="line">- e-val2        -- is from the probability of RD values within the region to be in</div><div class="line">the tails of a gaussian distribution describing frequencies of RD values in bins.</div><div class="line">- e-val3        -- same as e-val1 but for the middle of CNV</div><div class="line">- e-val4        -- same as e-val2 but for the middle of CNV</div><div class="line">- q0            -- fraction of reads mapped with q0 quality</div></pre></td></tr></table></figure>
<p><br></p>
<p><br></p>
<h1 id="Note"><a href="#Note" class="headerlink" title="Note"></a>Note</h1><ul>
<li><p>First thins you should do is try different bin_size settings like 100, 200, 500, 1000. And make sure that for the bin size you are using the ratio of average to its sd is around <strong>4-5</strong>. In the running log of <code>-stat</code> or <code>-eval</code>, something like this：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div></pre></td><td class="code"><pre><div class="line"># cnvnator -root scaffold100.root -chrom scaffold100 -stat 1000</div><div class="line">Making statistics for scaffold100 ...</div><div class="line">Average RD per bin (1-22) is 46.1805 +- 11.2085 (before GC correction)</div><div class="line">Average RD per bin (X,Y)  is 0 +- 0 (before GC correction)</div><div class="line">Correcting counts by GC-content for &apos;scaffold100&apos; ...</div><div class="line">Making statistics for scaffold100 after GC correction ...</div><div class="line">Average RD per bin (1-22) is 45.3823 +- 10.8865 (after GC correction)</div><div class="line">Average RD per bin (X,Y)  is 0 +- 0 (after GC correction)</div><div class="line"></div><div class="line"># OR</div><div class="line"># cnvnator -root scaffold100.root -chrom scaffold100 -eval 1000</div><div class="line">Average RD per bin (1-22) is 45.3823 +- 10.8865 4.16869</div><div class="line">Warning in &lt;Fit&gt;: Fit data is empty</div><div class="line">Warning in &lt;Fit&gt;: Fit data is empty</div><div class="line">Average RD per bin (X,Y)  is 0 +- 0 nan</div></pre></td></tr></table></figure>
</li>
<li><p>RAW cnv call的结果是normalized到1的深度, genotype的深度是normalized到2的深度。</p>
</li>
</ul>
<blockquote>
<p>The results of calling is RD signal normalized to 1, while genotyping returns estimated number of copies, which is 2 for all germline chromosomes except for X and Y in males.)</p>
</blockquote>
<ul>
<li>By default CNVnator estimates CN (<code>-genotype</code>, last two number in each line) using bin size that you provided and bins of <code>1000</code> bp. If histogram for 1k bp bins don’t exist then the software prints the warning and print -1 (the last field).</li>
</ul>
<ul>
<li>Larger events are more reliable. Also, regions with less repeats and segments duplication are more reliable (but this is taken care of by q0 filtering).</li>
</ul>
<ul>
<li><strong>First e-values is more conservative.</strong> You can ignore e-val3 and e-val4. By default CNVantor reports calls that are with &lt;0.05 either for e-val1 or e-val2.</li>
</ul>
<p><br><br><br></p>
<h1 id="Few-things-may-help"><a href="#Few-things-may-help" class="headerlink" title="Few things may help"></a>Few things may help</h1><ul>
<li><p><code>CNVnator_v0.2.7</code> and <code>root-5.34.36</code> is a safe chioce</p>
</li>
<li><p>the information about genome is taken from BAM header. Option <code>-genome</code> is only relevant when using with SAM files</p>
</li>
<li><p>looks like your file got corrupted. Not sure about the reason. If the issue persist, the walk around is to save histograms into a different .root file. e.g. <code>cnvnator -root tree.root -outroot his.root -his 200 -d /ref/sep/</code></p>
</li>
<li><p>by default CNVantor reports calls that are with &lt;0.05 either for e-val1 or e-val2.</p>
</li>
<li><p>dealing with scaffolds genome, you may want to remove few scaffold with abnormal RD and sd (and the abnormal ratio of average to its sd ).</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line"># for example, scaffold with too many repeat sequences</div><div class="line">Making statistics for scaffold1152 ...</div><div class="line">Average RD per bin (1-22) is 3.35436e-07 +- 96.0994 (before GC correction)</div><div class="line">Average RD per bin (X,Y)  is 0 +- 0 (before GC correction)</div><div class="line">Correcting counts by GC-content for &apos;scaffold1152&apos; ...</div><div class="line">Making statistics for scaffold1152 after GC correction ...</div><div class="line">Average RD per bin (1-22) is 3.29977e-07 +- 9.1335e-08 (after GC correction)</div><div class="line">Average RD per bin (X,Y)  is 0 +- 0 (after GC correction)</div></pre></td></tr></table></figure>
</li>
</ul>
<p><br></p>
<hr>
<h1 id="过滤参考"><a href="#过滤参考" class="headerlink" title="过滤参考"></a>过滤参考</h1><h2 id="参考CNV过滤标准："><a href="#参考CNV过滤标准：" class="headerlink" title="参考CNV过滤标准："></a>参考CNV过滤标准：</h2><ol>
<li><p>q0 &lt; 0.5, ie. calls with &gt;50% of q0 (zero mapping quality) reads within the CNV regions were removed (q0 filter）</p>
</li>
<li><p>CNV Calls &gt; 1kb (不必要)</p>
</li>
<li><p>P-value &lt;0.01 （e-val1 ）</p>
</li>
<li><p>不在基因组gap区域： it  does  not  <strong>overlap</strong> <strong>a gap</strong>  in  the  reference  genome  for  a  <strong>deletion</strong>,  and it is not <strong>within</strong> 0.5 Mb from a gap in the reference genome for a <strong>duplication</strong>; （nature11632-12-千人-s1）<em>/OR/</em> All CNV calls overlapping with gaps in the reference genome were excluded from consideration. (这个对于scaffolds的基因组太严了，尝试去掉50%以上)</p>
</li>
<li><p>重复序列区域： CNV calls that overlapped (more than 50% of their length) with masked regions were excluded from CNV analyses.</p>
</li>
<li><p>不好处理：a deletion must have at least two paired-end reads supporting the predictions  (overlapping  by  50%  reciprocally)  or  read  depth  in  the  middle  part  (1  kb  away  from  breakpoints)  of  the  called  region  should  be  statistically  different  from  average  read  depth.  Statistical  testing  was  done  the  same  way  as  when  calling  CNV regions. （nature11632-12-千人-s1）</p>
</li>
</ol>
<p><br></p>
<h2 id="千人SV过滤和merge方法（doi-10-1038-nature15394）"><a href="#千人SV过滤和merge方法（doi-10-1038-nature15394）" class="headerlink" title="千人SV过滤和merge方法（doi:10.1038/nature15394）"></a>千人SV过滤和merge方法（doi:10.1038/nature15394）</h2><blockquote>
<p>For  each  sample  we  subsequently  selected  confident  CNVnator  calls  as  follows:<br>1)  calls  having  <strong>paired-end</strong>  support;  (PE reads only in bam)<br>2)  calls  with  p-values  less  than  1e-5  (to  accounts  for  multiple  hypothesis testing, i.e., calling in ~2,500 samples), and with <strong>q0 &lt; 0.5</strong>;<br>3) deletion calls with  p-values less than 10^-5 and <strong>rd*(1 + q0) &lt; 0.75</strong> — whereby rd is the read-depth normalized  to  genome  average,  and  q0  is  fraction  of  reads  mapped  with  0  (zero)  mapping  quality.  </p>
<p>We  merged  CNV  calls  for  individuals  within  each  population. For  CNVnator  site  merging  we  initially  clustered  <strong>confident  overlapping  calls</strong>  and  <strong>averaged  coordinates  of  each  bound</strong>,  pursuing the merging initially by population and then across the entire sample set.</p>
</blockquote>
<p><br></p>
<h2 id="CNV处理方式"><a href="#CNV处理方式" class="headerlink" title="CNV处理方式"></a>CNV处理方式</h2><ol>
<li><p>不同个体间，大于50% reciprocal  overlap的CNV为一个CNV （0-GR-2011-CNVnator, doi:10.1101/gr.114876.110：  consider  two  calls  concordant  if  they  have  &gt;50%  reciprocal  overlap.  ）</p>
</li>
<li><p>For  each  population,  overlapping  calls  were  merged  by  selecting  the  largest  call  of  all  overlapping  ones.(to CNVR)<br>For comparisons of CNVs among multiple individuals, we merged all overlapping calls across individuals into unique CNVRs.</p>
</li>
</ol>
<p><br></p>
<h2 id="实际过滤情况"><a href="#实际过滤情况" class="headerlink" title="实际过滤情况"></a>实际过滤情况</h2><h3 id="实际使用过滤标准："><a href="#实际使用过滤标准：" class="headerlink" title="实际使用过滤标准："></a>实际使用过滤标准：</h3><ol>
<li><p>filter q0 &gt;= 0.5  </p>
</li>
<li><p>e-val1 &gt;= 0.01</p>
</li>
<li><p>deletion and duplication; dose not <strong>within</strong> a gap</p>
</li>
</ol>
<h3 id="实际使用个体间merge方法"><a href="#实际使用个体间merge方法" class="headerlink" title="实际使用个体间merge方法"></a>实际使用个体间merge方法</h3><ul>
<li>不同个体间，大于50% reciprocal overlap的CNV为一个CNV，平均两端的坐标（averaged  coordinates  of  each  bound）</li>
</ul>
<p><br></p>
<hr>
<h1 id="描述"><a href="#描述" class="headerlink" title="描述"></a>描述</h1><h2 id="相关定义"><a href="#相关定义" class="headerlink" title="相关定义"></a>相关定义</h2><ul>
<li>Please make sure that for the bin size you are using the ratio of average to its sd is around 4-5. （<strong>第一步挑选一个合适的bin_size</strong>）</li>
<li>most of false positives come from repetitive regions. So, the most confident calls are with q0 &lt; 0.5 (see README on how to generate it) （去除可能的重复序列区域结果）</li>
</ul>
<h2 id="CNV、genetic-CNV、CNV-gene定义"><a href="#CNV、genetic-CNV、CNV-gene定义" class="headerlink" title="CNV、genetic CNV、CNV gene定义"></a>CNV、genetic CNV、CNV gene定义</h2><p>(all from doi/10.1101/gr.187187.114.)</p>
<ul>
<li><strong>“CNVs”</strong> are all duplications or deletions ≥1 kb;</li>
<li><strong>“genic CNVs”</strong> are calls that contain at least ++one whole transcription unit++, based on the RefSeq- Gene database;</li>
<li>and <strong>“CNV genes”</strong> are transcription units that are completely contained within genic CNVs.</li>
<li>In addition, we use the term “CNV regions” (<strong>CNVRs</strong>) for genomic regions that include all partially or fully overlapping CNVs in any one of the analyzed animals. The borders of a CNVR are defined by the coordinates of the merged CNV calls across all individuals</li>
</ul>
<h3 id="描述数量"><a href="#描述数量" class="headerlink" title="描述数量"></a>描述数量</h3><p>Three  to  five thousand CNV calls were produced for each individual, ranging in size from 200  to  1,590,400  bp  (see  Figure  S3).  </p>
<h2 id="Vst分析-（千人SV文章-doi-10-1038-nature15394）"><a href="#Vst分析-（千人SV文章-doi-10-1038-nature15394）" class="headerlink" title="Vst分析 （千人SV文章, doi:10.1038/nature15394）"></a>Vst分析 （千人SV文章, doi:10.1038/nature15394）</h2><p><strong>计算方法</strong></p>
<p><img src="http://ofr9vioug.bkt.clouddn.com/vst_analysis.png?imageView/2/w/500" alt="image"></p>
<h2 id="copy-number-based-Vst"><a href="#copy-number-based-Vst" class="headerlink" title="copy number based Vst"></a>copy number based Vst</h2><ul>
<li>用copy number（CN）方差来反应群体之间的分化</li>
<li>取0.2作为阈值，鉴定高分化区域</li>
<li>去掉X染色体的</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/vst_table_display.png?imageView/2/w/500" alt="image"></p>
<p><br></p>
<hr>
<h1 id="个性化分析"><a href="#个性化分析" class="headerlink" title="个性化分析"></a>个性化分析</h1><h2 id="基本信息统计"><a href="#基本信息统计" class="headerlink" title="基本信息统计"></a>基本信息统计</h2><h3 id="基于个体的"><a href="#基于个体的" class="headerlink" title="基于个体的"></a>基于个体的</h3><ul>
<li><p>平均乘数、比对reads数、比对率、 All CNVs数量、Average CNV length、Genetic CNVs数量、CNV基因数量 （table）</p>
</li>
<li><p>统计presence and absence patterns of CNV alss between individuals(individuals within populations are more similar to each other than individuals between populations) (FigS5 家鼠 doi/10.1101/gr.187187.114.)</p>
</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/PairwiseCluster.png?imageView/2/w/600" alt="image"></p>
<ul>
<li>不同群体之间，两两个体overlap数（1bp，50%长度分别展示）</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/cnv_overlap_1bp_50percentage.png?imageView/2/w/600" alt="image"></p>
<h3 id="基于群体的"><a href="#基于群体的" class="headerlink" title="基于群体的"></a>基于群体的</h3><ul>
<li>CNV <strong>长度</strong>分布，分群体展示（FigS3 家鼠 doi/10.1101/gr.187187.114.)</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/CNV_size_distribution.png?imageView/2/w/600" alt="image"></p>
<ul>
<li>不同群体CNV calls/genetic CNVs/CNV genes<strong>数目</strong>，barplot (Fig1 家鼠 doi/10.1101/gr.187187.114.)</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">boxplot(CNV_number~sample_id,data=a)</div><div class="line">boxplot(a$V1,b$V1,outline = <span class="literal">F</span>,names=c(<span class="string">"A"</span>,<span class="string">"B"</span>),main=<span class="string">"title"</span>)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/CNVnumberBarplot?imageView/2/w/600" alt="image"></p>
<ul>
<li>不同群体share情况</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/cnv-human.png?imageView/2/w/400" alt="image"></p>
<p>share</p>
<p><img src="http://ofr9vioug.bkt.clouddn.com/CNV_share_within_without_pops.png?imageView/2/w/500" alt="image"></p>
<p>private</p>
<p><img src="http://ofr9vioug.bkt.clouddn.com/cnv_private.png?imageView/2/w/500" alt="image"></p>
<ul>
<li>和基因组不同区域的overlap情况</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/overlap_with_genomic_elements.png?imageView/2/w/500" alt="image"></p>
<blockquote>
<p>Extended Data Figure 7 | Enrichment of functional elements intersecting SVs. a, Shadow figure of Fig. 2a. Overlap enrichment analysis of deletions (with resolved breakpoints) versus genomic elements, using partial overlap statistic, deletions categorized into VAF bins. b, Similar to a. The only difference is that engulf overlap statistic is used instead of partial overlap statistic. Engulf overlap statistic is the count of genomic elements (for example, CDS) that are fully imbedded in at least one SV interval (for example, deletions). *no element intersected observed within data set. c, Similar to a and b, with the enrichment/depletion analysis pursued for common SNPs as well as more rare single nucleotide polymorphisms/variants (SNVs). Common SNV alleles show the highest levels of depletion for investigated genomic elements. d, Overlap enrichment analysis of various SV types versus genomic elements, using partial overlap statistic.</p>
</blockquote>
<ul>
<li><p>cnv长度和拷贝数的关系？</p>
</li>
<li><p>CNV频率分布（CNV Allele Frequency Distributions）对选择的影响（doi:10.1371/journal.pgen.1004830）</p>
</li>
</ul>
<blockquote>
<p>To test for the influence of selection on the allele frequency spectrum of CNVs, we used a Poisson Random Field approach implemented in <strong>prfreq</strong>, which evaluates the expected allele frequency distributions given different demographic and selection models. We compared the estimates of the scaled selection coefficient c of CNVs (both deletions and duplications) with intergenic SNPs, by fitting the ‘‘single point mass’’ model over a range of c (between 220 and 10). Estimates were taken modeling a stationary population, under the assumption that demographics affect intergenic SNPs and CNVs in a similar fashion.</p>
</blockquote>
<ul>
<li>cnv 注释，不同区域的数量、比例，用比例可以不然不同群体间/不同类型（Deletion/Duplication)数量差异在图上太明显</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/cnv-annot.png?imageView/2/w/500" alt="image"></p>
<p><br></p>
<p><br></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/18/Insert-blank-line-using-markdown/" itemprop="url">
                  Insert blank line using markdown
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-18T10:21:10+08:00" content="2017-04-18">
              2017-04-18
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>markdown里插入空行的方法，两个可用的:<br><code>&lt;br /&gt;</code><br><code>&lt;br&gt;&lt;/br&gt;</code></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/18/admixtools/" itemprop="url">
                  admixtools FAQs
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-18T09:40:20+08:00" content="2017-04-18">
              2017-04-18
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="常见的和格式有关的问题：more"><a href="#常见的和格式有关的问题：more" class="headerlink" title="常见的和格式有关的问题：more"></a>常见的和格式有关的问题：<a href="https://www.hsph.harvard.edu/alkes-price/eigensoft-frequently-asked-questions/" target="_blank" rel="external">more</a></h1><p><br></p>
<h2 id="ped、fam格式第六列的格式问题"><a href="#ped、fam格式第六列的格式问题" class="headerlink" title="ped、fam格式第六列的格式问题"></a>ped、fam格式第六列的格式问题</h2><p>Question: convertf decides to “ignore” all my samples. Why?</p>
<blockquote>
<p>Answer: A likely reason is that you are using a “fam” or “ped” file with a funny value (0, 9 or -9) in column 6. Try setting column 6 to 1.</p>
</blockquote>
<p><br></p>
<h2 id="推荐去除LD先"><a href="#推荐去除LD先" class="headerlink" title="推荐去除LD先 "></a>推荐去除LD先 <br></h2><p>Question: Should regions of long-range LD in the genome be removed prior to PCA?</p>
<blockquote>
<p>Answer: Yes, to avoid principal components that are artifacts of long-range LD it is ideal to remove such regions. See Table 1 of Price et al. 2008 AJHG. However, EIGENSTRAT can subsequently be run to compute disease association statistics using the full set of SNPs.</p>
</blockquote>
<p><br></p>
<h2 id="仅支持有限的SNP数量"><a href="#仅支持有限的SNP数量" class="headerlink" title="仅支持有限的SNP数量"></a>仅支持有限的SNP数量</h2><p>Question: Can I run EIGENSOFT on very large data sets? <br></p>
<blockquote>
<p>Answer: Yes. We currently support GWAS data sets up to 8 billion genotypes. For data sets between 2 billion and 8 billion genotypes, some care is required. See documentation for details.</p>
</blockquote>
<p><br></p>
<h2 id="命名太长的问题"><a href="#命名太长的问题" class="headerlink" title="命名太长的问题"></a>命名太长的问题</h2><p>Question: When I run I get an error message about “idnames too long”. What should I do?</p>
<blockquote>
<p>Answer: The software supports sample ID names up to a max of 39 characters. Longer sample ID names must be shortened. In addition, if your data is in PED format, the default is to concatenate the family ID and sample ID names so that their total length must meet this limit; however, you can set “familynames: NO” so that only the sample ID name will be used and must meet the 39 character limit.</p>
</blockquote>
<p><br></p>
<h2 id="只取一部分群体"><a href="#只取一部分群体" class="headerlink" title="只取一部分群体"></a>只取一部分群体</h2><p>Question: How do I compute principal components using only a subset of populations and project other populations onto those principal components?</p>
<blockquote>
<p>Answer: Use -w flag to smartpca.perl (see EIGENSTRAT/README), or use poplistname parameter to smartpca (see POPGEN/README).</p>
</blockquote>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/17/Getting-free-images-with-copyright/" itemprop="url">
                  Getting free images with copyright
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-17T16:54:21+08:00" content="2017-04-17">
              2017-04-17
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><a href="http://www.publicdomainpictures.net/" target="_blank" rel="external">PublicDomainPictures</a></p>
<p><a href="https://unsplash.com/" target="_blank" rel="external">Unsplash</a></p>
<p><a href="https://pixabay.com/" target="_blank" rel="external">Pixabay</a></p>
<p><a href="https://librestock.com/" target="_blank" rel="external">LibreStock</a></p>
<p><a href="https://www.pexels.com/" target="_blank" rel="external">pexels</a></p>
<p><a href="https://visualhunt.com/" target="_blank" rel="external">VisualHunt</a> 可以按颜色筛选</p>
<p><a href="http://www.ssyer.com/" target="_blank" rel="external">ssyer</a></p>
<p><a href="http://www.gratisography.com/" target="_blank" rel="external">gratisography</a></p>
<p><a href="http://streetwill.co/" target="_blank" rel="external">streetwill</a> 小众、街拍、风景</p>
<p><a href="http://fancycrave.com/" target="_blank" rel="external">fancycrave</a> 照片背后的故事</p>
<p><a href="www.shutterstock.com">shutterstock</a></p>
<p><a href="http://stokpic.com/" target="_blank" rel="external">stokpic</a></p>
<p><a href="finda.photo">finda</a></p>
<p><a href="www.gettyimages.com">gettyimages</a></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/12/vcftools/" itemprop="url">
                  vcftools
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-12T19:12:08+08:00" content="2017-04-12">
              2017-04-12
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="习惯：Always使用bgzip生成vcf-gz文件"><a href="#习惯：Always使用bgzip生成vcf-gz文件" class="headerlink" title="习惯：Always使用bgzip生成vcf.gz文件"></a>习惯：Always使用bgzip生成vcf.gz文件</h2><p>并用tabix生成index，使得vcftools和GATK中直接使用vcf.gz文件。</p>
<p>bgzip和tabix包含在<code>samtools/htslib</code>内。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">samtools/htslib/bgzip -c file.vcf &gt; file.vcf.gz</div><div class="line">samtools/htslib/tabix -p vcf file.vcf.gz</div></pre></td></tr></table></figure></p>
<hr>
<p><strong>小模块</strong><br>在<code>vcftools/vcftools-build/bin/</code>中的小程序，实现简单的特定功能。列出几个最常用的。<a href="http://vcftools.sourceforge.net/perl_module.html" target="_blank" rel="external">其他更多</a></p>
<p>使用前需要导入perl_module lib<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">export PERL5LIB=/home/wanglizhong/software/vcftools/vcftools-build/lib/perl5/site_perl/5.8.8:$PERL5LIB;</div></pre></td></tr></table></figure></p>
<h2 id="单个vcf"><a href="#单个vcf" class="headerlink" title="单个vcf"></a>单个vcf</h2><h3 id="重新计算INFO里的AN和AC"><a href="#重新计算INFO里的AN和AC" class="headerlink" title="重新计算INFO里的AN和AC"></a>重新计算<code>INFO</code>里的AN和AC</h3><p>fill-an-ac<br>重新计算AN和AC 并加入<code>INFO</code>中。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">zcat in.vcf.gz|fill-an-ac &gt; out.vcf</div></pre></td></tr></table></figure></p>
<p><br></p>
<h3 id="vcf2fa"><a href="#vcf2fa" class="headerlink" title="vcf2fa"></a>vcf2fa</h3><p>vcf转成一致性序列(consensus sequence)。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">cat ref.fa|vcf-consensus file.vcf.gz &gt; out.fa</div><div class="line"></div><div class="line"># 其中某一个个体</div><div class="line">cat ref.fa|vcf-consensus -s sampleA_ID file.vcf.gz &gt; out.sampleA_ID.fa</div><div class="line"></div><div class="line"># 只有某一个haplotype（1或者2）</div><div class="line">cat ref.fa|vcf-consensus -H 1 file.vcf.gz &gt; out.hap1.fa</div></pre></td></tr></table></figure></p>
<p><br></p>
<h3 id="取出部分个体"><a href="#取出部分个体" class="headerlink" title="取出部分个体"></a>取出部分个体</h3><p>vcf-subset -c<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">vcf-subset -c NA0001,NA0002 file.vcf.gz | bgzip -c &gt; out.vcf.gz</div></pre></td></tr></table></figure></p>
<p><br></p>
<h3 id="vcf里的信息自定义输出"><a href="#vcf里的信息自定义输出" class="headerlink" title="vcf里的信息自定义输出"></a>vcf里的信息自定义输出</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">vcf-query file.vcf.gz 1:10327-10330</div><div class="line">vcf-query file.vcf -f &apos;%CHROM:%POS %REF %ALT [ %DP]\n&apos;</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div></pre></td><td class="code"><pre><div class="line">Usage: vcf-query [OPTIONS] file.vcf.gz</div><div class="line">Options:</div><div class="line">   -c, --columns &lt;file|list&gt;           List of comma-separated column names or one column name per line in a file.</div><div class="line">   -f, --format &lt;string&gt;               The default is &apos;%CHROM:%POS\t%REF[\t%SAMPLE=%GT]\n&apos;</div><div class="line">   -l, --list-columns                  List columns.</div><div class="line">   -r, --region chr:from-to            Retrieve the region. (Runs tabix.)</div><div class="line">       --use-old-method                Use old version of API, which is slow but more robust.</div><div class="line">   -h, -?, --help                      This help message.</div><div class="line">Expressions:</div><div class="line">   %CHROM          The CHROM column (similarly also other columns)</div><div class="line">   %GT             Translated genotype (e.g. C/A)</div><div class="line">   %GTR            Raw genotype (e.g. 0/1)</div><div class="line">   %INFO/TAG       Any tag in the INFO column</div><div class="line">   %LINE           Prints the whole line</div><div class="line">   %SAMPLE         Sample name</div><div class="line">   []              The brackets loop over all samples</div><div class="line">   %*&lt;A&gt;&lt;B&gt;        All format fields printed as KEY&lt;A&gt;VALUE&lt;B&gt;</div><div class="line">Examples:</div><div class="line">   vcf-query file.vcf.gz 1:1000-2000 -c NA001,NA002,NA003</div><div class="line">   vcf-query file.vcf.gz -r 1:1000-2000 -f &apos;%CHROM:%POS\t%REF\t%ALT[\t%SAMPLE:%*=,]\n&apos;</div><div class="line">   vcf-query file.vcf.gz -f &apos;[%GT\t]%LINE\n&apos;</div><div class="line">   vcf-query file.vcf.gz -f &apos;[%GT\ ]%LINE\n&apos;</div><div class="line">   vcf-query file.vcf.gz -f &apos;%CHROM\_%POS\t%INFO/DP\t%FILTER\n&apos;</div></pre></td></tr></table></figure>
<p><br></p>
<h3 id="将个体顺序重排"><a href="#将个体顺序重排" class="headerlink" title="将个体顺序重排"></a>将个体顺序重排</h3><p>Reorder columns</p>
<p>将<code>file.vcf.gz</code>按照<code>template.vcf.gz</code>顺序重排<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">vcf-shuffle-cols -t template.vcf.gz file.vcf.gz &gt; out.vcf</div></pre></td></tr></table></figure></p>
<h3 id="vcf-postions-Reorder"><a href="#vcf-postions-Reorder" class="headerlink" title="vcf postions Reorder"></a>vcf postions Reorder</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">vcf-sort file.vcf.gz &gt; out.vcf</div><div class="line">zcat file.vcf.gz| vcf-sort &gt; out.vcf</div></pre></td></tr></table></figure>
<p><br></p>
<h3 id="转换成tab格式"><a href="#转换成tab格式" class="headerlink" title="转换成tab格式"></a>转换成tab格式</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">zcat file.vcf.gz | vcf-to-tab &gt; out.tab</div></pre></td></tr></table></figure>
<p>输出如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">#CHROM  POS REF 05537</div><div class="line">chr2    131 T   T/T</div><div class="line">chr2    437 G   G/G</div><div class="line">chr2    453 G   G/G</div><div class="line">chr2    469 G   G/G</div><div class="line">chr2    526 G   G/G</div><div class="line">chr2    618 G   G/G</div><div class="line">chr2    745 A   A/A</div><div class="line">chr2    756 T   T/T</div><div class="line">chr2    760 T   T/T</div></pre></td></tr></table></figure></p>
<p><br></p>
<h3 id="计算tstv"><a href="#计算tstv" class="headerlink" title="计算tstv"></a>计算tstv</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">cat file.vcf | vcf-tstv</div></pre></td></tr></table></figure>
<p><br></p>
<h3 id="vcf格式校验"><a href="#vcf格式校验" class="headerlink" title="vcf格式校验"></a>vcf格式校验</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">vcf-validator file.vcf.gz</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="多个vcf"><a href="#多个vcf" class="headerlink" title="多个vcf"></a>多个vcf</h2><h3 id="比较多个vcf里SNP位置的overlap"><a href="#比较多个vcf里SNP位置的overlap" class="headerlink" title="比较多个vcf里SNP位置的overlap"></a>比较多个vcf里SNP位置的overlap</h3><p>可以用结果画维恩图Venn-Diagram。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line"># run</div><div class="line">vcf-compare test.vcf.gz test2.vcf.gz test3.vcf.gz|grep ^VN | cut -f 2-</div><div class="line"></div><div class="line"># results</div><div class="line">VN      94      test.vcf.gz (9.5%)      test2.vcf.gz (22.3%)    test3.vcf.gz (48.5%)</div><div class="line">VN      100     test.vcf.gz (10.1%)     test3.vcf.gz (51.5%)</div><div class="line">VN      327     test.vcf.gz (32.9%)     test2.vcf.gz (77.7%)</div><div class="line">VN      473     test.vcf.gz (47.6%)</div></pre></td></tr></table></figure></p>
<p><br></p>
<h3 id="合并多个染色体chr-scaffold"><a href="#合并多个染色体chr-scaffold" class="headerlink" title="合并多个染色体chr/scaffold"></a>合并多个染色体chr/scaffold</h3><p>1.个体（header）需完全相同。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">vcf-concat chr1.vcf.gz chr2.vcf.gz | bgzip -c &gt; all.vcf.gz</div></pre></td></tr></table></figure></p>
<p>2.合并有不同个体的多个vcf</p>
<ul>
<li><p>最好有相同Postions，缺失的默认会被填上miss（.）</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">vcf-merge A.vcf.gz B.vcf.gz C.vcf.gz | bgzip -c &gt; out.vcf.gz</div></pre></td></tr></table></figure>
</li>
<li><p>把mis的用自定义genotype（0|0，0/0，1/1等等）填上</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">vcf-merge -R &apos;0|0&apos; A.vcf.gz B.vcf.gz C.vcf.gz | bgzip -c &gt; out.vcf.gz</div></pre></td></tr></table></figure>
</li>
<li><p>bcftools</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtools/bcftools/bcftools merge --merge all --no-version --threads 10 file1.vcf.gz file2.vcf.gz |bgzip -c &gt; merge.vcf.gz</div></pre></td></tr></table></figure>
</li>
</ul>
<p>3.一行命令合并vcf，巧用括号<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"># Merge (that is, concatenate) two VCF files into one, keeping the header</div><div class="line"># from first one only.</div><div class="line">(zcat A.vcf.gz | grep ^#; zcat A.vcf.gz | grep -v ^#; zcat B.vcf.gz | grep -v ^#; )| bgzip -c &gt; out.vcf.gz</div></pre></td></tr></table></figure></p>
<p><br></p>
<h3 id="找vcf-overlap-互补位置"><a href="#找vcf-overlap-互补位置" class="headerlink" title="找vcf overlap/互补位置"></a>找vcf overlap/互补位置</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line"># 输出两个vcf里共有的</div><div class="line">vcf-isec -n +2 A.vcf.gz B.vcf.gz | bgzip -c &gt; out.vcf.gz</div><div class="line"></div><div class="line"># 输出第一个vcf里有，其他vcf（B和C及后面更多）里没有的</div><div class="line">vcf-isec -c A.vcf.gz B.vcf.gz C.vcf.gz | bgzip -c &gt; out.vcf.gz</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/04/06/bam2fq/" itemprop="url">
                  bam2fq
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-04-06T14:58:39+08:00" content="2017-04-06">
              2017-04-06
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="samtools-bam2fq"><a href="#samtools-bam2fq" class="headerlink" title="samtools bam2fq"></a>samtools bam2fq</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtools.1.3.1 bam2fq input.bam |gzip -c &gt; output.fq.gz</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="picard-SamToFastq"><a href="#picard-SamToFastq" class="headerlink" title="picard SamToFastq"></a>picard SamToFastq</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">java -jar picard-tools-2.5.0/picard.jar SamToFastq I=input.bam F=out.1.fq.gz F2=out.2.fq.gz</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="提取-unmapped-reads"><a href="#提取-unmapped-reads" class="headerlink" title="提取 unmapped reads"></a>提取 unmapped reads</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># 提取unmapped reads比对到另一个ref（比如线粒体）上</div><div class="line">samtools.1.3.1 bam2fq -f 4 input.bam | bwa mem -t 30 -M -R &apos;@RG\tID:Sample01\tLB:Sample01\tSM:Sample01\tPL:Illumina\tPU:Illumina\tSM:Sample01\t&apos; mt.fa - | samtools.1.3.1 sort -O bam -T /dir/tmp -o mapMt.sort.bam</div></pre></td></tr></table></figure>
<p><br></p>
<h2 id="提取-unique-reads"><a href="#提取-unique-reads" class="headerlink" title="提取 unique reads"></a>提取 unique reads</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"># unique reads (a single read mapping at one best position)</div><div class="line"># -q INT Skip alignments with MAPQ smaller than INT [0]</div><div class="line">samtools view -bq 1 file.bam &gt; unique.bam</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/03/23/GATK-SNP-calling/" itemprop="url">
                  GATK SNP calling
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-03-23T12:59:39+08:00" content="2017-03-23">
              2017-03-23
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="1-gatk-calling"><a href="#1-gatk-calling" class="headerlink" title="1. gatk calling"></a>1. gatk calling</h2><h3 id="1-1-HaplotypeCaller"><a href="#1-1-HaplotypeCaller" class="headerlink" title="1.1 HaplotypeCaller"></a>1.1 HaplotypeCaller</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">java -Xmx3g -jar GenomeAnalysisTK-3.4-0/GenomeAnalysisTK.jar -R ref.fa -T HaplotypeCaller \</div><div class="line">-nct 3 \</div><div class="line">--emitRefConfidence GVCF \</div><div class="line">--variant_index_type LINEAR \</div><div class="line">--variant_index_parameter 128000 \</div><div class="line">-I in.bam \</div><div class="line">-L Chr1 \</div><div class="line">-o out.g.vcf.gz</div></pre></td></tr></table></figure>
<h3 id="1-2-GenotypeGVCFs"><a href="#1-2-GenotypeGVCFs" class="headerlink" title="1.2 GenotypeGVCFs"></a>1.2 GenotypeGVCFs</h3><p>NB. Chr1.gvcf.list: gvcf list, one individual per line<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line">java -Xmx6g -Djava.io.tmpdir=/tmp -jar GenomeAnalysisTK-3.4-0/GenomeAnalysisTK.jar -T GenotypeGVCFs \</div><div class="line">-R ref.fa \</div><div class="line">-nt 4 \</div><div class="line">--variant Chr1.gvcf.list \</div><div class="line">--dbsnp dbsnp.vcf \</div><div class="line">-stand_call_conf 30.0 \</div><div class="line">-stand_emit_conf 10.0 \</div><div class="line">-L Chr1 \</div><div class="line">-o raw.Chr1.vcf</div></pre></td></tr></table></figure></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"># snp SelectVariants</div><div class="line">java -Xmx2g -Djava.io.tmpdir=/tmp -jar GenomeAnalysisTK-3.4-0/GenomeAnalysisTK.jar -T SelectVariants \</div><div class="line">-R ref.fa \</div><div class="line">--variant raw.Chr1.vcf \</div><div class="line">-selectType SNP \</div><div class="line">-o raw.Chr1.snp.vcf</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"># indel SelectVariants</div><div class="line">java -Xmx2g -Djava.io.tmpdir=/tmp -jar GenomeAnalysisTK-3.4-0/GenomeAnalysisTK.jar -T SelectVariants \</div><div class="line">-R ref.fa  \</div><div class="line">--variant raw.Chr1.vcf \</div><div class="line">-selectType INDEL \</div><div class="line">-o raw.Chr1.indel.vcf</div></pre></td></tr></table></figure>
<hr>
<h2 id="2-samtools-calling"><a href="#2-samtools-calling" class="headerlink" title="2. samtools calling"></a>2. samtools calling</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"># samtools</div><div class="line">samtools mpileup -b bam.list -uf ref.fa -r Chr1 -DS -C50 -m 2 -F 0.002 -P ILLUMINA  \</div><div class="line">| bcftools view -p 0.99 -bcvg - | bcftools view  \</div><div class="line">| vcfutils.pl varFilter -1 1e-5 -4 1e-7 -d 2 -Q 10 -a 2 -w 3  | bgzip -c &gt; Chr1.flt.vcf.gz</div><div class="line"></div><div class="line">tabix -f Chr1.flt.vcf.gz</div></pre></td></tr></table></figure>
<hr>
<h2 id="3-confidence-snp-dataset-with-overlap-of-gatk-and-samtools；SNPQ-gt-30"><a href="#3-confidence-snp-dataset-with-overlap-of-gatk-and-samtools；SNPQ-gt-30" class="headerlink" title="3. confidence snp dataset with overlap of gatk and samtools；SNPQ&gt;30"></a>3. confidence snp dataset with overlap of gatk and samtools；SNPQ&gt;30</h2><p>Chr1.overlap.snp.vcf.gz</p>
<hr>
<h2 id="4-final-SNP"><a href="#4-final-SNP" class="headerlink" title="4. final SNP"></a>4. final SNP</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div></pre></td><td class="code"><pre><div class="line"># VariantRecalibrator &amp; ApplyRecalibration</div><div class="line">java -Xmx9g -Djava.io.tmpdir=/tmp -jar GenomeAnalysisTK-3.4-0/GenomeAnalysisTK.jar -T VariantRecalibrator \</div><div class="line">-l INFO \</div><div class="line">-R ref.fa  \</div><div class="line">-input raw.Chr1.snp.vcf.gz \</div><div class="line">-resource:concordantSet,VCF,known=true,training=true,truth=true,prior=10.0 Chr1.overlap.snp.vcf \</div><div class="line">-resource:dbsnp,known=true,training=false,truth=false,prior=2.0 dbSNP/dbsnp.vcf \</div><div class="line">-an QD  -an MQRankSum -an ReadPosRankSum -an FS -an MQ -an DP \</div><div class="line">-an InbreedingCoeff \</div><div class="line">-recalFile gatk.SNP.recal \</div><div class="line">-tranchesFile gatk.SNP.tranches \</div><div class="line">-rscriptFile gatk.SNP.plot.R \</div><div class="line">--TStranche 90.0 --TStranche 93.0 --TStranche 95.0 \</div><div class="line">--TStranche 97.0 --TStranche 99.0 --TStranche 100.0 \</div><div class="line">-mode SNP \</div><div class="line">-L Chr1</div><div class="line"></div><div class="line"></div><div class="line">java -Xmx6g -Djava.io.tmpdir=/tmp -jar GenomeAnalysisTK-3.4-0/GenomeAnalysisTK.jar -T ApplyRecalibration \</div><div class="line">-l INFO \</div><div class="line">-R ref.fa \</div><div class="line">-input raw.Chr1.snp.vcf.gz \</div><div class="line">--ts_filter_level 99.0 \</div><div class="line">-recalFile gatk.SNP.recal \</div><div class="line">-tranchesFile gatk.SNP.tranches \</div><div class="line">-mode SNP \</div><div class="line">-L Chr1 \</div><div class="line">-o gatk.snp.Chr1.VQSR.vcf</div><div class="line"></div><div class="line">grep -E &apos;PASS|#&apos; gatk.snp.Chr1.VQSR.vcf &gt;final.gatk.snp.Chr1.VQSR.vcf</div><div class="line">perl tabix.pl gatk.snp.Chr1.VQSR.vcf</div><div class="line">perl tabix.pl final.gatk.snp.Chr1.VQSR.vcf</div></pre></td></tr></table></figure>
<hr>
<h2 id="5-use-Beagle-to-inprove-genotype-calls-using-genotype-likelihoods-from-SAMtools-GATK"><a href="#5-use-Beagle-to-inprove-genotype-calls-using-genotype-likelihoods-from-SAMtools-GATK" class="headerlink" title="5. use Beagle to inprove genotype calls using genotype likelihoods from SAMtools/GATK"></a>5. use Beagle to inprove genotype calls using genotype likelihoods from SAMtools/GATK</h2><p>用Beagle，利用GL信息做genotype<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">java -Xmx3g -jar beagle/4.1/beagle.27Jul16.86a.jar \</div><div class="line">gl=input.vcf.gz \</div><div class="line">out=output.genotype</div></pre></td></tr></table></figure></p>
<h2 id="6-phase-with-beagle"><a href="#6-phase-with-beagle" class="headerlink" title="6. phase with beagle"></a>6. phase with beagle</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">java -Xmx3g -jar beagle/4.1/beagle.27Jul16.86a.jar \</div><div class="line">gt=output.genotype.vcf.gz \</div><div class="line">out=output.phased.vcf.gz</div></pre></td></tr></table></figure>
<h2 id="7-phase-with-shapeit-human"><a href="#7-phase-with-shapeit-human" class="headerlink" title="7. phase with shapeit (human)"></a>7. phase with shapeit (human)</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">shapeit --thread 4 --effective-size 14269 --input-vcf chr1.genotype.vcf.gz \</div><div class="line">-M 1kg/phasing/genetic_map_b37/genetic_map_chr1_combined_b37.txt \</div><div class="line">--output-max chr1.shapeit.haps.gz chr1.shapeit.haps.sample --output-log chr1.shapeit.log</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/02/12/PPT/" itemprop="url">
                  PPT
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-02-12T10:43:10+08:00" content="2017-02-12">
              2017-02-12
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h2 id="字体"><a href="#字体" class="headerlink" title="字体"></a>字体</h2><p>中文字体：<br>微软雅黑，方正兰亭黑简体，华文黑体，冬青黑体<br>雅痞-简</p>
<p>娃娃体，方正呐喊，小麦体，翩翩体 (娱乐化)</p>
<p>英文字体：<br>英文一般无衬线字体(san-serif font)，不小于28号<br>Verdana(标题)<br>Franklin Gothic Medium (标题)<br>Comic Sans MS (标题)<br>Arial (正文、标题)<br>Times New Roman (正文)<br>Palatino (正文)</p>
<h2 id="强调"><a href="#强调" class="headerlink" title="强调"></a>强调</h2><p>用黑体(bold)和斜体(italic)来强调，不要使用下划线或者其他字体</p>
<h2 id="字体大小"><a href="#字体大小" class="headerlink" title="字体大小"></a>字体大小</h2><p>对于海报来说，有一个字体大小的简单规则：</p>
<blockquote>
<p>To be easily readable the smallest font size, in points, should beat least 4 times the viewing distance in feet.</p>
<p>For example for viewing at 6 feet (4 x 6 = 24) the smallest fontsize you should use is 24 point. (this formula assumes goodeyesight so use larger sizes if possible)</p>
</blockquote>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2017/01/01/R-Cheetsheets/" itemprop="url">
                  R Cheetsheets
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2017-01-01T10:40:41+08:00" content="2017-01-01">
              2017-01-01
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="基本"><a href="#基本" class="headerlink" title="基本"></a>基本</h1><h2 id="基本元素、函数"><a href="#基本元素、函数" class="headerlink" title="基本元素、函数"></a>基本元素、函数</h2><h3 id="基本元素"><a href="#基本元素" class="headerlink" title="基本元素"></a>基本元素</h3><table>
<thead>
<tr>
<th>函数名</th>
<th>功能</th>
</tr>
</thead>
<tbody>
<tr>
<td>points(x,y)</td>
<td>添加点</td>
</tr>
<tr>
<td>lines(x,y)</td>
<td>添加线</td>
</tr>
<tr>
<td>text(x,y,labels,…)</td>
<td>在(x,y)处添加用labels指定的文字</td>
</tr>
<tr>
<td>mtext(text,side=3,line=0,…)</td>
<td>在边空添加用text指定的文字</td>
</tr>
<tr>
<td>segments(x0,y0,x1,y1)</td>
<td>从(x0,y0)各点到(x1,y1)各点画线段</td>
</tr>
<tr>
<td>arrows(x0,y0,x1,y1,…)</td>
<td>同上，但添加箭头</td>
</tr>
<tr>
<td>abline(a,b)</td>
<td>绘制斜率为b和截距为a的直线</td>
</tr>
<tr>
<td>abline(h=y)</td>
<td>在纵坐标y处画水平线, y可以是向量, abline(h=c(1,2,3))</td>
</tr>
<tr>
<td>abline(v=x)</td>
<td>在横坐标x处画垂直线, x可以是向量</td>
</tr>
<tr>
<td>abline(lm.obj)</td>
<td>画出lm.obj确定的回归线</td>
</tr>
<tr>
<td>rect(x1,y1,x2,y2)</td>
<td>绘制长方形，(x1,y1)为左下角，(x2,y2)为右上角</td>
</tr>
<tr>
<td>polygon(x,y)</td>
<td>绘制连接各x,y坐标确定的点的多边形</td>
</tr>
<tr>
<td>legend(x,y,legend)</td>
<td>在点(x,y)处添加图例，说明内容由legend给定</td>
</tr>
<tr>
<td>title()</td>
<td>添加标题，也可添加一个副标题</td>
</tr>
<tr>
<td>axis(side,vect)</td>
<td>画坐标轴</td>
</tr>
<tr>
<td>box()</td>
<td>在当前的图上加边框</td>
</tr>
<tr>
<td>rug(x)</td>
<td>在x轴上用短线画出x数据的位置</td>
</tr>
<tr>
<td>locator(n,type=”n”,…)</td>
<td>获取鼠标在图中点击处的坐标</td>
</tr>
</tbody>
</table>
<h3 id="常用绘图函数"><a href="#常用绘图函数" class="headerlink" title="常用绘图函数"></a>常用绘图函数</h3><table>
<thead>
<tr>
<th>函数名</th>
<th>功能</th>
</tr>
</thead>
<tbody>
<tr>
<td>plot(x)</td>
<td>以x的元素值为纵坐标、以序号为横坐标绘图</td>
</tr>
<tr>
<td>plot(x,y)</td>
<td>x与y的二元作图</td>
</tr>
<tr>
<td>pie(x)</td>
<td>饼图</td>
</tr>
<tr>
<td>boxplot(x)</td>
<td>盒形图(也称箱线图)</td>
</tr>
<tr>
<td>hist(x)</td>
<td>x的频率直方图</td>
</tr>
<tr>
<td>barplot(x)</td>
<td>x的值的条形图</td>
</tr>
<tr>
<td>pairs(x)</td>
<td>如果x是矩阵或是数据框，作x的各列之间的二元图</td>
</tr>
<tr>
<td>matplot(x,y)</td>
<td>二元图，其中x的第一列对应y的第一列，依次类推</td>
</tr>
<tr>
<td>qqnorm(x)</td>
<td>正态分位数-分位数图</td>
</tr>
<tr>
<td>image(x,y,z)</td>
<td>x,y,z三元图</td>
</tr>
<tr>
<td>heatmap(x)</td>
<td>热图</td>
</tr>
<tr>
<td>coplot(x~y&#124;z)</td>
<td>关于z的每个数值(或数值区间)绘制x与y的二元图</td>
</tr>
</tbody>
</table>
<h3 id="常见图，例子"><a href="#常见图，例子" class="headerlink" title="常见图，例子"></a>常见图，例子</h3><p><a href="http://docs.ggplot2.org/current/" target="_blank" rel="external">ggplot2 doc</a></p>
<p><a href="http://blog.csdn.net/bone_ace/article/details/47284065" target="_blank" rel="external">柱状图、饼图、玫瑰图、图例、坐标系、主题、颜色等汇总</a></p>
<p><a href="http://fibears.top/2016/01/24/RVisualization/" target="_blank" rel="external">R/ggplot2基础查询</a></p>
<p><a href="http://www.cnblogs.com/weibaar/p/4567791.html" target="_blank" rel="external">柱状图、分类柱状图、簇形图、百分比堆积图、饼图极坐标图、折线图</a></p>
<p><a href="http://fibears.top/2016/01/06/RMap/" target="_blank" rel="external">地图</a></p>
<p><a href="http://www.longqi.cf/Archive/#ggplot" target="_blank" rel="external">ggplot概念理解</a></p>
<p><a href="http://www.cnblogs.com/yumtaoist/p/4498729.html" target="_blank" rel="external">用ggplot2画qqplot</a></p>
<p><a href="http://www.cnblogs.com/yumtaoist/p/4240199.html" target="_blank" rel="external">ggplot2叠加多图层</a></p>
<h3 id="安装包方法"><a href="#安装包方法" class="headerlink" title="安装包方法"></a>安装包方法</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">source(&quot;http://www.bioconductor.org/biocLite.R&quot;)</div><div class="line"># 可以一次装多个包</div><div class="line">biocLite(c(&quot;gridExtra&quot;,&quot;ggplot2&quot;))</div></pre></td></tr></table></figure>
<h3 id="几何性质相关"><a href="#几何性质相关" class="headerlink" title="几何性质相关"></a>几何性质相关</h3><p>alpha, color, fill, linetype, shape, size</p>
<h4 id="shape"><a href="#shape" class="headerlink" title="shape"></a>shape</h4><p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170103-0@2x-shape.sample.png?imageView/2/w/500" alt="image"></p>
<h4 id="linetype："><a href="#linetype：" class="headerlink" title="linetype："></a>linetype：</h4><p>0 = blank, 1 = solid, 2 = dashed, 3 = dotted, 4 = dotdash, 5 = longdash, 6 = twodas</p>
<h3 id="数据操作"><a href="#数据操作" class="headerlink" title="数据操作"></a>数据操作</h3><p>矩阵操作<br>合并<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div></pre></td><td class="code"><pre><div class="line"># 合并两个矩阵</div><div class="line"># 纵向合并</div><div class="line">cbind(pop,admix)</div><div class="line"># 横向合并</div><div class="line">rbind(a,b)</div><div class="line"></div><div class="line">#横向合并</div><div class="line">ID&lt;-c(1,2,3,4)</div><div class="line">name&lt;-c(&quot;Jim&quot;,&quot;Tony&quot;,&quot;Lisa&quot;,&quot;Tom&quot;)</div><div class="line">score&lt;-c(89,22,78,78)</div><div class="line">student1&lt;-data.frame(ID,name)</div><div class="line">student2&lt;-data.frame(ID,score)</div><div class="line">total_student&lt;-merge(student1,student2,by=&quot;ID&quot;)</div><div class="line">total_student</div><div class="line">#纵向合并</div><div class="line">ID&lt;-c(1,2,3)</div><div class="line">name&lt;-c(&quot;Jame&quot;,&quot;Kevin&quot;,&quot;Sunny&quot;)</div><div class="line">student1&lt;-data.frame(ID,name)</div><div class="line">ID&lt;-c(4,5,6)</div><div class="line">name&lt;-c(&quot;Sun&quot;,&quot;Frame&quot;,&quot;Eric&quot;)</div><div class="line">student2&lt;-data.frame(ID,name)</div><div class="line">total&lt;-rbind(student1,student2)</div><div class="line">total</div></pre></td></tr></table></figure></p>
<p>取出部分行、列<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"># 取出1，2，3行、列</div><div class="line">admixname=admixsort[c(1,2,3),] # 行</div><div class="line">admixpatial=admixsort[,c(1,2,3)] # 列</div><div class="line"></div><div class="line"># 去掉1，2，3行、列</div><div class="line">admixpatial=admixsort[-c(1,2,3),] # 行</div><div class="line">admixpatial=admixsort[,-c(1,2,3)] # 列</div></pre></td></tr></table></figure></p>
<p>截取、排序、构造新列</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div></pre></td><td class="code"><pre><div class="line"># 截取大于x的所有数值</div><div class="line">a=rnorm(10)</div><div class="line">a=a[a&gt;0]</div><div class="line"></div><div class="line"># 排序</div><div class="line">对于矩阵pop</div><div class="line">针对第3列排序：</div><div class="line">pop2=pop[order(pop[,3]),]  # 从小到大</div><div class="line">pop2=pop[order(-pop[,3]),] # 从大到小</div><div class="line"></div><div class="line">先按第3列，再按第4列排序，再按第5列排序：</div><div class="line">pop2=pop[order(pop[,3],pop[,4],pop[,5]),]</div><div class="line"></div><div class="line">如果想以按照第3列从小到大，如果第3列一样，按照第4列从大到小排列：</div><div class="line">pop2=pop[order(pop[,3],-pop[,4]),]</div><div class="line"></div><div class="line"># R重新构造一列</div><div class="line">a$V3=paste(a$V1,a$V2,sep=&quot;-&quot;)</div></pre></td></tr></table></figure>
<h3 id="stat-pos-binwidth-scale"><a href="#stat-pos-binwidth-scale" class="headerlink" title="stat, pos, binwidth, scale"></a>stat, pos, binwidth, scale</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div></pre></td><td class="code"><pre><div class="line">stat_identity</div><div class="line">stat_bin</div><div class="line">stat=&apos;bin&apos; # 默认</div><div class="line">stat=&apos;identity&apos; #</div><div class="line"></div><div class="line">pos=&apos;stack&apos; # 不同类/组的X叠加起来，默认</div><div class="line">pos=&apos;fill&apos; # 将每个x值的数据归一化到1</div><div class="line">pos=&apos;dodge&apos; # 独立开来</div><div class="line"></div><div class="line">binwidth=30</div><div class="line"></div><div class="line"># scale</div><div class="line">scale_x_log10(...)</div><div class="line">scale_y_log10(...)</div><div class="line"></div><div class="line">scale_x_reverse(...)</div><div class="line">scale_y_reverse(...)</div><div class="line"></div><div class="line">scale_x_sqrt(...)</div><div class="line">scale_y_sqrt(...)</div><div class="line"></div><div class="line">scale_x_datetime()</div><div class="line">scale_x_date()</div><div class="line"></div><div class="line">scale_x_discrete()</div></pre></td></tr></table></figure>
<h3 id="标签"><a href="#标签" class="headerlink" title="标签"></a>标签</h3><p>labs, guides<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">+ labs(title=&quot;Title&quot;, x=&apos;Weight&apos;, y=&apos;Y:mpg&apos;)</div><div class="line">+ guides(color=guide_legend(title=&quot;Yan Se&quot;, title.pos = &apos;bottom&apos;),size=guide_legend(title=&apos;Size&apos;,reverse=TRUE))</div><div class="line"></div><div class="line"># guides也可以在相应的scale, theme函数里</div><div class="line">+ scale_color_brewer(palette=&apos;Set1&apos;, guide=guide_legend(title=&quot;Yan Se&quot;, title.pos = &apos;bottom&apos;))</div><div class="line"></div><div class="line"># guides 分几列显示</div><div class="line">+guides(color=guide_legend(ncol=2))</div><div class="line">+guides(shape=guide_legend(ncol=2))</div><div class="line">+guides(fill=guide_legend(ncol=2))</div></pre></td></tr></table></figure></p>
<p>单独画坐标、竖线（admixture图里面）<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line">position=c(0,tapply(1:nrow(admixname),admixname[,3],max))</div><div class="line"># 坐标</div><div class="line">axis(1,at=position,labels=F)</div><div class="line"># 一组竖线</div><div class="line">abline(v=position,lwd=0.2)</div></pre></td></tr></table></figure></p>
<h3 id="将绘图区划分成-m-行-n-列"><a href="#将绘图区划分成-m-行-n-列" class="headerlink" title="将绘图区划分成 m 行 n 列"></a>将绘图区划分成 m 行 n 列</h3><h4 id="用facet把图分成小图，多图"><a href="#用facet把图分成小图，多图" class="headerlink" title="用facet把图分成小图，多图"></a>用facet把图分成小图，多图</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div></pre></td><td class="code"><pre><div class="line">+ facet_wrap(~cut)</div><div class="line">+ facet_wrap(~cut, ncol=1)</div><div class="line">+ facet_wrap(~cut+clarity, ncol=4)</div><div class="line">+ facet_wrap(~clarity+cut, ncol=4, scales=&quot;free&quot;)</div><div class="line"></div><div class="line"># facet_grid与facet_wrap可以实现相同效果，</div><div class="line">+ facet_wrap(~gear, nrow=1)</div><div class="line">+ facet_grid(.~gear)</div><div class="line"></div><div class="line">+ facet_wrap(~gear, ncol=1)</div><div class="line">+ facet_grid(gear~.)</div><div class="line"></div><div class="line">+ facet_grid(clarity~cut)</div></pre></td></tr></table></figure>
<p>facet_wrap()说明<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div></pre></td><td class="code"><pre><div class="line">facet_wrap(facets, nrow = NULL, ncol = NULL, scales = &quot;fixed&quot;,</div><div class="line">  shrink = TRUE, as.table = TRUE, drop = TRUE)</div><div class="line"></div><div class="line">**Arguments**</div><div class="line"></div><div class="line">facets  </div><div class="line">一般是“~变量名”或者“~变量一+变量二”</div><div class="line"></div><div class="line">nrow    </div><div class="line">控制总行数</div><div class="line"></div><div class="line">ncol    </div><div class="line">控制总列数</div><div class="line"></div><div class="line">scales  </div><div class="line">这个变量控制不同图像间坐标的尺度关系。&quot;fixed&quot;是指多个坐标轴使用相同的尺度，这个是默认值； &quot;free&quot;是指根据当前坐标轴的内容确定尺度，&quot;free_x&quot;和&quot;free_y&quot;是根据x或者y的数据范围确定当前坐标轴的尺度。</div><div class="line"></div><div class="line">as.table    </div><div class="line">这个控制最大值的变量的位置，TURE是默认值，意思是最大值在右下角；可以设置成FALSE，那么最大值将在右上角。</div><div class="line"></div><div class="line">drop    </div><div class="line">这个如果是真会自动删除没有数据的坐标轴。默认是真。</div></pre></td></tr></table></figure></p>
<h4 id="mfrow方法"><a href="#mfrow方法" class="headerlink" title="mfrow方法"></a>mfrow方法</h4><p>par(mfrow=c(m,n))<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line"># 划分成 1行 2列</div><div class="line">par(mfrow=c(1,2))</div><div class="line"></div><div class="line"># 直接在原图上叠加</div><div class="line">par(new=TRUE)</div></pre></td></tr></table></figure></p>
<h4 id="gridExtra方法，可以和ggplot一起使用"><a href="#gridExtra方法，可以和ggplot一起使用" class="headerlink" title="gridExtra方法，可以和ggplot一起使用"></a>gridExtra方法，可以和ggplot一起使用</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line">library(grid)</div><div class="line">library(gridExtra)</div><div class="line">b=ggplot(a,aes(pos,value))+geom_point(col=1)</div><div class="line">c=ggplot(a,aes(pos,value))+geom_point(col=2)</div><div class="line">grid.arrange(b,c,ncol=2,nrow=2) # 4个小图，2x2排布</div></pre></td></tr></table></figure>
<hr>
<h1 id="常用图例子"><a href="#常用图例子" class="headerlink" title="常用图例子"></a>常用图例子</h1><h2 id="点图-point"><a href="#点图-point" class="headerlink" title="点图 point"></a>点图 point</h2><p>应用（PCA）</p>
<p>input file：“input.ggplot2”:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">name     PC1      PC2      species</div><div class="line">sample1    1.10     2.039    D</div><div class="line">sample2    0.83     1.928    D</div><div class="line">sample3    0.678    0.837    W</div><div class="line">sample4    1.828    1.282    W</div><div class="line">...</div></pre></td></tr></table></figure></p>
<h3 id="geom-point"><a href="#geom-point" class="headerlink" title="geom_point()"></a>geom_point()</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">a=read.table(<span class="string">"input.ggplot2"</span>,header=<span class="literal">T</span>);</div><div class="line">mycolor=c(<span class="string">"#BF812D"</span>,<span class="string">"#35978F"</span>)</div><div class="line"></div><div class="line">ggplot(data,aes(V2,V3,color=factor(V6)))+geom_point(size=<span class="number">8</span>,alpha=<span class="number">0.8</span>)+xlab(<span class="string">"PC1(25.0%)"</span>)+ylab(<span class="string">"PC2(12.0%)"</span>)+theme_bw()+theme(legend.title=element_blank(),panel.grid =element_blank(),  panel.grid.major=element_blank())+scale_color_manual(values=mycolor)</div><div class="line"></div><div class="line"><span class="comment"># OR</span></div><div class="line">p=ggplot(a,aes(PC1,PC2,color=species))+geom_point(alpha=<span class="number">0.9</span>,size=<span class="number">4</span>)</div><div class="line">p+theme_bw()+scale_color_manual(values=c(\<span class="string">"#2166ac\",\"#b2182b\"))</span></div></pre></td></tr></table></figure>
<h3 id="特殊例子，同时设定颜色color和形状，同时只显示一个legend"><a href="#特殊例子，同时设定颜色color和形状，同时只显示一个legend" class="headerlink" title="特殊例子，同时设定颜色color和形状，同时只显示一个legend"></a>特殊例子，同时设定颜色color和形状，同时只显示一个legend</h3><p>在<code>scale_colour_manual</code>和<code>scale_shape_manual</code>中使用同样的<code>name</code>和<code>labels</code><br><a href="https://stackoverflow.com/questions/12410908/combine-legends-for-color-and-shape-into-a-single-legend" target="_blank" rel="external">link</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div></pre></td><td class="code"><pre><div class="line"># 数据</div><div class="line">state1 &lt;- c(rep(c(rep(&quot;N&quot;, 7), rep(&quot;Y&quot;, 7)), 2))</div><div class="line">year &lt;- rep(c(2003:2009), 4)</div><div class="line">group1 &lt;- c(rep(&quot;C&quot;, 14), rep(&quot;E&quot;, 14))</div><div class="line">group2 &lt;- paste(state1, group1, sep = &quot;&quot;)</div><div class="line">beta &lt;- c(0.16,0.15,0.08,0.08,0.18,0.48,0.14,0.19,0.00,0.00,0.04,0.08,0.27,0.03,0.11,0.12,0.09,0.09,0.10,0.19,0.16,0.00,0.11,0.07,0.08,0.09,0.19,0.10)</div><div class="line">    lcl &lt;- c(0.13,0.12,0.05,0.05,0.12,0.35,0.06,0.13,0.00,0.00,0.01,0.04,0.20,0.00,0.09,0.09,0.06,0.06,0.07,0.15,0.11,0.00,0.07,0.03,0.05,0.06,0.15,0.06)</div><div class="line">    ucl &lt;- c(0.20,0.20,0.13,0.14,0.27,0.61,0.28,0.27,0.00,1.00,0.16,0.16,0.36,0.82,0.14,0.15,0.13,0.13,0.15,0.23,0.21,0.00,0.15,0.14,0.12,0.12,0.23,0.16)</div><div class="line">data &lt;- data.frame(state1,year,group1,group2,beta,lcl,ucl)</div><div class="line"></div><div class="line"># 画图</div><div class="line">library(ggplot2)</div><div class="line">pd &lt;- position_dodge(.65)</div><div class="line">ggplot(data = data,aes(x= year, y = beta, colour = group2, shape = group2)) +    </div><div class="line">  geom_point(position = pd, size = 4) +</div><div class="line">  geom_errorbar(aes(ymin = lcl, ymax = ucl), colour = &quot;black&quot;, width = 0.5, position = pd) +</div><div class="line">  scale_colour_manual(name = &quot;Treatment &amp; State&quot;,</div><div class="line">                      labels = c(&quot;Control, Non-F&quot;, &quot;Control, Flwr&quot;, &quot;Exclosure, Non-F&quot;, &quot;Exclosure, Flwr&quot;),</div><div class="line">                      values = c(&quot;blue&quot;, &quot;red&quot;, &quot;blue&quot;, &quot;red&quot;)) +   </div><div class="line">  scale_shape_manual(name = &quot;Treatment &amp; State&quot;,</div><div class="line">                     labels = c(&quot;Control, Non-F&quot;, &quot;Control, Flwr&quot;, &quot;Exclosure, Non-F&quot;, &quot;Exclosure, Flwr&quot;),</div><div class="line">                     values = c(19, 19, 17, 17))</div></pre></td></tr></table></figure></p>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170927-191813.png?imageView/2/w/500" alt="image"></p>
<p>手动指定颜色、形状，例子：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">ggplot(a,aes(PC1,PC2,shape=Regions,color=Regions))+geom_point(size=2)+scale_color_manual(name=&quot;test&quot;,labels=c(&quot;AWD&quot;,&quot;EMZ&quot;,&quot;NQA&quot;,&quot;RDA&quot;,&quot;MODA&quot;,&quot;BCS&quot;,&quot;GCN&quot;,&quot;BMN&quot;,&quot;BSI&quot;,&quot;AME&quot;,&quot;GME&quot;,&quot;DOP&quot;,&quot;TEX&quot;,&quot;BGE&quot;,&quot;GUR&quot;,&quot;GAR&quot;,&quot;SUM&quot;,&quot;AFS&quot;,&quot;AW&quot;,&quot;CC&quot;,&quot;IROA&quot;,&quot;KRS&quot;,&quot;KR&quot;,&quot;NDZ&quot;,&quot;SKZ&quot;,&quot;AWT&quot;,&quot;ALT&quot;,&quot;BAS&quot;,&quot;BAY&quot;,&quot;DUL&quot;,&quot;HET&quot;,&quot;KAZ&quot;,&quot;QIB&quot;,&quot;TUB&quot;,&quot;GLT&quot;,&quot;HUL&quot;,&quot;HUS&quot;,&quot;JZS&quot;,&quot;LLT&quot;,&quot;LOP&quot;,&quot;LTH&quot;,&quot;LZM&quot;,&quot;SNT&quot;,&quot;SSF&quot;,&quot;STH&quot;,&quot;TAN&quot;,&quot;THF&quot;,&quot;TON&quot;,&quot;UJI&quot;,&quot;WDS&quot;,&quot;YFT&quot;,&quot;CHA&quot;,&quot;DQS&quot;,&quot;GBF&quot;,&quot;HZS&quot;,&quot;MBF&quot;,&quot;OLA&quot;,&quot;TCS&quot;,&quot;TIB&quot;,&quot;TIBQ&quot;,&quot;TIBS&quot;,&quot;WNS&quot;,&quot;LPB&quot;,&quot;NLB&quot;,&quot;SPG&quot;,&quot;ZTS&quot;,&quot;IROO&quot;,&quot;IROV&quot;,&quot;MPS&quot;,&quot;OCAN&quot;,&quot;ODAL&quot;),values=c(&quot;#E41A1C&quot;,&quot;#E41A1C&quot;,&quot;#E41A1C&quot;,&quot;#E41A1C&quot;,&quot;#E41A1C&quot;,&quot;#377EB8&quot;,&quot;#377EB8&quot;,&quot;#377EB8&quot;,&quot;#377EB8&quot;,&quot;#984EA3&quot;,&quot;#984EA3&quot;,&quot;#984EA3&quot;,&quot;#984EA3&quot;,&quot;#FF7F00&quot;,&quot;#FF7F00&quot;,&quot;#FF7F00&quot;,&quot;#FF7F00&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#A65628&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#b2df8a&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#4DAF4A&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#e7298a&quot;,&quot;#4eb3d3&quot;,&quot;#4eb3d3&quot;,&quot;#4eb3d3&quot;,&quot;#4eb3d3&quot;,&quot;#F781BF&quot;,&quot;#000000&quot;,&quot;#000000&quot;,&quot;#000000&quot;,&quot;#000000&quot;))+scale_shape_manual(name=&quot;test&quot;,labels=c(&quot;AWD&quot;,&quot;EMZ&quot;,&quot;NQA&quot;,&quot;RDA&quot;,&quot;MODA&quot;,&quot;BCS&quot;,&quot;GCN&quot;,&quot;BMN&quot;,&quot;BSI&quot;,&quot;AME&quot;,&quot;GME&quot;,&quot;DOP&quot;,&quot;TEX&quot;,&quot;BGE&quot;,&quot;GUR&quot;,&quot;GAR&quot;,&quot;SUM&quot;,&quot;AFS&quot;,&quot;AW&quot;,&quot;CC&quot;,&quot;IROA&quot;,&quot;KRS&quot;,&quot;KR&quot;,&quot;NDZ&quot;,&quot;SKZ&quot;,&quot;AWT&quot;,&quot;ALT&quot;,&quot;BAS&quot;,&quot;BAY&quot;,&quot;DUL&quot;,&quot;HET&quot;,&quot;KAZ&quot;,&quot;QIB&quot;,&quot;TUB&quot;,&quot;GLT&quot;,&quot;HUL&quot;,&quot;HUS&quot;,&quot;JZS&quot;,&quot;LLT&quot;,&quot;LOP&quot;,&quot;LTH&quot;,&quot;LZM&quot;,&quot;SNT&quot;,&quot;SSF&quot;,&quot;STH&quot;,&quot;TAN&quot;,&quot;THF&quot;,&quot;TON&quot;,&quot;UJI&quot;,&quot;WDS&quot;,&quot;YFT&quot;,&quot;CHA&quot;,&quot;DQS&quot;,&quot;GBF&quot;,&quot;HZS&quot;,&quot;MBF&quot;,&quot;OLA&quot;,&quot;TCS&quot;,&quot;TIB&quot;,&quot;TIBQ&quot;,&quot;TIBS&quot;,&quot;WNS&quot;,&quot;LPB&quot;,&quot;NLB&quot;,&quot;SPG&quot;,&quot;ZTS&quot;,&quot;IROO&quot;,&quot;IROV&quot;,&quot;MPS&quot;,&quot;OCAN&quot;,&quot;ODAL&quot;),values=c(1,2,3,4,5,1,2,3,4,1,2,3,4,1,2,3,4,1,2,3,4,5,6,7,8,9,1,2,3,4,5,6,7,8,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,1,2,3,4,5,6,7,8,9,10,11,1,2,3,4,1,2,3,4,5))+theme_bw()+theme(plot.background=element_blank(),panel.background=element_blank(),legend.background=element_blank(),legend.position=&apos;right&apos;,panel.grid =element_blank(),legend.title = element_blank(),axis.title =element_text(size=9,color=&quot;black&quot;),axis.text=element_text(size=8,color=&quot;black&quot;))+xlab(&quot;Principal component 1&quot;)+ylab(&quot;Principal component 2&quot;)+guides(shape=guide_legend(ncol=5),color=guide_legend(ncol=5))</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170928-182556.png?imageView/2/w/500" alt="image"></p>
<h3 id="plot-amp-points"><a href="#plot-amp-points" class="headerlink" title="plot &amp; points"></a>plot &amp; points</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div></pre></td><td class="code"><pre><div class="line"># 用不同的颜色区分不同群体</div><div class="line">d1=read.table(&quot;D1.txt&quot;,header=T);</div><div class="line">d2=read.table(&quot;D2.txt&quot;,header=T);</div><div class="line">w=read.table(&quot;W.txt&quot;,header=T);</div><div class="line"></div><div class="line">d1x=d1$PC1</div><div class="line">d1y=d1$PC2</div><div class="line">d2x=d2$PC1</div><div class="line">d2y=d2$PC2</div><div class="line">wx=w$PC1</div><div class="line">wy=w$PC2</div><div class="line"></div><div class="line">pdf(file=&quot;PC12.pdf&quot;,width=2,height=2)</div><div class="line">plot(d1x,d1y,pch=4,col=&quot;#89c876&quot;,cex=2,cex.lab=1.5,lwd=2,xlab=&quot;PC1 (3.24%)&quot;,ylab=&quot;PC2 (1.36%)&quot;,xlim=c(-0.3,0.1),ylim=c(-0.6,0.2))</div><div class="line">points(d2x,d2y,pch=4,col=&quot;#2166ac&quot;,cex=2,lwd=2)</div><div class="line">points(wx,wy,pch=4,col=&quot;#b2182b&quot;,cex=2,lwd=2)</div><div class="line">dev.off()</div></pre></td></tr></table></figure>
<h2 id="barplot-柱状图"><a href="#barplot-柱状图" class="headerlink" title="barplot 柱状图"></a>barplot 柱状图</h2><h3 id="barplot"><a href="#barplot" class="headerlink" title="barplot"></a>barplot</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line"># admixture</div><div class="line"># k2.txt</div><div class="line">0.000000001	0.999999999</div><div class="line">0.029249134	0.970750866</div><div class="line">0.000000001	0.999999999</div><div class="line">8.65071E-08	0.999999913</div><div class="line">0.000000001	0.999999999</div><div class="line">1.49306E-08	0.999999985</div><div class="line">0.043220336	0.956779664</div><div class="line">0.04214767	0.95785233</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">admix=t(as.matrix(read.table(&quot;k2.txt&quot;,header=F)))</div><div class="line">barplot(admix,col=c(&quot;#2166ac&quot;,&quot;#b2182b&quot;),space=0,border=NULL ,xlab=&quot;Individuals&quot;,ylab=&quot;K=2&quot;,cex.lab=2,yaxt=&apos;n&apos;)</div></pre></td></tr></table></figure>
<h3 id="barplot-for-admixture-results-example"><a href="#barplot-for-admixture-results-example" class="headerlink" title="barplot for admixture results example:"></a>barplot for admixture results example:</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line"># 02.poplist2</div><div class="line">...</div><div class="line"># small_pop_id sample_id small_pop_order big_pop_id  big_pop_order</div><div class="line">TIB     SRR501845       60      Dom_EA_TIB      9</div><div class="line">TIB     SRR501863       60      Dom_EA_TIB      9</div><div class="line">BGE     SRR501854       27      Dom_SA  6</div><div class="line">BGE     SRR501873       27      Dom_SA  6</div><div class="line">GAR     SRR501902       26      Dom_SA  6</div><div class="line">...</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div></pre></td><td class="code"><pre><div class="line">par(mfrow=c(10,1), mar=c(0,3,1.5,0.5), las = 0, mgp=c(0.5, 0.5, 0))</div><div class="line">color=c(&quot;#E41A1C&quot;,&quot;#377EB8&quot;,&quot;#4DAF4A&quot;,&quot;#984EA3&quot;,&quot;#FF7F00&quot;,&quot;#A65628&quot;,&quot;#F781BF&quot;)</div><div class="line"></div><div class="line">pop&lt;-read.table(&quot;02.poplist2&quot;,as.is=T)</div><div class="line">admix&lt;-read.table(&quot;input.2.Q&quot;,as.is=T)</div><div class="line">admixmerge=t(as.matrix(cbind(pop,admix)))</div><div class="line">admixsort=admixmerge[,order(admixmerge[3,],admixmerge[6,],admixmerge[7,])]</div><div class="line">admixpatial=admixsort[-c(1,2,3,4,5),]</div><div class="line">admixname=t(admixsort[c(1,2,3,4,5),])</div><div class="line">barplot(admixpatial,col=color[1:2],space=0,border=NA,xlab=&quot;&quot;,ylab=&quot;K=2&quot;,yaxt=&apos;n&apos;)</div><div class="line"></div><div class="line"># small population axis</div><div class="line">#position=c(0,tapply(1:nrow(admixname),admixname[,3],max))</div><div class="line">#axis(1,at=position,labels=F)</div><div class="line"></div><div class="line"># small population line</div><div class="line">position=c(0,tapply(1:nrow(admixname),admixname[,3],max))</div><div class="line">abline(v=position,lwd=0.2)</div><div class="line"></div><div class="line"># big population line</div><div class="line">position2=c(0,tapply(1:nrow(admixname),admixname[,5],max))</div><div class="line">abline(v=position2,lwd=1)</div><div class="line"></div><div class="line"># name</div><div class="line">barplot(admixpatial,col=c(&quot;white&quot;),space=0,border=NA,xlab=&quot;&quot;,ylab=&quot; &quot;,xaxt=&apos;n&apos;, yaxt=&apos;n&apos;)</div><div class="line">text(tapply(1:nrow(admixname),admixname[,3],mean),0.8,unique(admixname[,1]),xpd=T,srt=90,cex=0.45)</div><div class="line">barplot(admixpatial,col=c(&quot;white&quot;),space=0,border=NA,xlab=&quot;&quot;,ylab=&quot; &quot;,xaxt=&apos;n&apos;, yaxt=&apos;n&apos;)</div><div class="line">text(tapply(1:nrow(admixname),admixname[,5],mean),0.8,unique(admixname[,4]),xpd=T,srt=0,cex=0.45)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170901-153652.png?imageView/2/w/500" alt="image"></p>
<h3 id="geom-bar"><a href="#geom-bar" class="headerlink" title="geom_bar"></a>geom_bar</h3><p>给定数据的柱状图<br><a href="http://blog.csdn.net/bone_ace/article/details/47267981" target="_blank" rel="external">http://blog.csdn.net/bone_ace/article/details/47267981</a><br><a href="http://shujuren.org/article/44.html" target="_blank" rel="external">http://shujuren.org/article/44.html</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># x轴为文本标签时，加上 stat=&apos;identity&apos;</div><div class="line">ggplot(a,mapping=aes(x=V1,y=V2,fill=V3))+geom_bar(stat=&apos;identity&apos;)+scale_x_discrete(limits=c(&quot;1/302&quot;,&quot;2/302&quot;,&quot;3/302&quot;,&quot;4/302&quot;,&quot;5/302&quot;,&quot;6/302&quot;,&quot;7/302&quot;,&quot;8/302&quot;,&quot;9/302&quot;,&quot;10/302&quot;,&quot;11/302&quot;,&quot;12/302&quot;,&quot;13/302&quot;,&quot;14/302&quot;,&quot;15/302&quot;,&quot;16/302&quot;,&quot;17/302&quot;,&quot;18/302&quot;,&quot;19/302&quot;,&quot;20/302&quot;,&quot;21/302&quot;,&quot;22/302&quot;,&quot;23/302&quot;,&quot;24/302&quot;,&quot;25/302&quot;,&quot;26/302&quot;,&quot;27/302&quot;,&quot;28/302&quot;,&quot;29/302&quot;,&quot;30/302&quot;,&quot;0.1-0.2&quot;,&quot;0.2-1&quot;))+theme_bw()+theme(panel.border=element_blank(),panel.grid =element_blank(),legend.title=element_blank(),axis.text.x=element_text(angle =90))+scale_fill_manual(values=c(&quot;#1F78B4&quot;,&quot;#FF7F00&quot;))+labs(x=&quot;Allele frequency&quot;,y=&quot;SNP number (M)&quot;)</div></pre></td></tr></table></figure></p>
<h2 id="geom-histogram"><a href="#geom-histogram" class="headerlink" title="geom_histogram()"></a>geom_histogram()</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">pdf(file=&quot;B1.final.cnv.V3.pdf&quot;)</div><div class="line">library(&quot;ggplot2&quot;)</div><div class="line">a &lt;- read.table(&quot;B1.final.cnv&quot;,head=F)</div><div class="line"># logx</div><div class="line">ggplot(a,aes(x=log(a$V3)/log(10)))+geom_histogram()+xlab(&quot;CNV length (log10)&quot;)+ylab(&quot;Count&quot;)+xlim(3,7)</div><div class="line"># ggplot(a,aes(x=a$V3))+geom_histogram()+xlab(&quot;CNV length (log10)&quot;)+ylab(&quot;Count&quot;)</div><div class="line">dev.off()</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line"># histogram纵坐标变成density，默认是count</div><div class="line">ggplot(qcctau,aes(x=V1,y=..density..,fill=factor(V2)))+ geom_histogram(position=&apos;dodge&apos;)</div><div class="line"></div><div class="line"># position （比如直方图）</div><div class="line">dodge 并列</div><div class="line">fill 标准化堆叠</div><div class="line">identity 重叠</div><div class="line">jitter	使用jitter方式保证不重叠</div><div class="line">stack	堆叠</div></pre></td></tr></table></figure>
<h2 id="小提琴图-geom-violin"><a href="#小提琴图-geom-violin" class="headerlink" title="小提琴图 geom_violin()"></a>小提琴图 geom_violin()</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">geom_violin()</div><div class="line"># 参数：adjust调整平滑程度，scale=&quot;count&quot;使得图的面积与每组观测值数目成正比</div></pre></td></tr></table></figure>
<h2 id="boxplot-箱线图"><a href="#boxplot-箱线图" class="headerlink" title="boxplot 箱线图"></a>boxplot 箱线图</h2><h3 id="boxplot"><a href="#boxplot" class="headerlink" title="boxplot"></a>boxplot</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">par(mfrow=c(1,2))</div><div class="line">boxplot(bb1$V2,bc1$V2,cc1$V2,outline = F,names=c(&quot;B-B&quot;,&quot;B-C&quot;,&quot;C-C&quot;),main=&quot;1 bp overlap&quot;)</div><div class="line">boxplot(bb5$V2,bc5$V2,cc5$V2,outline = F,names=c(&quot;B-B&quot;,&quot;B-C&quot;,&quot;C-C&quot;),main=&quot;50% reciprocal overlap&quot;)</div></pre></td></tr></table></figure>
<h3 id="geom-boxplot"><a href="#geom-boxplot" class="headerlink" title="geom_boxplot"></a>geom_boxplot</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line"># 根据 V3 大小值排序（从小到大）</div><div class="line">ggplot(a,aes(x=reorder(V1,V3),y=V3)+geom_boxplot()</div><div class="line"></div><div class="line"># 手动调整顺序</div><div class="line">ggplot(b,aes(V1,V4/V5))+geom_boxplot()+xlab(&quot;Individuals&quot;)+ylab(&quot;HET/HOM&quot;)+theme( legend.position = &quot;none&quot;)+scale_x_discrete(limits=c(&quot;BRM&quot;,&quot;GIR&quot;,&quot;LQC&quot;,&quot;NEL&quot;,&quot;LXC&quot;,&quot;NYC&quot;,&quot;QCC&quot;,&quot;YBC&quot;,&quot;YNC&quot;,&quot;FLV&quot;,&quot;HOL&quot;,&quot;JBC&quot;,&quot;JER&quot;,&quot;LIM&quot;,&quot;RAN&quot;))</div><div class="line"></div><div class="line"># x、y轴转换，横着的boxplot</div><div class="line">+coord_flip()</div><div class="line"></div><div class="line"># 参数 width,outlier.size,outlier.shape</div></pre></td></tr></table></figure>
<h3 id="点-erro-bar"><a href="#点-erro-bar" class="headerlink" title="点+erro bar"></a>点+erro bar</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"># 如ABBA-BABA test结果</div><div class="line"># range: Mean-3*SE ~ Mean+3*SE</div><div class="line">+geom_pointrange(aes(ymin=V4-3*V5,ymax=V4+3*V5))</div></pre></td></tr></table></figure>
<h2 id="pheatmap-热图"><a href="#pheatmap-热图" class="headerlink" title="pheatmap 热图"></a>pheatmap 热图</h2><p>输入格式 <code>input.txt</code><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div></pre></td><td class="code"><pre><div class="line">        B1	B2	B3	B4	B5	B6	C1	C2	C3	C4	C5</div><div class="line">B1	1	0.624833110814419	0.616958399636281	0.648058868307536	0.63029315960912	0.628120893561104	0.615183929037308	0.632938076416337	0.637414484548242	0.6375	0.616299232103047</div><div class="line">B2	0.624833110814419	1	0.593837535014006	0.65446941975954	0.676940319417204	0.659620596205962	0.653577192038731	0.665760869565217	0.612027158098933	0.626946847960445	0.615502294747578</div><div class="line">B3	0.616958399636281	0.593837535014006	1	0.574553571428571	0.538953350698556	0.558471454880295	0.563129002744739	0.557452699584679	0.606948514022604	0.575867205788466	0.572989510489511</div><div class="line">B4	0.648058868307536	0.65446941975954	0.574553571428571	1	0.645949535192563	0.629439011837365	0.625958099131323	0.636738906088751	0.62962962962963	0.649846734260788	0.627489072365226</div><div class="line">B5	0.63029315960912	0.676940319417204	0.538953350698556	0.645949535192563	1	0.675117112152108	0.667213563029806	0.676982591876209	0.611863155303963	0.635542168674699	0.632996632996633</div><div class="line">B6	0.628120893561104	0.659620596205962	0.558471454880295	0.629439011837365	0.675117112152108	1	0.68078348332451	0.652941176470588	0.60755258126195	0.627040194884287	0.650426921145153</div><div class="line">C1	0.615183929037308	0.653577192038731	0.563129002744739	0.625958099131323	0.667213563029806	0.68078348332451	1	0.64968152866242	0.611585944919278	0.631986450520203	0.650872817955112</div><div class="line">C2	0.632938076416337	0.665760869565217	0.557452699584679	0.636738906088751	0.676982591876209	0.652941176470588	0.64968152866242	1	0.625778629611883	0.652991452991453	0.64551863041289</div><div class="line">C3	0.637414484548242	0.612027158098933	0.606948514022604	0.62962962962963	0.611863155303963	0.60755258126195	0.611585944919278	0.625778629611883	1	0.612910332672395	0.581521739130435</div><div class="line">C4	0.6375	0.626946847960445	0.575867205788466	0.649846734260788	0.635542168674699	0.627040194884287	0.631986450520203	0.652991452991453	0.612910332672395	1	0.682038275305511</div><div class="line">C5	0.616299232103047	0.615502294747578	0.572989510489511	0.627489072365226	0.632996632996633	0.650426921145153	0.650872817955112	0.64551863041289	0.581521739130435	0.682038275305511	1</div></pre></td></tr></table></figure></p>
<p>例子<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">a=read.table(&quot;input.txt”)</div><div class="line">a=as.matrix(a)</div><div class="line"></div><div class="line"># 聚类</div><div class="line">pdf(&quot;out.cluster.pdf&quot;,width=10,height=10,onefile=FALSE)</div><div class="line">pheatmap(a,cellwidth = 18, cellheight = 18, fontsize=12, fontsize_row=9,main=&quot;title&quot;)</div><div class="line">dev.off()</div><div class="line"></div><div class="line"># 不聚类</div><div class="line">pheatmap(a,cellwidth = 18, cellheight = 18, fontsize=12, fontsize_row=9,main=&quot;title&quot;,cluster_rows=0,cluster_cols=0)</div></pre></td></tr></table></figure></p>
<p>其他例子<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">pheatmap(data,fontsize=9, fontsize_row=6) #最简单地直接出图</div><div class="line">pheatmap(data, scale = &quot;row&quot;, clustering_distance_row = &quot;correlation&quot;, fontsize=9, fontsize_row=6) #改变排序算法</div><div class="line">pheatmap(data, color = colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50), fontsize=9, fontsize_row=6) #自定义颜色</div><div class="line">pheatmap(data, cluster_row=FALSE, fontsize=9, fontsize_row=6) #关闭按行排序</div><div class="line">pheatmap(data, legend = FALSE, fontsize=9, fontsize_row=6) #关闭图例</div><div class="line">pheatmap(data, cellwidth = 6, cellheight = 5, fontsize=9, fontsize_row=6) #设定格子的尺寸</div></pre></td></tr></table></figure></p>
<h2 id="ggplot-调整作图细节"><a href="#ggplot-调整作图细节" class="headerlink" title="ggplot 调整作图细节"></a>ggplot 调整作图细节</h2><h3 id="保存文件"><a href="#保存文件" class="headerlink" title="保存文件"></a>保存文件</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">pdf(file=&quot;plot.allpop.chr.pl.pdf&quot;,width=10,height=10)</div><div class="line">ggplot ... # 具体图形</div><div class="line">dev.off()</div></pre></td></tr></table></figure>
<h3 id="自动加label"><a href="#自动加label" class="headerlink" title="自动加label"></a>自动加label</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">library(ggrepel)</div><div class="line"></div><div class="line">ggplot(mtcars,aes(wt, mpg,color=&apos;red&apos;))+geom_point() +geom_text_repel(aes(wt,mpg, label=rownames(mtcars)))</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20161230-0@2x-biaodian.png?imageView/2/w/500" alt="image"></p>
<h3 id="添加注释"><a href="#添加注释" class="headerlink" title="添加注释"></a>添加注释</h3><h4 id="添加文本注释"><a href="#添加文本注释" class="headerlink" title="添加文本注释"></a>添加文本注释</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># 可以加多个</div><div class="line">annotate(&quot;text&quot;, x=3, y=48, label=&quot;Group 1&quot;) +annotate(&quot;text&quot;, x=4.5, y=66, label=&quot;Group 2&quot;)</div></pre></td></tr></table></figure>
<h4 id="添加数学表达式注释"><a href="#添加数学表达式注释" class="headerlink" title="添加数学表达式注释"></a>添加数学表达式注释</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">+ annotate(&quot;text&quot;, x=2, y=0.3, parse=TRUE,label=&quot;frac(1, sqrt(2 * pi)) * e ^ &#123;-x^2 / 2&#125;&quot;)</div><div class="line"># ?plotmath可以见到更多使用数学表达式的例子</div></pre></td></tr></table></figure>
<h4 id="添加线条注释"><a href="#添加线条注释" class="headerlink" title="添加线条注释"></a>添加线条注释</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line"># linetype=&quot;dashed&quot;, size=1</div><div class="line">+ geom_hline(yintercept=60) # 横线</div><div class="line">+ geom_vline(xintercept=14) # 竖线</div><div class="line">+ geom_abline(intercept=37.4, slope=1.75) # 斜线；intercept截距；slope斜率</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">+geom_vline() +geom_hline()</div><div class="line"># linetype线的类型（=2 虚线）</div><div class="line"># 指定X轴坐标的竖线</div><div class="line">+geom_vline(xintercept=1000,color=&quot;black&quot;, linetype=2)</div><div class="line"></div><div class="line"># 制定Y轴坐标的横线</div><div class="line">+ geom_hline(yintercept=10,color=&quot;black&quot;, linetype=2)</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">library(plyr)</div><div class="line">hw_means &lt;- ddply(heightweight, &quot;sex&quot;, summarise, heightIn=mean(heightIn))</div><div class="line">p + geom_hline(aes(yintercept=heightIn, colour=sex), data=hw_means,linetype=&quot;dashed&quot;, size=1)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170103-0@2x.png?imageView/2/w/500" alt="image"></p>
<h4 id="添加分割标记"><a href="#添加分割标记" class="headerlink" title="添加分割标记"></a>添加分割标记</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">p &lt;- ggplot(subset(climate, Source==&quot;Berkeley&quot;), aes(x=Year, y=Anomaly10y)) +geom_line()</div><div class="line">p+ annotate(&quot;segment&quot;, x=1950, xend=1980, y=-.25, yend=-.25)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170103-0@2x-annotate.png?imageView/2/w/500" alt="image"></p>
<h4 id="长方形阴影"><a href="#长方形阴影" class="headerlink" title="长方形阴影"></a>长方形阴影</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">p &lt;- ggplot(subset(climate, Source==&quot;Berkeley&quot;), aes(x=Year, y=Anomaly10y)) +geom_line()</div><div class="line">p + annotate(&quot;rect&quot;, xmin=1950, xmax=1980, ymin=-1, ymax=1, alpha=.1,fill=&quot;blue&quot;)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170103-0@2x-3.png?imageView/2/w/500" alt="image"></p>
<h4 id="添加误差线"><a href="#添加误差线" class="headerlink" title="添加误差线"></a>添加误差线</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"># geom_line geom_point同时用，画出线和点</div><div class="line"># geom_errorbar，设置ymin和ymax</div><div class="line">ce &lt;- subset(cabbage_exp, Cultivar == &quot;c39&quot;)</div><div class="line">ggplot(ce, aes(x=Date, y=Weight)) +</div><div class="line">geom_line(aes(group=1)) +</div><div class="line">geom_point(size=4) +</div><div class="line">geom_errorbar(aes(ymin=Weight-se, ymax=Weight+se), width=.2)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170103-0@2x-4.png?imageView/2/w/500" alt="image"></p>
<h4 id="每个小平面增加注释，多小图"><a href="#每个小平面增加注释，多小图" class="headerlink" title="每个小平面增加注释，多小图"></a>每个小平面增加注释，多小图</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">p &lt;- ggplot(mpg, aes(x=displ, y=hwy)) + geom_point() + facet_grid(. ~ drv)</div><div class="line">#构造注释数据框</div><div class="line">f_labels &lt;- data.frame(drv = c(&quot;4&quot;, &quot;f&quot;, &quot;r&quot;), label = c(&quot;4wd&quot;, &quot;Front&quot;, &quot;Rear&quot;))</div><div class="line">p + geom_text(x=6, y=40, aes(label=label), data=f_labels)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170103-0@2x-annot-123small-plot.png?imageView/2/w/500" alt="image"></p>
<h3 id="加x、y坐标名字；标题"><a href="#加x、y坐标名字；标题" class="headerlink" title="加x、y坐标名字；标题"></a>加x、y坐标名字；标题</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">p+xlab(&quot;X轴名&quot;)+ylab(&quot;Y轴名&quot;)+ ggtitle(&quot;标题&quot;)</div><div class="line">p+labs(x=&quot;Xname&quot;,y=&quot;Yname&quot;,title=&quot;title&quot;)</div><div class="line"></div><div class="line"># 坐标格式： 数值百分号、美元符号、带逗号</div><div class="line">p &lt;- ggplot(df, aes(x, y)) + geom_point()</div><div class="line">p + scale_y_continuous(labels = scales::percent)</div><div class="line">p + scale_y_continuous(labels = scales::dollar)</div><div class="line">p + scale_x_continuous(labels = scales::comma)</div></pre></td></tr></table></figure>
<h3 id="x、y坐标反转"><a href="#x、y坐标反转" class="headerlink" title="x、y坐标反转"></a>x、y坐标反转</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">p+coord_flip()</div></pre></td></tr></table></figure>
<h3 id="自定义顺序"><a href="#自定义顺序" class="headerlink" title="自定义顺序"></a>自定义顺序</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">p+scale_x_discrete(limits=c(&quot;BRM&quot;,&quot;GIR&quot;,&quot;LQC&quot;,&quot;NEL&quot;,&quot;LXC&quot;,&quot;NYC&quot;,&quot;QCC&quot;,&quot;YBC&quot;,&quot;YNC&quot;,&quot;FLV&quot;,&quot;HOL&quot;,&quot;JBC&quot;,&quot;JER&quot;,&quot;LIM&quot;,&quot;RAN&quot;))</div></pre></td></tr></table></figure>
<h3 id="自定义颜色"><a href="#自定义颜色" class="headerlink" title="自定义颜色"></a>自定义颜色</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"># set colors; 26</div><div class="line">mycolor=c(&quot;#A6CEE3&quot;, &quot;#1F78B4&quot;, &quot;#B2DF8A&quot;, &quot;#33A02C&quot;, &quot;#FB9A99&quot;, &quot;#E31A1C&quot;, &quot;#FDBF6F&quot;, &quot;#FF7F00&quot;, &quot;#CAB2D6&quot;, &quot;#6A3D9A&quot;, &quot;#FFFF99&quot;, &quot;#B15928&quot;, &quot;#A6CEE3&quot;, &quot;#1F78B4&quot;, &quot;#B2DF8A&quot;, &quot;#33A02C&quot;, &quot;#FB9A99&quot;, &quot;#E31A1C&quot;, &quot;#FDBF6F&quot;, &quot;#FF7F00&quot;, &quot;#CAB2D6&quot;, &quot;#6A3D9A&quot;, &quot;#FFFF99&quot;, &quot;#B15928&quot;, &quot;#A6CEE3&quot;, &quot;#1F78B4&quot;)</div><div class="line">p+scale_color_manual(values=mycolor) # 注意颜色数量要一致，26种颜色对应26个分类</div></pre></td></tr></table></figure>
<h4 id="颜色组合"><a href="#颜色组合" class="headerlink" title="颜色组合"></a>颜色组合</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># 4种</div><div class="line">mycolor4=c(&quot;#E31A1C&quot;,&quot;#1F78B4&quot;,&quot;#6A3D9A&quot;,&quot;#FF7F00&quot;) # 红蓝紫橙</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170830-180135.png?imageView/2/w/500" alt="image"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># 8种</div><div class="line">mycolor8=c(&quot;#E41A1C&quot;,&quot;#377EB8&quot;,&quot;#4DAF4A&quot;,&quot;#984EA3&quot;,&quot;#FF7F00&quot;,&quot;#A65628&quot;,&quot;#F781BF&quot;,&quot;#000000&quot;) # 红蓝绿紫橙褐粉黑</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170830-180104.png?imageView/2/w/500" alt="image"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"># 12种</div><div class="line">mycolor12=c(&quot;#A6CEE3&quot;,&quot;#1F78B4&quot;,&quot;#B2DF8A&quot;,&quot;#33A02C&quot;,&quot;#FB9A99&quot;,&quot;#E31A1C&quot;,&quot;#FDBF6F&quot;,&quot;#FF7F00&quot;,&quot;#CAB2D6&quot;,&quot;#6A3D9A&quot;,&quot;#FFFF99&quot;,&quot;#B15928&quot;)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170830-180556.png?imageView/2/w/500" alt="image"></p>
<h3 id="好看的颜色ColorBrewer"><a href="#好看的颜色ColorBrewer" class="headerlink" title="好看的颜色ColorBrewer"></a>好看的颜色ColorBrewer</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div></pre></td><td class="code"><pre><div class="line"># ColorBrewer</div><div class="line">p + geom_boxplot(aes(fill=factor(cyl)))+ scale_fill_brewer(palette=&apos;Paired&apos;)</div><div class="line"># 对于一个单连续变量仅突出显示一端的色板：</div><div class="line">Blues BuGn BuPu GnBu Greens Greys Oranges OrRd PuBu PuBuGn PuRd Purples RdPu Reds YlGn YlGnBu YlOrBr YlOrRd</div><div class="line"># 对于一个单连续变量突出显示两端的色板</div><div class="line">BrBG PiYG PRGn PuOr RdBu RdGy RdYlBu RdYlGn Spectral</div><div class="line"># 离散变量的色板</div><div class="line">Accent Dark2 Paired Pastel1 Pastel2 Set1 Set2 Set3</div><div class="line"></div><div class="line">连续变量用Brewer色板需要调用 scale_*_distiller</div><div class="line">scale_*_brewer</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div></pre></td><td class="code"><pre><div class="line">library(RColorBrewer)</div><div class="line"># color names : https://github.com/lh3/gnuplot-colorbrewer</div><div class="line"># 查看所有颜色</div><div class="line">display.brewer.all()</div><div class="line"># 查看颜色说明</div><div class="line">brewer.pal.info</div><div class="line"></div><div class="line"># 测试颜色 test colors</div><div class="line">mycolor =brewer.pal(n=8, &apos;Paired&apos;)</div><div class="line">image(1:8,1,as.matrix(1:8),col=mycolor,xlab=&quot; &quot;,ylab=&quot;&quot;,xaxt=&quot;n&quot;,yaxt=&quot;n&quot;,bty=&quot;n&quot;)</div><div class="line"></div><div class="line"># 取其中几个</div><div class="line">subcolor=mycolor[1]</div><div class="line">subcolor=c(mycolor[4],mycolor[1:3],mycolor[7:5])</div><div class="line"></div><div class="line"># ggplot manual colors</div><div class="line">p+scale_color_manual(values=mycolor)</div><div class="line">p+scale_fill_manual(values=mycolor)</div><div class="line">p+scale_color_manual(values=c(&quot;BRM&quot;=&quot;#E31A1C&quot;,&quot;GIR&quot;=&quot;#E31A1C&quot;,&quot;NEL&quot;=&quot;#E31A1C&quot;))</div><div class="line">p+scale_color_manual(values=c(&quot;#E31A1C&quot;,&quot;#1F78B4&quot;,&quot;#6A3D9A&quot;,&quot;#FF7F00&quot;)）#四个颜色 红蓝紫橙</div><div class="line"></div><div class="line"></div><div class="line"># 直接用</div><div class="line">p+scale_color_brewer(palette=&apos;Paired&apos;)</div></pre></td></tr></table></figure>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20161207-0-colors.png?imageView/2/w/500" alt="image"></p>
<p><img src="https://raw.githubusercontent.com/lh3/gnuplot-colorbrewer/master/examples/diverging.png" alt="image"></p>
<p><img src="https://raw.githubusercontent.com/lh3/gnuplot-colorbrewer/master/examples/qualitative.png" alt="image"></p>
<p><img src="https://raw.githubusercontent.com/lh3/gnuplot-colorbrewer/master/examples/sequential.png" alt="image"></p>
<h3 id="theme主题设置，X轴、Y轴操作"><a href="#theme主题设置，X轴、Y轴操作" class="headerlink" title="theme主题设置，X轴、Y轴操作"></a>theme主题设置，X轴、Y轴操作</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div><div class="line">33</div><div class="line">34</div><div class="line">35</div><div class="line">36</div><div class="line">37</div></pre></td><td class="code"><pre><div class="line"># 去掉多余的白色背景、框线，坐标文字黑色</div><div class="line">+theme_bw()</div><div class="line">+theme(plot.background=element_blank(),panel.background=element_blank(),legend.background=element_blank(),panel.grid =element_blank(),legend.title = element_blank(),axis.title =element_text(color=&quot;black&quot;),axis.text=element_text(color=&quot;black&quot;),legend.position=&apos;right&apos;)</div><div class="line"></div><div class="line"># 白色主题模板; theme_bw()放前面</div><div class="line">p+theme_bw()+theme()</div><div class="line"></div><div class="line">p+ theme(</div><div class="line">  # 去掉网格线，分别去掉主要、次要网格线</div><div class="line">  panel.grid =element_blank(),</div><div class="line">  panel.grid.major=element_blank(),</div><div class="line">  panel.grid.minor=element_blank(),</div><div class="line">  # 去掉外边框</div><div class="line">  panel.border=element_blank(),</div><div class="line">  # 加外边框</div><div class="line">  panel.border = element_rect(fill=NA, color=&quot;gray20&quot;, size=0.5, linetype=&quot;solid&quot;),</div><div class="line">  # 再加上X轴、Y轴，无刻度、无标签</div><div class="line">  axis.line = element_line(colour = &quot;black&quot;,size = 0.2),</div><div class="line">  # 去掉图例title</div><div class="line">  legend.title = element_blank()</div><div class="line">  # 去掉图例</div><div class="line">  legend.position = &quot;none&quot;,</div><div class="line">  # X轴标签：大小、字体、颜色、加粗、位置、角度 &lt;http://docs.ggplot2.org/current/element_text.html&gt;，字体需要单独加载字体family=“”，其他element_text()规则一样</div><div class="line">  ## face取值：plain普通，bold加粗，italic斜体，bold.italic斜体加粗</div><div class="line">  axis.title.x =element_text(size=12,color=&quot;black&quot;,face=&quot;bold&quot;,vjust = 0.5, hjust = 0.5, angle = 45),</div><div class="line">  # X轴刻度标签名</div><div class="line">  axis.text = element_blank(), # 去掉所有刻度标签</div><div class="line">  axis.text.x=element_blank(), # 去掉x刻度标签</div><div class="line">  axis.text.y=element_blank(), # 去掉y刻度标签</div><div class="line">  # Y轴</div><div class="line">  axis.title.y=element_text(size=8),</div><div class="line">  axis.text.y = element_text(size=8),</div><div class="line">  # X轴刻度线，不显示</div><div class="line">  axis.ticks = element_blank(), # 去掉所有刻度</div><div class="line">  axis.ticks.x=element_blank(), # 去掉x刻度</div><div class="line">  axis.ticks.y=element_blank(), # 去掉y刻度</div><div class="line">  )</div></pre></td></tr></table></figure>
<h4 id="默认主题"><a href="#默认主题" class="headerlink" title="默认主题"></a>默认主题</h4><ul>
<li>theme_classic，有x，y轴和没有网格线 #（这个不错）</li>
<li>theme_bw，高对比度主题</li>
<li>theme_gray，默认</li>
<li>theme_light</li>
<li>theme_linedraw，网格线变成黑实线</li>
<li>theme_minimal，没有背景内容</li>
<li>更多 library(ggthemes)</li>
</ul>
<h4 id="theme关键词表"><a href="#theme关键词表" class="headerlink" title="theme关键词表"></a>theme关键词表</h4><table>
<thead>
<tr>
<th>参数</th>
<th>设置内容</th>
<th>继承自</th>
</tr>
</thead>
<tbody>
<tr>
<td>line</td>
<td>所有线属性</td>
<td></td>
</tr>
<tr>
<td>rect</td>
<td>所有矩形区域属性</td>
<td></td>
</tr>
<tr>
<td>text</td>
<td>所有文本相关属性</td>
<td></td>
</tr>
<tr>
<td>title</td>
<td>所有标题属性</td>
<td></td>
</tr>
<tr>
<td>axis.title</td>
<td>坐标轴标题</td>
<td>text</td>
</tr>
<tr>
<td>axis.title.x</td>
<td>x轴属性</td>
<td>axis.title</td>
</tr>
<tr>
<td>axis.title.y</td>
<td>y轴属性</td>
<td>axis.title</td>
</tr>
<tr>
<td>axis.text</td>
<td>坐标轴刻度标签属性</td>
<td>text</td>
</tr>
<tr>
<td>axis.text.x</td>
<td>属性和继承和前面类似，不再重复</td>
<td></td>
</tr>
<tr>
<td>axis.text.y</td>
<td></td>
<td></td>
</tr>
<tr>
<td>axis.ticks</td>
<td>坐标轴刻度线</td>
<td>line</td>
</tr>
<tr>
<td>axis.ticks.x</td>
<td></td>
<td></td>
</tr>
<tr>
<td>axis.ticks.y</td>
<td></td>
<td></td>
</tr>
<tr>
<td>axis.ticks.length</td>
<td>刻度线长度</td>
<td></td>
</tr>
<tr>
<td>axis.ticks.margin</td>
<td>刻度线和刻度标签之间的间距</td>
<td></td>
</tr>
<tr>
<td>axis.line</td>
<td>坐标轴线</td>
<td>line</td>
</tr>
<tr>
<td>axis.line.x</td>
<td></td>
<td></td>
</tr>
<tr>
<td>axis.line.y</td>
<td></td>
<td></td>
</tr>
<tr>
<td>legend.background</td>
<td>图例背景</td>
<td>rect</td>
</tr>
<tr>
<td>legend.margin</td>
<td>图例边界</td>
<td></td>
</tr>
<tr>
<td>legend.key</td>
<td>图例符号</td>
<td></td>
</tr>
<tr>
<td>legend.key.size</td>
<td>图例符号大小</td>
<td></td>
</tr>
<tr>
<td>legend.key.height</td>
<td>图例符号高度</td>
<td></td>
</tr>
<tr>
<td>legend.key.width</td>
<td>图例符号宽度</td>
<td></td>
</tr>
<tr>
<td>legend.text</td>
<td>图例文字标签</td>
<td></td>
</tr>
<tr>
<td>legend.text.align</td>
<td>图例文字标签对齐方式</td>
<td>0为左齐，1为右齐</td>
</tr>
<tr>
<td>legend.title</td>
<td>图例标题</td>
<td>text</td>
</tr>
<tr>
<td>legend.title.align</td>
<td>图例标题对齐方式</td>
<td></td>
</tr>
<tr>
<td>legend.position</td>
<td>图例位置</td>
<td>left, right, bottom, top, 两数字向量</td>
</tr>
<tr>
<td>legend.direction</td>
<td>图例排列方向</td>
<td>“horizontal” or “vertical”</td>
</tr>
<tr>
<td>legend.justification</td>
<td>居中方式</td>
<td>center或两数字向量</td>
</tr>
<tr>
<td>legend.box</td>
<td>多图例的排列方式</td>
<td>“horizontal” or “vertical”</td>
</tr>
<tr>
<td>legend.box.just</td>
<td>多图例居中方式</td>
<td></td>
</tr>
<tr>
<td>panel.background</td>
<td>绘图区背景</td>
<td>rect</td>
</tr>
<tr>
<td>panel.border</td>
<td>绘图区边框</td>
<td>rect</td>
</tr>
<tr>
<td>panel.margin</td>
<td>分面绘图区之间的边距</td>
<td></td>
</tr>
<tr>
<td>panel.grid</td>
<td>绘图区网格线</td>
<td>line</td>
</tr>
<tr>
<td>panel.grid.major</td>
<td>主网格线</td>
<td></td>
</tr>
<tr>
<td>panel.grid.minor</td>
<td>次网格线</td>
<td></td>
</tr>
<tr>
<td>panel.grid.major.x</td>
<td></td>
<td></td>
</tr>
<tr>
<td>panel.grid.major.y</td>
<td></td>
<td></td>
</tr>
<tr>
<td>panel.grid.minor.x</td>
<td></td>
<td></td>
</tr>
<tr>
<td>panel.grid.minor.y</td>
<td></td>
<td></td>
</tr>
<tr>
<td>plot.background</td>
<td>整个图形的背景</td>
<td></td>
</tr>
<tr>
<td>plot.title</td>
<td>图形标题</td>
<td></td>
</tr>
<tr>
<td>plot.margin</td>
<td>图形边距</td>
<td>top, right, bottom, left</td>
</tr>
<tr>
<td>strip.background</td>
<td>分面标签背景</td>
<td>rect</td>
</tr>
<tr>
<td>strip.text</td>
<td>分面标签文本</td>
<td>text</td>
</tr>
<tr>
<td>strip.text.x</td>
<td></td>
<td></td>
</tr>
<tr>
<td>strip.text.y</td>
<td></td>
</tr>
</tbody>
</table>
<h3 id="加载字体"><a href="#加载字体" class="headerlink" title="加载字体"></a>加载字体</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line"># require(&quot;sysfonts&quot;)</div><div class="line"># require(&quot;showtext&quot;)</div><div class="line"># 查看字体列表，可以使用系统已安装的所有字体</div><div class="line">font.files()</div><div class="line"># 更改字体 family参数</div><div class="line">ggplot(data.frame(x = rnorm(100))) +</div><div class="line">    geom_histogram(aes(x), fill = &apos;purple&apos;, alpha = 0.6) +</div><div class="line">    labs(x = &apos;X 取值&apos;, y = &apos;频数 Count&apos;) +</div><div class="line">    theme(text = element_text(family = &apos;Arial Black&apos;))</div></pre></td></tr></table></figure>
<h3 id="x、y轴，数据范围"><a href="#x、y轴，数据范围" class="headerlink" title="x、y轴，数据范围"></a>x、y轴，数据范围</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"># 方法一</div><div class="line">p+ scale_y_continuous(limits=c(0,1.8))</div><div class="line"># scale_x_continuous(limits=c(0,10))</div><div class="line"></div><div class="line"># 方法二</div><div class="line">p+xlim(0,10) # 不一定都适用</div><div class="line">p + xlim(min(dt$A, 0)*1.2, max(dt$A)*1.2)   ## 一般使用倍数来限定大小，注意定义最小值的方式</div></pre></td></tr></table></figure>
<h3 id="坐标轴刻度间隔大小"><a href="#坐标轴刻度间隔大小" class="headerlink" title="坐标轴刻度间隔大小"></a>坐标轴刻度间隔大小</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">p + scale_x_continuous(breaks=seq(0, 10, 5)) # X轴，每隔5个单位</div></pre></td></tr></table></figure>
<hr>
<h1 id="检验"><a href="#检验" class="headerlink" title="检验"></a>检验</h1><h2 id="wilcox-test"><a href="#wilcox-test" class="headerlink" title="wilcox.test()"></a>wilcox.test()</h2><p>Wilcoxon rank sum test / Wilcoxon秩和检验<br>Mann–Whitney U test(曼-惠特尼U检验) / Mann–Whitney–Wilcoxon (MWW)</p>
<blockquote>
<p>在R中，wilcox.test()函数可以用来做Wilcoxon秩和检验，也可以用于做 Mann-Whitney U检验。</p>
<ul>
<li>当参数为<code>单个样本</code>/<code>或者是两个样本相减</code>/<code>或者是两个参数，paired=T</code>时，是Wilcoxon秩和检验。</li>
<li>当<code>两个参数，paired = FALSE（独立样本）</code>时，就是 Mann-Whitney U检验</li>
</ul>
<p>If only x is given, or if both x and y are given and paired is TRUE, a Wilcoxon signed rank test of the null that the distribution of x (in the one sample case) or of x - y (in the paired two sample case) is symmetric about mu is performed.</p>
<p>Otherwise, if both x and y are given and paired is FALSE, a Wilcoxon rank sum test (equivalent to the Mann-Whitney test: see the Note) is carried out. In this case, the null hypothesis is that the distributions of x and y differ by a location shift of mu and the alternative is that they differ by some other location shift (and the one-sided alternative “greater” is that x is shifted to the right of y).</p>
</blockquote>
<h3 id="配对Wilcoxon符号秩检验"><a href="#配对Wilcoxon符号秩检验" class="headerlink" title="配对Wilcoxon符号秩检验"></a>配对Wilcoxon符号秩检验</h3><h4 id="例8-1"><a href="#例8-1" class="headerlink" title="例8-1"></a>例8-1</h4><p>对12份血清分别用原方法（old, 检测时间20分钟）和新方法（new, 检测时间10分钟）测谷-丙转氨酶。问两法所得结果有无差别？<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">old &lt;- c(60,142,195,80,242,220,190,25,198,38,236,95)</div><div class="line">new &lt;- c(76,152,243,82,240,220,205,38,243,44,190,100)</div><div class="line">wilcox.test(old,new,paired = TRUE)</div><div class="line"></div><div class="line"># 运行结果</div><div class="line">&gt; wilcox.test(old,new,paired = TRUE) #配对wilcoxon检验</div><div class="line"></div><div class="line">        Wilcoxon signed rank test with continuity correction</div><div class="line"></div><div class="line">data:  old and new</div><div class="line">V = 11.5, p-value = 0.06175</div><div class="line">alternative hypothesis: true location shift is not equal to 0</div><div class="line"></div><div class="line">Warning messages:</div><div class="line">1: In wilcox.test.default(old, new, paired = TRUE) :</div><div class="line">  cannot compute exact p-value with ties</div><div class="line">2: In wilcox.test.default(old, new, paired = TRUE) :</div><div class="line">  cannot compute exact p-value with zeroes</div><div class="line"></div><div class="line"># 结果分析： p-value=0.06175，因此不能认为两法测谷丙转氨酶有差别。</div></pre></td></tr></table></figure></p>
<h3 id="独立样本Wilcoxon检验（Mann-Whitney-U检验）"><a href="#独立样本Wilcoxon检验（Mann-Whitney-U检验）" class="headerlink" title="独立样本Wilcoxon检验（Mann-Whitney U检验）"></a>独立样本Wilcoxon检验（Mann-Whitney U检验）</h3><h4 id="例8-2"><a href="#例8-2" class="headerlink" title="例8-2"></a>例8-2</h4><p>已知某地正常人尿氟含量的中位数为45.30μmol/L 。今在该地某厂随机抽取12名工人，测得尿氟含量。问该厂工人的尿氟含量是否高于当地正常人的尿氟含量？<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">data82 &lt;- c(44.21,45.30,46.39,49.47,51.05,53.16,53.26,54.37,57.16,67.37,71.05,87.37)</div><div class="line">wilcox.test(data82-45.30)</div><div class="line"></div><div class="line"># 运行结果：</div><div class="line">&gt; wilcox.test(data82,mu=45.30)</div><div class="line"></div><div class="line">        Wilcoxon signed rank test with continuity correction</div><div class="line"></div><div class="line">data:  data82</div><div class="line">V = 65, p-value = 0.005099</div><div class="line">alternative hypothesis: true location is not equal to 45.3</div><div class="line"></div><div class="line">Warning message:</div><div class="line">In wilcox.test.default(data82, mu = 45.3) :</div><div class="line">  cannot compute exact p-value with zeroes</div><div class="line"># 结果分析：p值小于0.01，可以视为该工厂的工人尿氟含量高于正常人。</div></pre></td></tr></table></figure></p>
<h4 id="例8-3"><a href="#例8-3" class="headerlink" title="例8-3"></a>例8-3</h4><p>对10例肺癌病人和12例矽肺0期工人用X光片测量肺门横径右侧距RD值（cm）。问肺癌病人的RD值是否高于矽肺0期工人的RD值？<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">lc &lt;- c(2.78,3.23,4.2,4.87,5.12,6.21,7.18,8.05,8.56,9.6)</div><div class="line">si &lt;- c(3.23,3.5,4.04,4.15,4.28,4.34,4.47,4.64,4.75,4.82,4.95,5.1)</div><div class="line">wilcox.test(lc,si)</div><div class="line"></div><div class="line">#运行结果：</div><div class="line">&gt; wilcox.test(lc,si,alternative=&quot;greater&quot;,exact=F)</div><div class="line"></div><div class="line">        Wilcoxon rank sum test with continuity correction</div><div class="line"></div><div class="line">data:  lc and si</div><div class="line">W = 86.5, p-value = 0.04318</div><div class="line">alternative hypothesis: true location shift is greater than 0</div><div class="line"># 在R中，wilcox.test()函数可以用来做Wilcoxon秩和检验，也可以用于做Mann-Whitney U检验。当参数为单个样本，或者是两个样本相减，或者是两个参数，paired=F时，是Wilcoxon秩和检验。当paired = FALSE（独立样本）时，就是Mann-Whitney U检验，在这个题目中，用的就是Mann-Whitney U检验，虽然结果中W=86.5与书中的T=141.5不一样，但本质上是一样的，换算如下：W1=141.5-10*(10+1)/2=86.5；W2=111.5-12*(12+1)/2=33.5</div></pre></td></tr></table></figure></p>
<h4 id="例8-4"><a href="#例8-4" class="headerlink" title="例8-4"></a>例8-4</h4><p>已知39名吸烟工人和40名不吸烟工人的碳氧血红蛋白HbCO(%)含量，问吸烟工人的HbCO(%)含量是否高于不吸烟工人的HbCO(%)含量？</p>
<p>答：本题书中用的是wilcox检验，其实用Ridit分析更合适一些，下面分别用这两种方法进行检验：</p>
<ul>
<li>Wilcox.test,或Kruskal检验</li>
</ul>
<p>Wilcox.test（或Kruskal检验）<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line">smoke &lt;- c(1,8,16,10,4)</div><div class="line">no.smoke &lt;-c(2,23,11,4,0)</div><div class="line">rank.c &lt;- c(1:5)</div><div class="line">group1 &lt;- rep(rank.c,smoke)</div><div class="line">group2 &lt;- rep(rank.c,no.smoke)</div><div class="line">data84 &lt;- c(group1,group2)</div><div class="line">group.f &lt;-factor(c(rep(1,length(group1)),rep(2,length(group2))))</div><div class="line">wilcox.test(data84~group.f)</div><div class="line">kruskal.test(data84~group.f)</div></pre></td></tr></table></figure></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div></pre></td><td class="code"><pre><div class="line"># 运行结果：</div><div class="line"># wilcox.test</div><div class="line">wilcox.test(data84~group.f)</div><div class="line"></div><div class="line">Wilcoxon rank sum test with continuity correction</div><div class="line"></div><div class="line">data: data84 by group.f</div><div class="line">W = 1137, p-value = 0.0002181</div><div class="line">alternative hypothesis: true location shift is not equal to 0</div><div class="line"></div><div class="line">Warning message:</div><div class="line">In wilcox.test.default(x = c(1L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, :</div><div class="line">cannot compute exact p-value with ties</div><div class="line"></div><div class="line"></div><div class="line"># kruskal.test检验</div><div class="line">&gt; kruskal.test(data84~group.f)</div><div class="line"></div><div class="line">Kruskal-Wallis rank sum test</div><div class="line"></div><div class="line">data: data84 by group.f</div><div class="line">Kruskal-Wallis chi-squared = 13.707, df = 1, p-value = 0.0002137</div></pre></td></tr></table></figure>
<ul>
<li>此题用Ridit检验更合适一些</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div></pre></td><td class="code"><pre><div class="line">data84 &lt;- matrix(c(1,8,16,10,4,2,23,11,4,0),nrow=5)</div><div class="line">dimnames(data84) &lt;- list(cone=c(&quot;very low&quot;,&quot;low&quot;,&quot;median&quot;,&quot;a bit high&quot;,&quot;high&quot;),</div><div class="line">worker=c(&quot;smoker&quot;,&quot;no-smoker&quot;))</div><div class="line">library(Ridit)</div><div class="line">ridit(data84,2)</div><div class="line">chisq.test(data84)</div><div class="line"></div><div class="line"># 运行结果：</div><div class="line">&gt; library(Ridit)</div><div class="line">&gt; ridit(data84,2)</div><div class="line"></div><div class="line">Ridit Analysis:</div><div class="line"></div><div class="line">Group Label Mean Ridit</div><div class="line">----- ----- ----------</div><div class="line">1 smoker 0.6159</div><div class="line">2 no-smoker 0.387</div><div class="line"></div><div class="line">Reference: Total of all groups</div><div class="line">chi-squared = 13.707, df = 1, p-value = 0.0002137</div><div class="line"></div><div class="line">&gt; chisq.test(data84)</div><div class="line"></div><div class="line">Pearsons Chi-squared test</div><div class="line"></div><div class="line">data: data84</div><div class="line">X-squared = 15.0785, df = 4, p-value = 0.004541</div></pre></td></tr></table></figure>
<h2 id="Kurskal-Wallis检验"><a href="#Kurskal-Wallis检验" class="headerlink" title="Kurskal-Wallis检验"></a>Kurskal-Wallis检验</h2><p>Kurskal-Wallis检验是Wilcoxon方法（其实是Mann-Whitney检验）用于多于两个样本的时候的升级版。当对两个样本进行比较的时候，Kurskal-Wallis检验与Mann-Whitney检验是等价的。</p>
<h4 id="例8-5"><a href="#例8-5" class="headerlink" title="例8-5"></a>例8-5</h4><p>用三种药物杀灭钉螺，每批用200只活钉螺，用药后清点每批钉螺的死亡数、再计算死亡率（%），问三种药物杀灭钉螺的效果有无差别？</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">drug &lt;-rep(c(&quot;甲药&quot;,&quot;乙药&quot;,&quot;丙药&quot;),each=5)</div><div class="line">data &lt;- c(32.5,35.5,40.5,46,49,16,20.5,22.5,29,36,6.5,9.0,12.5,18,24)</div><div class="line">data85 &lt;- data.frame(drug,data)</div><div class="line">kruskal.test(data85$data~data85$drug)</div><div class="line"></div><div class="line"># 运行结果：</div><div class="line">&gt; kruskal.test(data85$data~data85$drug)</div><div class="line"></div><div class="line">        Kruskal-Wallis rank sum test</div><div class="line"></div><div class="line">data:  data85$data by data85$drug</div><div class="line">Kruskal-Wallis chi-squared = 9.74, df = 2, p-value = 0.007673</div><div class="line"># 结果分析：Kruskal-Wallis的统计量是H值。</div></pre></td></tr></table></figure>
<h4 id="例8-6"><a href="#例8-6" class="headerlink" title="例8-6"></a>例8-6</h4><p>比较小白鼠接种三种不同菌型伤寒杆菌9D、11C和DSC1后存活日数，问小白鼠接种三种不同菌型伤寒杆菌的存活日数有无差别？</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">mice &lt;- as.factor(c(rep(&quot;9D&quot;,10),rep(&quot;11C&quot;,9),rep(&quot;DSC1&quot;,11)))</div><div class="line">data86 &lt;- c(2,2,2,3,4,4,4,5,7,7,5,5,6,6,6,7,8,10,12,3,5,6,6,6,7,7,9,10,11,11)</div><div class="line">data86 &lt;- data.frame(mice,data)</div><div class="line">kruskal.test(data86$data~data86$mice)</div><div class="line"></div><div class="line"># 运行结果：</div><div class="line">&gt; kruskal.test(data86$data~data86$mice)</div><div class="line"></div><div class="line">        Kruskal-Wallis rank sum test</div><div class="line"></div><div class="line">data:  data86$data by data86$mice</div><div class="line">Kruskal-Wallis chi-squared = 9.9405, df = 2, p-value = 0.006941</div></pre></td></tr></table></figure>
<h4 id="例8-7"><a href="#例8-7" class="headerlink" title="例8-7"></a>例8-7</h4><p>四种疾病患者痰液内嗜酸性白细胞的检查结果见表8-11。问四种疾病患者痰液内的嗜酸性白细胞有无差别？注：这道例题与《医学统计学及SAS应用》（上海交通大学）的9.11类似</p>
<p><img src="http://ofr9vioug.bkt.clouddn.com/8-7-table-KWtest.png?imageView/2/w/500" alt="image"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">x1 &lt;- c(0,2,9,6)</div><div class="line">x2 &lt;- c(3,5,5,2)</div><div class="line">x3 &lt;- c(5,7,3,2)</div><div class="line">x4 &lt;- c(3,5,3,0)</div><div class="line"></div><div class="line">freq &lt;- function(x)&#123;</div><div class="line">count &lt;-c()</div><div class="line">for(i in 1:4)&#123;</div><div class="line">count1 &lt;-c(rep(i,x[i]))</div><div class="line">count &lt;- append(count,count1)</div><div class="line">&#125;</div><div class="line">return(count)&#125;</div><div class="line">data87 &lt;-c(freq(x1),freq(x2),freq(x3),freq(x4))</div><div class="line">group &lt;- c(rep(1,sum(x1)),rep(2,sum(x2)),rep(3,sum(x3)),rep(4,sum(x4)))</div><div class="line">kruskal.test(data87~group)</div><div class="line"></div><div class="line"># 运行结果：</div><div class="line">&gt; kruskal.test(data87~group)</div><div class="line"></div><div class="line">        Kruskal-Wallis rank sum test</div><div class="line"></div><div class="line">data:  data87 by group</div><div class="line">Kruskal-Wallis chi-squared = 15.5058, df = 3, p-value = 0.001432</div></pre></td></tr></table></figure>
<h4 id="例8-8"><a href="#例8-8" class="headerlink" title="例8-8"></a>例8-8</h4><p>对例8-6资料作三个样本间的两两比较，Nemenyi检验。<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div><div class="line">33</div><div class="line">34</div><div class="line">35</div><div class="line">36</div><div class="line">37</div><div class="line">38</div><div class="line">39</div><div class="line">40</div><div class="line">41</div><div class="line">42</div><div class="line">43</div><div class="line">44</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line"># 需要安装下列包：</div><div class="line"># install.packages(&quot;pgirmess&quot;)</div><div class="line"># install.packages(&quot;coin&quot;)</div><div class="line"># install.packages(&quot;multcomp&quot;)</div><div class="line"># library(pgirmess)</div><div class="line"># library(coin)</div><div class="line"># library(multcomp) # 安装并加载要用到的包</div><div class="line">mice &lt;- as.factor(c(rep(&quot;9D&quot;,10),rep(&quot;11C&quot;,9),rep(&quot;DSC1&quot;,11)))</div><div class="line">data &lt;- c(2,2,2,3,4,4,4,5,7,7,5,5,6,6,6,7,8,10,12,3,5,6,6,6,7,7,9,10,11,11)</div><div class="line">data88 &lt;- data.frame(mice,data)</div><div class="line">kruskal.test(data~mice,data=data88)</div><div class="line">kruskalmc(data~mice, data=data88, probs=0.05) # 使用kruskalmc函数做两两比较，但此方法不能给出具体的值</div><div class="line"># 下面构建函数计算具体的p值</div><div class="line">mult &lt;- oneway_test(data ~ mice, data = data88,</div><div class="line">ytrafo = function(data) trafo(data, numeric_trafo = rank),</div><div class="line">xtrafo = function(data) trafo(data, factor_trafo = function(x)</div><div class="line">model.matrix(~x - 1) %*% t(contrMat(table(x), &quot;Tukey&quot;))),</div><div class="line">teststat = &quot;max&quot;, distribution = approximate(B = 90000))</div><div class="line">pvalue(mult, method = &quot;single-step&quot;) #计算具体的p值，这段代码是复制丁香园的[1]</div><div class="line"></div><div class="line"></div><div class="line"># 运行结果：</div><div class="line">&gt; kruskal.test(data~mice,data=data88)</div><div class="line"></div><div class="line">    Kruskal-Wallis rank sum test</div><div class="line"></div><div class="line">data:  data by mice</div><div class="line">Kruskal-Wallis chi-squared = 9.9405, df = 2, p-value = 0.006941</div><div class="line"></div><div class="line">&gt; kruskalmc(data~mice, data=data88, probs=0.05) # 使用kruskalmc函数做两两比较，但此方法不能给出具体的值</div><div class="line">Multiple comparison test after Kruskal-Wallis</div><div class="line">p.value: 0.05</div><div class="line">Comparisons</div><div class="line">            obs.dif critical.dif difference</div><div class="line">11C-9D   10.3777778     9.683378       TRUE</div><div class="line">11C-DSC1  0.4949495     9.472590      FALSE</div><div class="line">9D-DSC1  10.8727273     9.208410       TRUE</div><div class="line"></div><div class="line">&gt; pvalue(mult, method = &quot;single-step&quot;) #计算具体的p值</div><div class="line"></div><div class="line">9D - 11C   0.01870000</div><div class="line">DSC1 - 11C 0.95110000</div><div class="line">DSC1 - 9D  0.01088889</div></pre></td></tr></table></figure></p>
<h4 id="例8-9"><a href="#例8-9" class="headerlink" title="例8-9"></a>例8-9</h4><p>已知8名受试对象在相同实验条件下分别接受4种不同频率声音的刺激，他们的反应率（%）资料。问4种频率声音刺激的反应率是否有差别？</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div></pre></td><td class="code"><pre><div class="line"># 源代码：</div><div class="line">freA &lt;- c(8.40,11.60,9.40,9.80,8.30,8.60,8.90,7.80)</div><div class="line">freB &lt;- c(9.60,12.70,9.10,8.70,8.00,9.80,9.00,8.20)</div><div class="line">freC &lt;- c(9.80,11.80,10.40,9.90,8.60,9.60,10.60,8.50)</div><div class="line">freD &lt;- c(11.70,12.00,9.80,12.00,8.60,10.60,11.40,10.80)</div><div class="line">matrix89 &lt;- matrix(c(freA,freB,freC,freD),nrow=8,dimnames=list(no=1:8,c(&quot;A&quot;,&quot;B&quot;,&quot;C&quot;,&quot;D&quot;)))</div><div class="line">friedman.test(matrix89)</div><div class="line"></div><div class="line">运行结果：</div><div class="line">&gt; friedman.test(matrix89)</div><div class="line"></div><div class="line">        Friedman rank sum test</div><div class="line"></div><div class="line">data:  matrix89</div><div class="line">Friedman chi-squared = 15.1519, df = 3, p-value = 0.001691</div></pre></td></tr></table></figure>
<h2 id="手动计算PCA"><a href="#手动计算PCA" class="headerlink" title="手动计算PCA"></a>手动计算PCA</h2><h3 id="prcomp"><a href="#prcomp" class="headerlink" title="prcomp"></a>prcomp</h3><p>输入格式：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"># in.txt</div><div class="line">B1	B2	B3	B4	B5	B6	C1	C2	C3	C4	C5</div><div class="line">2 	2 	1 	1 	2 	2 	1 	1 	1 	1 	1</div><div class="line">1 	1 	1 	1 	1 	0 	0 	0 	1 	0 	0</div><div class="line">3 	3 	3 	3 	2 	3 	2 	2 	2 	2 	2</div><div class="line">......</div><div class="line">......</div></pre></td></tr></table></figure></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div></pre></td><td class="code"><pre><div class="line">y=read.table(<span class="string">"in.txt"</span>,header=<span class="literal">T</span>)</div><div class="line">y=as.matrix(y)</div><div class="line"><span class="comment"># 转置矩阵</span></div><div class="line">y=t(y)</div><div class="line">p=prcomp(y,scale=<span class="literal">T</span>)</div><div class="line">p2=summary(p)</div><div class="line"><span class="comment"># 显示PC1-PC4</span></div><div class="line">p2[<span class="number">5</span>]$x[,<span class="number">1</span>:<span class="number">4</span>]</div><div class="line"><span class="comment"># 查看PC解释度 Proportion of Variance</span></div><div class="line">p2[<span class="number">6</span>]$importance[,<span class="number">1</span>:<span class="number">4</span>]</div><div class="line"></div><div class="line"><span class="comment"># 存到文件里</span></div><div class="line">write.table(p3[<span class="number">5</span>]$x[,<span class="number">1</span>:<span class="number">4</span>],file=<span class="string">"pca.out"</span>,sep=<span class="string">" "</span>,quote=<span class="literal">F</span>)</div><div class="line">write.table(p3[<span class="number">6</span>]$importance[,<span class="number">1</span>:<span class="number">6</span>],file=<span class="string">"pca.out.log"</span>,sep=<span class="string">" "</span>,quote=<span class="literal">F</span>)</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/12/12/Bioinfomatics-Index/" itemprop="url">
                  Awesome-Bioinfomatics-List
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-12-12T15:31:23+08:00" content="2016-12-12">
              2016-12-12
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <blockquote>
<p>Ctrl+F, and search the key words</p>
</blockquote>
<h4 id="linux"><a href="#linux" class="headerlink" title="linux"></a>linux</h4><p>Windows Subsystem for Linux <a href="https://github.com/goreliu/wsl-terminal" target="_blank" rel="external">here</a></p>
<h4 id="Coding-language"><a href="#Coding-language" class="headerlink" title="Coding language"></a>Coding language</h4><p>Perl  <a href="http://www.perl.org/" target="_blank" rel="external">here</a><br>Python <a href="https://www.python.org/" target="_blank" rel="external">here</a><br>R <a href="http://www.r-project.org/" target="_blank" rel="external">here</a><br>Cookbook for R <a href="http://www.cookbook-r.com/" target="_blank" rel="external">here</a></p>
<h4 id="Plot"><a href="#Plot" class="headerlink" title="Plot"></a>Plot</h4><p>ggplot2 <a href="http://docs.ggplot2.org/current/" target="_blank" rel="external">here</a><br>Circos <a href="http://circos.ca/" target="_blank" rel="external">here</a></p>
<p>Color<br>Colorbrewer2 <a href="http://colorbrewer2.org/#type=sequential&amp;scheme=BuGn&amp;n=3" target="_blank" rel="external">here</a></p>
<h4 id="data-handling"><a href="#data-handling" class="headerlink" title="data handling"></a>data handling</h4><p>trim fastq reads <a href="https://github.com/lh3/trimadap" target="_blank" rel="external">trimadap</a><br>SeqKit <a href="http://bioinf.shenwei.me/seqkit/usage/" target="_blank" rel="external">here</a></p>
<h4 id="Mapping"><a href="#Mapping" class="headerlink" title="Mapping"></a>Mapping</h4><p>BWA <a href="http://bio-bwa.sourceforge.net/" target="_blank" rel="external">here</a><br>Maq (out-of-date) <a href="http://maq.sourceforge.net/" target="_blank" rel="external">here</a><br>stampy (out-of-date) <a href="http://www.well.ox.ac.uk/project-stampy" target="_blank" rel="external">here</a><br>Bowtie (out-of-date) <a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_blank" rel="external">here</a><br>Bowtie2 <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank" rel="external">here</a><br>minimap <a href="https://github.com/lh3/minimap" target="_blank" rel="external">here</a><br>SAM/BAM flag explain <a href="https://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="external">here</a><br>lastz <a href="http://www.bx.psu.edu/miller_lab/dist/README.lastz-1.02.00/README.lastz-1.02.00a.html" target="_blank" rel="external">here</a></p>
<ul>
<li>axt format <a href="http://genome.ucsc.edu/goldenPath/help/axt.html" target="_blank" rel="external">here</a></li>
<li>maf format <a href="http://genome.ucsc.edu/FAQ/FAQformat#format5" target="_blank" rel="external">here</a></li>
</ul>
<h4 id="SNP-Indel-calling"><a href="#SNP-Indel-calling" class="headerlink" title="SNP/Indel calling"></a>SNP/Indel calling</h4><p>samtools <a href="http://samtools.sourceforge.net/samtools.shtml" target="_blank" rel="external">here</a><br>gatk <a href="http://www.broadinstitute.org/gatk/" target="_blank" rel="external">here</a><br>angsd <a href="http://popgen.dk/angsd/index.php/Main_Page" target="_blank" rel="external">here</a><br>soapsnp <a href="http://soap.genomics.org.cn/soapsnp.html" target="_blank" rel="external">here</a><br>soapindel <a href="http://soap.genomics.org.cn/soapindel.html" target="_blank" rel="external">here</a><br>pindel <a href="http://gmt.genome.wustl.edu/pindel/0.2.4/index.html" target="_blank" rel="external">here</a><br>dindel <a href="http://www.sanger.ac.uk/resources/software/dindel/" target="_blank" rel="external">here</a></p>
<p>Variant call format<br>about vcf format <a href="http://vcftools.sourceforge.net/specs.html" target="_blank" rel="external">here</a><br>vcftools <a href="http://vcftools.sourceforge.net/index.html" target="_blank" rel="external">here</a></p>
<h4 id="CNV-call"><a href="#CNV-call" class="headerlink" title="CNV call"></a>CNV call</h4><p>CNVnator <a href="https://github.com/abyzovlab/CNVnator" target="_blank" rel="external">here</a><br>Genome STRiP <a href="http://software.broadinstitute.org/software/genomestrip/" target="_blank" rel="external">here</a><br>XHMM <a href="http://atgu.mgh.harvard.edu/xhmm/index.shtml" target="_blank" rel="external">here</a><br><a href="https://secure.genome.ucla.edu/index.php/ExomeCNV_User_Guide" target="_blank" rel="external">ExomeCNV</a><br><a href="http://contra-cnv.sourceforge.net" target="_blank" rel="external">CONTRA</a><br><a href="http://www.bioconductor.org/packages/2.9/bioc/html/exomeCopy.html" target="_blank" rel="external">ExomeCopy</a><br><a href="http://cran.r-project.org/web/packages/ExomeDepth/index.html" target="_blank" rel="external">ExomeDepth</a><br><a href="http://conifer.sourceforge.net/index.html" target="_blank" rel="external">CoNIFER</a><br><a href="http://bioconductor.org/packages/release/bioc/html/cn.mops.html" target="_blank" rel="external">cn.mops - Mixture of Poissons for CNV detection in NGS data</a> <a href="https://academic.oup.com/nar/article/40/9/e69/1136601/cn-MOPS-mixture-of-Poissons-for-discovering-copy" target="_blank" rel="external">paper</a><br><a href="https://sourceforge.net/projects/cnvhitseq/" target="_blank" rel="external">cnvHiTSeq</a></p>
<h4 id="Genotype-phasing-imputation"><a href="#Genotype-phasing-imputation" class="headerlink" title="Genotype phasing, imputation"></a>Genotype phasing, imputation</h4><p>beagle <a href="http://faculty.washington.edu/browning/beagle/beagle.html" target="_blank" rel="external">here</a><br>impute2 <a href="http://genome.sph.umich.edu/wiki/IMPUTE2:_1000_Genomes_Imputation_Cookbook" target="_blank" rel="external">here</a><br>phase <a href="http://stephenslab.uchicago.edu/software.html#phase" target="_blank" rel="external">here</a><br>fastphase <a href="http://depts.washington.edu/fphase/Old/" target="_blank" rel="external">here</a><br>SHAPEIT <a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html" target="_blank" rel="external">here</a></p>
<h4 id="Phylogenetic-tree"><a href="#Phylogenetic-tree" class="headerlink" title="Phylogenetic tree"></a>Phylogenetic tree</h4><p>fasttree <a href="http://www.microbesonline.org/fasttree/" target="_blank" rel="external">here</a><br>RAxML <a href="http://sco.h-its.org/exelixis/web/software/raxml/index.html" target="_blank" rel="external">here</a><br>Phylip <a href="http://evolution.genetics.washington.edu/phylip.html" target="_blank" rel="external">here</a><br>PAUP* <a href="http://www.molecularevolution.org/software/phylogenetics/paup" target="_blank" rel="external">here</a>   <a href="http://paup.sc.fsu.edu/port.html" target="_blank" rel="external">here</a><br>Phylogen Programs <a href="http://evolution.genetics.washington.edu/phylip/software.html" target="_blank" rel="external">here</a>  </p>
<ul>
<li><p>draw a nice tree <a href="http://www.evolgenius.info/evolview/#mytrees/DEMOS/bars" target="_blank" rel="external">here</a>   <a href="http://icytree.org/" target="_blank" rel="external">here</a></p>
</li>
<li><p>compare two trees <a href="https://cran.r-project.org/web/packages/dendextend/vignettes/introduction.html" target="_blank" rel="external">dendextend</a></p>
</li>
</ul>
<h4 id="Analysis-tools"><a href="#Analysis-tools" class="headerlink" title="Analysis tools"></a>Analysis tools</h4><p>plink <a href="http://pngu.mgh.harvard.edu/~purcell/plink/index.shtml" target="_blank" rel="external">here</a><br>ngstools <a href="https://github.com/mfumagalli/ngsTools" target="_blank" rel="external">here</a></p>
<h4 id="LD"><a href="#LD" class="headerlink" title="LD"></a>LD</h4><p>haploview<a href="http://www.broadinstitute.org/scientific-community/science/programs/medical-and-population-genetics/haploview/haploview" target="_blank" rel="external">here</a></p>
<h4 id="IBD"><a href="#IBD" class="headerlink" title="IBD"></a>IBD</h4><p>beagle <a href="https://faculty.washington.edu/browning/beagle/beagle.html" target="_blank" rel="external">here</a><br>germline <a href="http://www.cs.columbia.edu/~gusev/germline/" target="_blank" rel="external">here</a></p>
<h4 id="Population-Structure-analysis"><a href="#Population-Structure-analysis" class="headerlink" title="Population Structure analysis"></a>Population Structure analysis</h4><p>frappe <a href="http://med.stanford.edu/tanglab/software/frappe.html" target="_blank" rel="external">here</a><br>structure <a href="http://pritchardlab.stanford.edu/structure.html" target="_blank" rel="external">here</a><br>admixture <a href="http://www.genetics.ucla.edu/software/admixture/" target="_blank" rel="external">here</a><br>ngsAdmix <a href="http://www.popgen.dk/software/index.php/NgsAdmix" target="_blank" rel="external">here</a></p>
<h4 id="PCA"><a href="#PCA" class="headerlink" title="PCA"></a>PCA</h4><p>eigensoft <a href="http://genetics.med.harvard.edu/reich/Reich_Lab/Software.html" target="_blank" rel="external">here</a></p>
<h4 id="Demographic-history"><a href="#Demographic-history" class="headerlink" title="Demographic history"></a>Demographic history</h4><p>dadi <a href="https://code.google.com/p/dadi/" target="_blank" rel="external">here</a><br>psmc <a href="https://github.com/lh3/psmc" target="_blank" rel="external">here</a><br>msmc <a href="https://github.com/stschiff/msmc" target="_blank" rel="external">here</a><br>smc++ <a href="https://github.com/popgenmethods/smcpp" target="_blank" rel="external">here</a><br>treemix <a href="https://code.google.com/p/treemix/" target="_blank" rel="external">here</a><br>fastsimcoal2 <a href="http://cmpg.unibe.ch/software/fastsimcoal2/" target="_blank" rel="external">here</a><br>ChromoPainter/fineStructure/GLOBETROTTER <a href="http://www.paintmychromosomes.com/" target="_blank" rel="external">here</a><br>G-PhoCS <a href="http://compgen.cshl.edu/GPhoCS/" target="_blank" rel="external">here</a><br>SLim <a href="https://messerlab.org/slim/" target="_blank" rel="external">here</a><br>BEAST <a href="http://beast.bio.ed.ac.uk/" target="_blank" rel="external">here</a></p>
<p>Admix relatd:<br>ADMIXTOOLS <a href="https://reich.hms.harvard.edu/software" target="_blank" rel="external">here</a> <a href="https://github.com/DReichLab/AdmixTools" target="_blank" rel="external">here</a><br>ALDER <a href="https://reich.hms.harvard.edu/software" target="_blank" rel="external">here</a></p>
<p>Simulator：<br>msPrime <a href="https://msprime.readthedocs.io/en/stable/" target="_blank" rel="external">here</a><br>msms <a href="https://github.com/delt0r/msms" target="_blank" rel="external">here</a></p>
<h4 id="Seletive-sweeps"><a href="#Seletive-sweeps" class="headerlink" title="Seletive sweeps"></a>Seletive sweeps</h4><p>sweep <a href="http://www.broadinstitute.org/mpg/sweep/" target="_blank" rel="external">here</a><br>ihs and xpehh <a href="http://hgdp.uchicago.edu/Software/" target="_blank" rel="external">here</a><br>nSL <a href="http://cteg.berkeley.edu/software.html" target="_blank" rel="external">here</a><br>SweepFinder <a href="http://people.binf.ku.dk/rasmus/webpage/sf.html" target="_blank" rel="external">here</a><br>SweeD <a href="http://sco.h-its.org/exelixis/web/software/sweed/index.html" target="_blank" rel="external">here</a><br>xp-clr <a href="http://genetics.med.harvard.edu/reichlab/Reich_Lab/Software.html" target="_blank" rel="external">here</a><br>selscan <a href="https://github.com/szpiech/selscan" target="_blank" rel="external">here</a></p>
<h4 id="Transcriptome"><a href="#Transcriptome" class="headerlink" title="Transcriptome"></a>Transcriptome</h4><p>Trinity <a href="http://trinityrnaseq.sourceforge.net/" target="_blank" rel="external">here</a><br>velvet <a href="http://www.ebi.ac.uk/~zerbino/velvet/" target="_blank" rel="external">here</a><br>Tophat <a href="http://ccb.jhu.edu/software/tophat/index.shtml" target="_blank" rel="external">here</a><br>cufflinks <a href="http://cole-trapnell-lab.github.io/cufflinks/" target="_blank" rel="external">here</a><br>HISAT2 <a href="https://ccb.jhu.edu/software/hisat2/manual.shtml" target="_blank" rel="external">here</a><br>WGCNA (a R package for weighted correlation network analysis)  <a href="https://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/" target="_blank" rel="external">here</a><br>Network plot: <a href="http://www.cytoscape.org/" target="_blank" rel="external">Cytoscape</a><br><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170803-093337@2x.png?imageView/2/w/500" alt="image"></p>
<h4 id="Database"><a href="#Database" class="headerlink" title="Database"></a>Database</h4><p>DDBJ <a href="http://www.ddbj.nig.ac.jp/index-e.html" target="_blank" rel="external">here</a><br>ENA <a href="http://www.ebi.ac.uk/ena/home" target="_blank" rel="external">here</a><br>NCBI <a href="http://www.ncbi.nlm.nih.gov/" target="_blank" rel="external">here</a><br>ensembl <a href="http://asia.ensembl.org/index.html" target="_blank" rel="external">here</a><br>ensembl plant <a href="http://plants.ensembl.org/index.html" target="_blank" rel="external">here</a><br>KEGG <a href="http://www.genome.jp/kegg/" target="_blank" rel="external">here</a><br>GO <a href="http://www.geneontology.org/" target="_blank" rel="external">here</a><br>Plant transcription factor database (PlantTFDB) <a href="http://planttfdb.cbi.pku.edu.cn/" target="_blank" rel="external">here</a><br>Human Phenotype Ontology <a href="http://human-phenotype-ontology.github.io/" target="_blank" rel="external">here</a><br>GSA, Genome sequence archive of Beijing Institutes of Genomics(BIG) <a href="http://gsa.big.ac.cn/" target="_blank" rel="external">here</a><br>水稻rice <a href="http://www.rmbreeding.cn/Index/" target="_blank" rel="external">here</a> <a href="http://cgm.sjtu.edu.cn/3kricedb/" target="_blank" rel="external">here</a><br>人human <a href="http://genetics.med.harvard.edu/reichlab/Reich_Lab/Datasets.html" target="_blank" rel="external">David Reich lab</a></p>
<p>GOO/KEGG Enrichment<br><a href="https://www.r-bloggers.com/kegg-enrichment-analysis-with-latest-online-data-using-clusterprofiler/" target="_blank" rel="external">clusterProfiler</a><br><a href="http://www.genome.jp/tools/kaas/" target="_blank" rel="external">kaas</a><br><a href="http://geneontology.org/page/go-enrichment-analysis" target="_blank" rel="external">Gene Ontology</a><br><a href="http://pantherdb.org/" target="_blank" rel="external">PANTHER</a><br><a href="http://cbl-gorilla.cs.technion.ac.il/" target="_blank" rel="external">GORILLA</a><br><a href="https://david.ncifcrf.gov/summary.jsp" target="_blank" rel="external">David</a></p>
<h4 id="GWAS"><a href="#GWAS" class="headerlink" title="GWAS"></a>GWAS</h4><p>FAST-LMM <a href="http://research.microsoft.com/en-us/um/redmond/projects/mscompbio/fastlmm/" target="_blank" rel="external">here</a><br>GAPIT <a href="https://groups.google.com/forum/#!forum/gapit-forum" target="_blank" rel="external">here</a><br>EMMAX <a href="http://csg.sph.umich.edu/kang/emmax/download/index.html" target="_blank" rel="external">here</a></p>
<h4 id="Metagenomics-MGWAS"><a href="#Metagenomics-MGWAS" class="headerlink" title="Metagenomics(MGWAS)"></a>Metagenomics(MGWAS)</h4><p>MaAsLin <a href="https://huttenhower.sph.harvard.edu/maaslin" target="_blank" rel="external">here</a></p>
<h4 id="RAD-seq"><a href="#RAD-seq" class="headerlink" title="RAD-seq"></a>RAD-seq</h4><p>Stacks <a href="http://catchenlab.life.illinois.edu/stacks/manual/" target="_blank" rel="external">here</a></p>
<h4 id="SNP-annotation"><a href="#SNP-annotation" class="headerlink" title="SNP annotation"></a>SNP annotation</h4><p>SNPeff <a href="http://snpeff.sourceforge.net/SnpEff_manual.html#intro" target="_blank" rel="external">here</a><br>NGS-SNP <a href="http://www.ualberta.ca/~stothard/downloads/NGS-SNP/" target="_blank" rel="external">here</a><br>ANNOVAR <a href="http://annovar.openbioinformatics.org/en/latest/user-guide/startup/" target="_blank" rel="external">here</a></p>
<h4 id="Gene-annotation"><a href="#Gene-annotation" class="headerlink" title="Gene annotation"></a>Gene annotation</h4><p>EVidenceModeler(EVM) <a href="https://evidencemodeler.github.io/" target="_blank" rel="external">here</a><br>DAVID <a href="https://david.ncifcrf.gov/home.jsp" target="_blank" rel="external">here</a><br>Gene2Function <a href="http://www.gene2function.org/search/" target="_blank" rel="external">here</a><br>Gene Structure Display Server <a href="http://gsds.cbi.pku.edu.cn/" target="_blank" rel="external">GSDS2</a></p>
<h4 id="Protein-functional-effects"><a href="#Protein-functional-effects" class="headerlink" title="Protein functional effects"></a>Protein functional effects</h4><p>PolyPhen-2 <a href="http://genetics.bwh.harvard.edu/pph2/" target="_blank" rel="external">here</a><br>Protein-Protein Interaction Networks 基因结构画图展示<a href="http://string-db.org/" target="_blank" rel="external">here</a></p>
<h4 id="Tools"><a href="#Tools" class="headerlink" title="Tools"></a>Tools</h4><p>Primer3 <a href="https://www.plob.org/article/4529.html" target="_blank" rel="external">here</a> <a href="https://www.plob.org/article/4530.html" target="_blank" rel="external">here</a><br>TMHMM (domain prediction) <a href="http://www.cbs.dtu.dk/services/TMHMM/" target="_blank" rel="external">here</a><br>Wego <a href="http://wego.genomics.org.cn/cgi-bin/wego/index.pl" target="_blank" rel="external">here</a><br>Species &amp; general Chinese name <a href="http://wiki.cnki.com.cn/HotWord/28322.htm" target="_blank" rel="external">here</a><br>tree of life <a href="http://tolweb.org/tree/" target="_blank" rel="external">here</a><br>制作流程图 <a href="https://www.draw.io/" target="_blank" rel="external">here</a> <a href="https://www.processon.com/" target="_blank" rel="external">here</a><br>Venn diagram 维恩图 <a href="https://github.com/linguoliang/VennPainter/" target="_blank" rel="external">VennPainter</a>  <a href="http://bioinfogp.cnb.csic.es/tools/venny/index.html" target="_blank" rel="external">Venny2.1</a></p>
<p>draw a plot like this <a href="http://weblogo.berkeley.edu/logo.cgi" target="_blank" rel="external">weblogo</a><br><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170419-145649-atgc.png?imageView/2/w/500" alt="image"></p>
<h4 id="Cancer-related"><a href="#Cancer-related" class="headerlink" title="Cancer related"></a>Cancer related</h4><p>The Cancer Genome Atlas(TCGA) pan-cancer anlysis <a href="https://cancergenome.nih.gov/" target="_blank" rel="external">here</a> <a href="http://www.nature.com/tcga/" target="_blank" rel="external">here</a><br>International Cancer Genome Consortium (ICGC) <a href="http://icgc.org/" target="_blank" rel="external">here</a><br>National Cancer Institute (NCI) <a href="https://www.cancer.gov/research/areas/genomics" target="_blank" rel="external">here</a><br>National Comprehensive Cancer Network (NCCN), including NCCN guidelines for clinical application. <a href="https://www.nccn.org/" target="_blank" rel="external">here</a></p>
<h4 id="Disease"><a href="#Disease" class="headerlink" title="Disease"></a>Disease</h4><p>CDC Public Health Genomics Knowledge Base <a href="https://phgkb.cdc.gov/GAPPKB/phgHome.do?action=home" target="_blank" rel="external">here</a><br>疾病位点打分相关 <a href="http://bejerano.stanford.edu/mcap/" target="_blank" rel="external">M-CAP</a> <a href="http://bejerano.stanford.edu/mcap/" target="_blank" rel="external">M-CAP online</a><br>疾病位点打分相关 <a href="https://sites.google.com/site/revelgenomics/about" target="_blank" rel="external">REVEL</a><br><a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank" rel="external">ClinVar</a><br><a href="https://sites.google.com/site/jpopgen/dbNSFP" target="_blank" rel="external">dbNSFP</a><br><a href="http://www.hgmd.cf.ac.uk/" target="_blank" rel="external">Human Gene Mutation Database</a><br><a href="http://swissvar.expasy.org/" target="_blank" rel="external">SwissVar</a></p>
<h4 id="evolution"><a href="#evolution" class="headerlink" title="evolution"></a>evolution</h4><p><a href="http://evolution.berkeley.edu/evolibrary/home.php" target="_blank" rel="external">UC berkeley, Understanding Evolution</a><br><a href="http://tiktaalik.uchicago.edu/resources.html#" target="_blank" rel="external">tiktaalik roseae</a> <a href="http://open.163.com/movie/2015/10/9/5/MB617DHCA_MB619EC95.html" target="_blank" rel="external">你身体里的鱼 Your inner fish</a></p>
<h4 id="Phenotype"><a href="#Phenotype" class="headerlink" title="Phenotype"></a>Phenotype</h4><p>mouse gene knockout and Phenotype <a href="http://www.mousephenotype.org/" target="_blank" rel="external">here</a></p>
<h4 id="查基因功能网站："><a href="#查基因功能网站：" class="headerlink" title="查基因功能网站："></a>查基因功能网站：</h4><p><a href="http://www.genecards.org/" target="_blank" rel="external">GeneCards</a><br><a href="http://www.wikigenes.org/?referer-ignore=1" target="_blank" rel="external">wikigenes</a></p>
<hr>
<h4 id="Contact"><a href="#Contact" class="headerlink" title="Contact"></a><a href="https://github.com/wlz0726/Images/blob/master/Wechat/QRcode-wanglizhong0726.jpg" target="_blank" rel="external">Contact</a></h4><p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20170615-092735@2x.png?imageView/2/w/200" alt="image"></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/11/15/libgd-and-GD-problem-installing-Circos/" itemprop="url">
                  libgd and GD problem (installing Circos)
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-11-15T12:47:33+08:00" content="2016-11-15">
              2016-11-15
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><a href="http://zientzilaria.herokuapp.com/blog/2012/06/03/installing-circos-on-os-x/" target="_blank" rel="external">clear explaination</a></p>
<h3 id="1-install-libgd-2-1-0"><a href="#1-install-libgd-2-1-0" class="headerlink" title="1. install libgd-2.1.0"></a>1. install libgd-2.1.0</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div></pre></td><td class="code"><pre><div class="line">wget https://github.com/libgd/libgd/releases/download/gd-2.1.0/libgd-2.1.0.tar.gz</div><div class="line">tar zxvf libgd-2.1.0.tar.gz</div><div class="line">cd libgd-2.1.0</div><div class="line">./configure</div><div class="line">...</div><div class="line">** Configuration summary for libgd 2.1.0:</div><div class="line"></div><div class="line">   Support for Zlib:                 yes</div><div class="line">   Support for PNG library:          yes</div><div class="line">   Support for JPEG library:         yes</div><div class="line">   Support for VPX library:          no</div><div class="line">   Support for TIFF library:         yes</div><div class="line">   Support for Freetype 2.x library: yes</div><div class="line">   Support for Fontconfig library:   yes</div><div class="line">   Support for Xpm library:          no</div><div class="line">   Support for pthreads:             yes</div><div class="line"></div><div class="line">...</div></pre></td></tr></table></figure>
<p>If Freetype and Fontconfig are missing, you would have to download, configure, make and install them like this:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div></pre></td><td class="code"><pre><div class="line">curl -O http://www.ijg.org/files/jpegsrc.v8d.tar.gz</div><div class="line">tar -xzvf jpegsrc.v8d.tar.gz</div><div class="line">cd jpeg-8d</div><div class="line">./configure --prefix=/path/to/jpeg-8d</div><div class="line">make</div><div class="line">make install (or &quot;sudo make install&quot;)</div><div class="line"></div><div class="line">curl -O ftp://ftp.simplesystems.org/pub/libpng/png/src/libpng16/libpng-1.6.2.tar.gz</div><div class="line">tar -xzvf libpng-1.6.2.tar.gz</div><div class="line">cd libpng-1.6.2.tar.gz</div><div class="line">./configure --prefix=/path/to/jpeg-8d</div><div class="line">make</div><div class="line">make install (or &quot;sudo make install&quot;)</div></pre></td></tr></table></figure>
<p>Finish install libgd<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">./configure --prefix=/path/to/jpeg-8d</div><div class="line">make</div><div class="line">make install(or &quot;sudo make install&quot;)</div></pre></td></tr></table></figure></p>
<h3 id="2-install-GD-manually-perl-module"><a href="#2-install-GD-manually-perl-module" class="headerlink" title="2. install GD manually (perl module)"></a>2. install GD manually (perl module)</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">curl -O http://www.cpan.org/authors/id/L/LD/LDS/GD-2.49.tar.gz (if curl fails copy and past on your browser)</div><div class="line">tar -xzvf GD-2.49.tar.gz</div><div class="line">cd GD-2.49</div><div class="line">GD-2.49$ perl Makefile.PL</div><div class="line">GD-2.49$ make</div><div class="line">GD-2.49$ sudo make install</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/11/07/Fix-pheatmap-bug/" itemprop="url">
                  Fix pheatmap bug
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-11-07T19:55:11+08:00" content="2016-11-07">
              2016-11-07
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Work/" itemprop="url" rel="index">
                    <span itemprop="name">Work</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>pheatmap画图保存为pdf的时候会产生空白页，添加<code>onefile=FALSE</code>可以解决<br><figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">pdf(<span class="string">"test.pdf"</span>, onefile=<span class="literal">FALSE</span>)</div></pre></td></tr></table></figure></p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/11/01/perl-cheetsheets/" itemprop="url">
                  perl cheetsheets
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-11-01T20:22:32+08:00" content="2016-11-01">
              2016-11-01
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>比整个屏幕长的脚本都应该加上<code>use strict;</code></p>
<h2 id="数据元素"><a href="#数据元素" class="headerlink" title="数据元素"></a>数据元素</h2><h3 id="变量"><a href="#变量" class="headerlink" title="变量"></a>变量</h3><p>当你需要声明一个复杂的字符串时，可以使用heredoc语法：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> $blurb =&lt;&lt;<span class="string">'END_BLURB'</span>;</div><div class="line">He looked up. <span class="string">"Change is the constant on which they all</span></div><div class="line">can agree. We instead, born out of time, remain perfect</div><div class="line">and perfectly self-aware. We only suffer change as we</div><div class="line">pursue it. It is against our nature. We rebel against</div><div class="line">that change. Shall we consider them greater for it?"</div><div class="line">END_BLURB</div></pre></td></tr></table></figure>
<h3 id="Undef"><a href="#Undef" class="headerlink" title="Undef"></a>Undef</h3><p>Perl里的undef表示一个未分配、不确定、未知的值。<br>一个声明了但是没有定义的标量值就是undef：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> $name = <span class="keyword">undef</span>;     <span class="comment"># unnecessary assignment</span></div><div class="line"><span class="keyword">my</span> $rank;     <span class="comment"># also contains undef</span></div></pre></td></tr></table></figure>
<h3 id="数字"><a href="#数字" class="headerlink" title="数字"></a>数字</h3><p>Perl支持整型数字和浮点数字，你可以使用不同的方式来表示它们，二进制、八进制、十进制、十六进制：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> $integer = <span class="number">42</span>;    <span class="comment">#整型</span></div><div class="line"><span class="keyword">my</span> $float = <span class="number">0</span>.<span class="number">007</span>;    <span class="comment">#浮点型</span></div><div class="line"><span class="keyword">my</span> $sci_float = <span class="number">1.02</span>e14;    <span class="comment">#科学计数法，浮点数</span></div><div class="line"><span class="keyword">my</span> $binary = 0b10101<span class="number">0</span>;    <span class="comment">#二进制0b前缀</span></div><div class="line"><span class="keyword">my</span> $octal = <span class="number">052</span>;    <span class="comment">#八进制0前缀</span></div><div class="line"><span class="keyword">my</span> $hex = <span class="number">0x20</span>;    <span class="comment">#十六进制0x前缀</span></div></pre></td></tr></table></figure>
<p>还可以使用下划线来增加可读性：（注意不是逗号，因为逗号在Perl中有特殊意义）</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> $billion = <span class="number">1000000000</span>;</div><div class="line"><span class="keyword">my</span> $billion = <span class="number">1_000_000_000</span>;</div><div class="line"><span class="keyword">my</span> $billion = <span class="number">10_0_00_00_0_0_0</span>;</div></pre></td></tr></table></figure>
<h3 id="列表"><a href="#列表" class="headerlink" title="列表"></a>列表</h3><p>列表list 标量的有序集合；数组array 储存列表的变量</p>
<p>使用范围操作符（..）可以很方便的创建列表：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> @chars = <span class="string">'a'</span> .. <span class="string">'z'</span>;</div><div class="line"><span class="keyword">my</span> @count = <span class="number">13</span> .. <span class="number">27</span>;</div></pre></td></tr></table></figure>
<p>可以使用qw()操作符分隔空白来产生字符串列表：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> @stooges = <span class="string">qw( Larry Curly Moe Shemp Joey Kenny )</span>;</div></pre></td></tr></table></figure>
<h3 id="数组操作"><a href="#数组操作" class="headerlink" title="数组操作"></a>数组操作</h3><p>压入、弹出<br><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> $fred = <span class="keyword">pop</span>(@array)   <span class="comment"># pop从右边弹出；shift从左边弹出</span></div><div class="line"><span class="keyword">push</span> (@array,$a)         <span class="comment"># push从右边压入；unshift从左边压入</span></div></pre></td></tr></table></figure></p>
<p>最后一个元素的索引 <code>$trees[$#trees]</code></p>
<p>reverse 反向 <code>@fred = reverse @fred</code> #本身不修改</p>
<p>排序<br><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> @result = <span class="keyword">sort</span> by_number @some_numbers</div><div class="line"><span class="keyword">my</span> @result = <span class="keyword">sort</span> &#123; $a &lt;=&gt; $b&#125; @some_numbers;  <span class="comment"># &lt;=&gt; 符号表示数字的比较和排序</span></div><div class="line"><span class="keyword">my</span> @string = <span class="keyword">sort</span> &#123;$a cmp $b&#125; @any_strings;  <span class="comment"># cmp 表示字符串的比较和排序</span></div><div class="line"><span class="keyword">my</span> @string = <span class="keyword">sort</span> &#123;<span class="string">"\L$a"</span> cmp <span class="string">"\L$b"</span>&#125; @any_strings;  <span class="comment"># \L表示忽略大小写</span></div></pre></td></tr></table></figure></p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"><span class="function"><span class="keyword">sub</span> <span class="title">by_score_and_name</span> </span>&#123;</div><div class="line">  &#123;$score&#123;$b&#125; &lt;=&gt; $score&#123;$a&#125;&#125;  <span class="comment"># 按照value对key做降序排列</span></div><div class="line"></div><div class="line">  <span class="keyword">or</span></div><div class="line"></div><div class="line">  $a cmp $b <span class="comment"># 按照字母顺序再排序</span></div><div class="line">&#125;</div></pre></td></tr></table></figure>
<h2 id="流程控制"><a href="#流程控制" class="headerlink" title="流程控制"></a>流程控制</h2><h3 id="条件分支、循环"><a href="#条件分支、循环" class="headerlink" title="条件分支、循环"></a>条件分支、循环</h3><p>使用<strong>后缀形式</strong>，添加括号增加可读性：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">print</span> <span class="string">"yes\n"</span> <span class="keyword">if</span> ($a&gt;<span class="number">1</span>);</div><div class="line"></div><div class="line"><span class="keyword">print</span> <span class="string">"$_"</span> <span class="keyword">for</span> <span class="number">1</span>..<span class="number">10</span></div></pre></td></tr></table></figure>
<h3 id="三目操作符"><a href="#三目操作符" class="headerlink" title="三目操作符"></a>三目操作符</h3><p>三目操作符（? :）可以根据给定的条件选择执行表达式.</p>
<p>条件表达式为真时执行表达式1，条件表达式为假时执行表达式2：<br>条件表达式 ? 表达式1 : 表达式2</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> $time_suffix = after_noon($time)</div><div class="line">? <span class="string">'afternoon'</span></div><div class="line">: <span class="string">'morning'</span>;</div><div class="line"></div><div class="line"><span class="keyword">push</span> @&#123; <span class="keyword">rand</span>() &gt; <span class="number">0</span>.<span class="number">5</span> ? \@red_team : \@blue_team &#125;, Player-&gt;new;</div></pre></td></tr></table></figure>
<h3 id="循环控制"><a href="#循环控制" class="headerlink" title="循环控制"></a>循环控制</h3><p>next-立即进入下一循环：</p>
<p>last-立即结束循环。</p>
<p>redo-重新进入当前迭代，返回本次循环的顶端（无需评估循环条件）。</p>
<p>循环嵌套时，可能不容易让人理清楚结构，这时可以使用标签：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">LINE:</div><div class="line"><span class="keyword">while</span> (&lt;$fh&gt;)&#123;</div><div class="line">  <span class="keyword">chomp</span>;  </div><div class="line">  PREFIX:</div><div class="line">  <span class="keyword">for</span> <span class="keyword">my</span> $prefix (@prefixes)&#123;</div><div class="line">    <span class="keyword">next</span> LINE <span class="keyword">unless</span> $prefix;</div><div class="line">    <span class="keyword">say</span> <span class="string">"$prefix: $_"</span>;</div><div class="line">    <span class="comment"># next PREFIX is implicit here</span></div><div class="line">  &#125;</div><div class="line">&#125;</div></pre></td></tr></table></figure>
<h2 id="哈希"><a href="#哈希" class="headerlink" title="哈希"></a>哈希</h2><h3 id="访问哈希"><a href="#访问哈希" class="headerlink" title="访问哈希"></a>访问哈希</h3><p>访问键<br><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">for</span> <span class="keyword">my</span> $addressee (<span class="keyword">keys</span> %addresses)&#123;</div><div class="line"><span class="keyword">say</span> <span class="string">"Found an address for $addressee!"</span>;</div><div class="line">&#125;</div></pre></td></tr></table></figure></p>
<p>访问值<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">for my $address (values %addresses)&#123;</div><div class="line">say &quot;Someone lives at $address&quot;;</div><div class="line">&#125;</div></pre></td></tr></table></figure></p>
<p><strong>访问键值对</strong><br>each 函数,唯一使用地方就是在while循环中<br><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">while</span> (<span class="keyword">my</span> ($addressee, $address) = <span class="keyword">each</span> %addresses)&#123;</div><div class="line"><span class="keyword">say</span> <span class="string">"$addressee lives at $address"</span>;</div><div class="line">&#125;</div></pre></td></tr></table></figure></p>
<h3 id="用哈希去重复用法"><a href="#用哈希去重复用法" class="headerlink" title="用哈希去重复用法"></a>用哈希去重复用法</h3><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">my</span> %uniq;</div><div class="line"><span class="keyword">undef</span> @uniq&#123; @items &#125;;</div><div class="line"><span class="keyword">my</span> @uniques = <span class="keyword">keys</span> %uniq;</div><div class="line"></div><div class="line"><span class="comment"># @items是需要去重的数据，@uniques是去重后的</span></div><div class="line"><span class="comment"># 对哈希切片使用undef就是将所有的切片值设置为undef。</span></div></pre></td></tr></table></figure>
<h3 id="exists-函数-检查哈四中是否有某个键"><a href="#exists-函数-检查哈四中是否有某个键" class="headerlink" title="exists 函数:检查哈四中是否有某个键"></a>exists 函数:检查哈四中是否有某个键</h3><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">if</span> (<span class="keyword">exists</span> $books&#123;<span class="string">"dino"</span>&#125;) &#123;</div><div class="line">    <span class="keyword">print</span> <span class="string">"Hey, there's a library card for dino!\n"</span>;</div><div class="line">&#125;</div></pre></td></tr></table></figure>
<h3 id="反转-松绑哈希"><a href="#反转-松绑哈希" class="headerlink" title="反转 松绑哈希"></a>反转 松绑哈希</h3><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">%inverse_hash = <span class="keyword">reverse</span> %any_hash <span class="comment"># 反转哈希  键值互换</span></div><div class="line">@any_array = %some_hash;  <span class="comment">#哈希松绑，把哈希转化成 键/值 对的列表</span></div></pre></td></tr></table></figure>
<h3 id="去除哈希里的键"><a href="#去除哈希里的键" class="headerlink" title="去除哈希里的键"></a>去除哈希里的键</h3><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">delete</span> $h&#123;$a&#125;</div></pre></td></tr></table></figure>
<h2 id="符号"><a href="#符号" class="headerlink" title="符号"></a>符号</h2><h3 id="数学操作符"><a href="#数学操作符" class="headerlink" title="数学操作符"></a>数学操作符</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div></pre></td><td class="code"><pre><div class="line">+ 加</div><div class="line">- 减</div><div class="line">* 乘</div><div class="line">/ 除</div><div class="line">** 乘方</div><div class="line">% 模，取余 10%3  余数为1</div><div class="line">+=,-=, *=, /=, **=,  %=  组合形式</div><div class="line">--  前缀自减</div><div class="line">== 相等</div><div class="line">!= 不等</div><div class="line">&gt; 大于</div><div class="line">&lt; 小于</div><div class="line">&gt;= 大于等于</div><div class="line">&lt;= 小于等于</div><div class="line">&lt;=&gt; 排序中的比较符号</div></pre></td></tr></table></figure>
<h3 id="自增-自减"><a href="#自增-自减" class="headerlink" title="自增 自减"></a>自增 自减</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">自增++ 自减--</div><div class="line">操作符++放在变量之前就能先增加变量的值，然后获取标量的值。</div><div class="line"></div><div class="line">$m = 5;</div><div class="line">my $n = ++$m; # $m 增加到 6，并把该值赋给 $n</div><div class="line"></div><div class="line">$m = 5;</div><div class="line">my $d = $m++; # $d 得到的值为 $m 之前的值 5 ，然后 $m 增加到6</div></pre></td></tr></table></figure>
<h3 id="双引号里的转义字符"><a href="#双引号里的转义字符" class="headerlink" title="双引号里的转义字符"></a>双引号里的转义字符</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div></pre></td><td class="code"><pre><div class="line">\n	#换行</div><div class="line">\r	#光标回到最左边</div><div class="line">\t	#水平制表符</div><div class="line">\f	#换页符</div><div class="line">\b	#退格</div><div class="line">\a	#系统响铃</div><div class="line">\e	#Esc（ASCII编码的转义字符）</div><div class="line">\007	#任何八进制的ASCII值（此例中007表示系统响铃）</div><div class="line">\x7f	#任何十六进制的ASCII值（此例中7f表示删除键的控制代码）</div><div class="line">\cC	#控制符，也就是control键的代码（此例中表示同时按下Ctrl和C键的返回码）</div><div class="line">\\	#反斜线</div><div class="line">\&quot;	#双引号</div><div class="line">\l	#将下个字符转为小写</div><div class="line">\L	#将到\E为止的所有字符转为小写</div><div class="line">\u	#将下个字符转为大写</div><div class="line">\U	#将到\E未知的所有字符转为大写</div><div class="line">\Q	#将到\E为止的非单词（non-word）字符串加上反斜线</div><div class="line">\E	#结束\L，\U或\Q</div></pre></td></tr></table></figure>
<h3 id="字符串操作符"><a href="#字符串操作符" class="headerlink" title="字符串操作符"></a>字符串操作符</h3><p>字符串操作符为操作数提供字符串语境。比如<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line">正则表达式的绑定操作符（=~， !~）</div><div class="line">字符串连接符（.）</div><div class="line">字符串比较操作符（eq）</div><div class="line">不等 ne</div><div class="line">小于 lt</div><div class="line">大于 gt</div><div class="line">小于或等于 le</div><div class="line">大于或等于 ge</div><div class="line">还有sort排序中的字符串比较操作符（cmp）</div></pre></td></tr></table></figure></p>
<h3 id="逻辑操作符"><a href="#逻辑操作符" class="headerlink" title="逻辑操作符"></a>逻辑操作符</h3><p>逻辑操作符为操作数提供布尔语境。如&amp;&amp;, ||, and， or，//，三目操作符（?:） 逻辑非 （!）， not （优先级比!低） ，还有xor。<br>注意//和||的区别，操作符//只要目标操作数有定义就为真，即使是0和空字符串。</p>
<p>优先级：<br>括号  &gt;  量词  &gt;  锚位和序列  &gt;  择一（竖线|）  &gt;  字符、字符集、反向引用</p>
<h3 id="位操作符"><a href="#位操作符" class="headerlink" title="位操作符"></a>位操作符</h3><p>位操作符提供的是数字语境。这些操作符不太常见。<br>左移位（&lt;&lt;），右移位（&gt;&gt;）， 逐位与运算（&amp;）， 逐位或运算（|）， 逐位异或运算（^）， 以及组合形式（&lt;&lt;=，&gt;&gt;=，&amp;=，|=，^=）。</p>
<h3 id="特殊操作符"><a href="#特殊操作符" class="headerlink" title="特殊操作符"></a>特殊操作符</h3><p>自增操作符（++）的行为比较特别，当操作数是数字时就数值加一。如果是字符串就增长字符串。</p>
<p>重复操作符（x）也有着复杂的行为，在列表上下文中将列表重复；在标量上下文中，重复并连接成一个字符串。</p>
<p>范围操作符（..），在列表语境下会生成从一个操作数到另一个操作数的列表，可以是数字也可以是字符串。<br>在布尔语境下行为很不一样：操作符初始值为假，在左操作数为真时整体变为真值，持续这个状态直到右操作数为真时，整体再转变变为假值。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div></pre></td><td class="code"><pre><div class="line"><span class="comment">#列表语境</span></div><div class="line"><span class="keyword">my</span> @cards = ( <span class="number">2</span> .. <span class="number">10</span>, <span class="string">'J'</span>, <span class="string">'Q'</span>, <span class="string">'K'</span>, <span class="string">'A'</span> );</div><div class="line"></div><div class="line"><span class="comment">#布尔语境</span></div><div class="line"> <span class="keyword">foreach</span> <span class="keyword">my</span> $x ( <span class="number">1</span>..<span class="number">10</span> )&#123;<span class="keyword">print</span> $x <span class="keyword">if</span> $x == <span class="number">3</span>..$x == <span class="number">4</span>&#125;</div><div class="line"><span class="comment"># 打印 3 4</span></div><div class="line"></div><div class="line"><span class="comment">#利用该特性来方便地提取邮件正文：</span></div><div class="line"><span class="keyword">while</span> (<span class="regexp">/Hello, $user/</span> .. /Sincerely,<span class="regexp">/)</span></div><div class="line">&#123;</div><div class="line">say "&gt; $_";</div><div class="line">&#125;</div></pre></td></tr></table></figure>
<h2 id="正则表达式"><a href="#正则表达式" class="headerlink" title="正则表达式"></a>正则表达式</h2><p>竖线 | 或 意思是左边或右边匹配都行</p>
<p>正则表达式 可进行双引号内插</p>
<p>可选修饰符：  可连起来使用，顺序不影响结果<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line">/i  # 不分大小写</div><div class="line">/s  # 用点号(.)匹配任意字符</div><div class="line">/x  # 加入空白</div><div class="line">/g  # 全局匹配修饰 第一次匹配后 立即进行下一次匹配/替换</div><div class="line"></div><div class="line"></div><div class="line">/m  # 跨行模式匹配</div></pre></td></tr></table></figure></p>
<p>锚位<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">^   # 表示字符串的开头</div><div class="line">$   # 表示字符串的结尾</div><div class="line"></div><div class="line">\b  # 单词锚位    /\bfred\b/ 整词搜索</div></pre></td></tr></table></figure></p>
<p>命名捕获<br>为捕获串加标签  <code>(?&lt;name&gt;PATTERN) print &quot;$name&quot;</code></p>
<p>通用变量<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div></pre></td><td class="code"><pre><div class="line">&#123;&#125; 重复次数的范围</div><div class="line">/a&#123;5,15&#125;/   字母a重复5到15次的匹配</div><div class="line">/(fred)&#123;3,&#125;/ fred重复3次以上</div><div class="line">/(fred)&#123;8&#125;/  重复8次</div><div class="line"></div><div class="line">*   &#123;0,&#125;</div><div class="line">+   &#123;1,&#125;</div><div class="line">?   &#123;0,1&#125;</div><div class="line"></div><div class="line"># 非贪婪量词</div><div class="line"># 在贪婪量词后面加？</div><div class="line">*   *?</div><div class="line">+   +?</div><div class="line">?   ??</div></pre></td></tr></table></figure></p>
<h2 id="函数"><a href="#函数" class="headerlink" title="函数"></a>函数</h2><h3 id="函数参数"><a href="#函数参数" class="headerlink" title="函数参数"></a>函数参数</h3><figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line"><span class="function"><span class="keyword">sub</span> <span class="title">show_pets_by_type</span></span>&#123;</div><div class="line"><span class="keyword">my</span> ($type, %pets) = @_;    <span class="comment">#记得要先将标量分出来</span></div><div class="line">  <span class="keyword">while</span> (<span class="keyword">my</span> ($name, $species) = <span class="keyword">each</span> %pets)&#123;</div><div class="line">    <span class="keyword">next</span> <span class="keyword">unless</span> $species eq $type;</div><div class="line">    <span class="keyword">say</span> <span class="string">"$name is a $species"</span>;</div><div class="line">  &#125;</div><div class="line">&#125;</div><div class="line"></div><div class="line"><span class="comment"># 列表赋值是贪婪的，所以上面例子中%pets会吞掉 @_ 里面所有剩下的值。所以如果$type参数的位置是在后面，那么就会报错，因为所有的值先被哈希吞掉，但是却发现单了一个。</span></div></pre></td></tr></table></figure>
<h3 id="验证参数"><a href="#验证参数" class="headerlink" title="验证参数"></a>验证参数</h3><p>某些时候参数验证是很容易的，比如验证参数的个数：</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line"><span class="function"><span class="keyword">sub</span> <span class="title">add_numbers</span></span>&#123;</div><div class="line">croak <span class="string">'Expected two numbers, received: '</span> . @_</div><div class="line"><span class="keyword">unless</span> @_ == <span class="number">2</span>;</div><div class="line">...</div><div class="line">&#125;</div></pre></td></tr></table></figure>
<p>有时则比较麻烦，比如要验证参数的类型。因为Perl中，类型可以发生转换。如果你有这方面的需求，可以看看这些模块：Params::Validate和MooseX::Method::Signatures。</p>
<h2 id="输入和输出"><a href="#输入和输出" class="headerlink" title="输入和输出"></a>输入和输出</h2><p>调用参数 <code>@ARGV</code>由调用参数组成的列表,开始运行时，@ARGV已经塞满了调用参数。</p>
<p>6个特殊的文件句柄是perl保留的：</p>
<p><code>STDIN</code>   <code>STDOUT</code>  <code>STDERR</code>  <code>DATA</code>    <code>ARGV</code>    <code>ARGVOUT</code></p>
<p>使用say来输出 与print相比会再每行输出的结尾z自动加上换行符</p>
<h3 id="printf格式化输出"><a href="#printf格式化输出" class="headerlink" title="printf格式化输出"></a>printf格式化输出</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">printf &quot;Hello, %s; your password expires in %d days!\n&quot;, $user, $days_to_die;</div><div class="line"># %s 代表字符串 $user； %d 代表数字 $days_to_die</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">%d    十进制整数，自动舍去小数点后数字</div><div class="line">%s    字符串格式，可设定字段宽度 printf &quot;%10s\n&quot;,&quot;wilma&quot;; (正数右对齐，负数左对齐)</div><div class="line">%f    转换格式（浮点数，四舍五入），可指定小数点后输出位数 printf &quot;12.3f\n&quot;,42 + 2/3;   (宽度12，小数点3位)</div><div class="line">%%    要输出百分号，使用 %%</div></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line"># 直接给字符串</div><div class="line"># Format number with up to 8 leading zeroes</div><div class="line">my $result = sprintf(&quot;%08d&quot;, $number);</div><div class="line"></div><div class="line"># Round number to 3 digits after decimal point</div><div class="line">my $rounded = sprintf(&quot;%.3f&quot;, $number);</div></pre></td></tr></table></figure>
<h2 id="perl套路"><a href="#perl套路" class="headerlink" title="perl套路"></a>perl套路</h2><h3 id="perl打乱list顺序"><a href="#perl打乱list顺序" class="headerlink" title="perl打乱list顺序"></a>perl打乱list顺序</h3><p><code>perl -MList::Util=shuffle -e &#39;print shuffle(&lt;STDIN&gt;);&#39; &lt; myfile</code></p>
<h3 id="perl随机打乱数组"><a href="#perl随机打乱数组" class="headerlink" title="perl随机打乱数组"></a>perl随机打乱数组</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">use List::Util;</div><div class="line">@array = List::Util::shuffle @array;</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/11/01/The-Art-of-Charm/" itemprop="url">
                  The Art of Charm
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-11-01T20:01:35+08:00" content="2016-11-01">
              2016-11-01
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Social/" itemprop="url" rel="index">
                    <span itemprop="name">Social</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><a href="https://www.youtube.com/watch?v=HoQkiy9Elcw&amp;index=14&amp;list=PLyqPLT4V7ooLJ7QYNguRX2d-xXhzywfvV&amp;spfreload=10" target="_blank" rel="external">Learn to be a social ninja!</a></p>
<ol>
<li>meet somebody new and <strong>be genuinely interested</strong> in people</li>
<li>assume you’re familiar with somebody</li>
<li><strong>smile</strong> often</li>
<li>giving compliment freely</li>
<li>the ability for you to laugh at yourself, self-deprecating</li>
<li>learn to listen</li>
<li>treat people with kindness and respect</li>
</ol>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/11/01/mac关闭屏幕方式/" itemprop="url">
                  mac关闭屏幕方式
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-11-01T19:56:56+08:00" content="2016-11-01">
              2016-11-01
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>快捷键</p>
<ul>
<li>Ctrl+Shift+Power: 关闭屏幕</li>
<li>Cmd+Opt+Power: 睡眠 (sleep)</li>
<li>Cmd+Ctrl+Power: 重启 (restart)</li>
<li>Cmd+Ctrl+Opt+Power: 关机 (shutdown)</li>
</ul>
<p>应用程序-&gt;实用工具-&gt;钥匙链访问<br>在钥匙串访问程序菜单中选择 偏好设置-&gt;通用，勾选“在菜单栏中显示状态”，之后在菜单栏会出现一个小锁, 选择第一个“锁定屏幕”。</p>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/10/28/Insert-pictures-with-markdown/" itemprop="url">
                  为Markdown插入图片
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-10-28T22:04:50+08:00" content="2016-10-28">
              2016-10-28
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h3 id="使用七牛云做为markdown图床"><a href="#使用七牛云做为markdown图床" class="headerlink" title="使用七牛云做为markdown图床"></a><a href="http://cnfeat.com/blog/2015/11/30/cli-qiniu/#section-8" target="_blank" rel="external">使用七牛云做为markdown图床</a></h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">http://cnfeat.com/blog/2015/11/30/cli-qiniu/#section-8</div></pre></td></tr></table></figure>
<h3 id="qshell-命令行同步工具"><a href="#qshell-命令行同步工具" class="headerlink" title="qshell 命令行同步工具"></a>qshell 命令行同步工具</h3><p>(2016.11.8 更新) qrsync被官方弃用了…<br>现在可以使用<a href="http://developer.qiniu.com/code/v6/tool/qshell.html" target="_blank" rel="external">qshell工具</a>实现<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">http://developer.qiniu.com/code/v6/tool/qshell.html</div></pre></td></tr></table></figure></p>
<h3 id="自动上传图片并获得图片外链地址"><a href="#自动上传图片并获得图片外链地址" class="headerlink" title="自动上传图片并获得图片外链地址"></a>自动上传图片并获得图片外链地址</h3><ul>
<li><p>图片保存至上传目录 <code>/Users/wanglizhong/software/7niu/PicUpload/</code></p>
</li>
<li><p>写个简单的脚本获得图片地址<br>get_url.pl内容：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">#/usr/bin/perl -w</div><div class="line">`/usr/local/bin/qshell qupload  /Users/wanglizhong/software/7niu/conf.json 2&gt;/Users/wan\</div><div class="line">glizhong/software/7niu/.tmp`;</div><div class="line">my $pic=`ls /Users/wanglizhong/software/7niu/PicUpload/*`;</div><div class="line">chomp $pic;</div><div class="line">$pic =~ /\/([^\/]+)$/;</div><div class="line">print &quot;![image](http://ofr9vioug.bkt.clouddn.com/$1)&quot;;</div><div class="line">`rm $pic`;</div></pre></td></tr></table></figure>
</li>
</ul>
<p>终端运行 <code>perl get_url.pl</code>，可以进一步使用Keyboard Maestro绑定快捷键。</p>
<ul>
<li>插入图片地址<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">![image](http://ofr9vioug.bkt.clouddn.com/QQ20161028-0@2x-chimpanzee.png)</div></pre></td></tr></table></figure>
</li>
</ul>
<p><img src="http://ofr9vioug.bkt.clouddn.com/QQ20161028-0@2x-chimpanzee.png" alt="image"></p>
<h3 id="调整markdown插入图片尺寸"><a href="#调整markdown插入图片尺寸" class="headerlink" title="调整markdown插入图片尺寸"></a>调整markdown插入图片尺寸</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">![image](http://ofr9vioug.bkt.clouddn.com/PairwiseCluster.png)</div><div class="line">添加后缀 “?imageView/2/w/600”，600为自定义宽度</div><div class="line">![image](http://ofr9vioug.bkt.clouddn.com/PairwiseCluster.png?imageView/2/w/600)</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/10/12/Deal-with-Emails/" itemprop="url">
                  Deal with Emails
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-10-12T11:57:40+08:00" content="2016-10-12">
              2016-10-12
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Work/" itemprop="url" rel="index">
                    <span itemprop="name">Work</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <h1 id="处理邮件的规则："><a href="#处理邮件的规则：" class="headerlink" title="处理邮件的规则："></a>处理邮件的规则：</h1><ol>
<li>迅速回复，回应可以很简短，回头处理意味着重读邮件，更浪费时间</li>
<li>每个字都很重要，干净简洁</li>
<li>经常清理收件箱：尽可能一次处理完，不能立马回复的集中时间处理并在当天清空</li>
<li>先处理后到的邮件，有的时候先前的问题可能已经被解决了</li>
<li>将有用信息转发给相关人员</li>
<li>回复邮件的时候，去掉不相关的收件人</li>
<li>不要用电邮发火</li>
<li>需要跟进的邮件，抄送给自己，届时需要查看进度的时候可以将原邮件再发送一遍，并加上类似“这件事情做的怎么样？”的内容；</li>
<li>转发给自己，加上相关标签、可能用到的搜索词条（描述性关键词），帮助未来的自己搜索</li>
</ol>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/10/08/Bioinfomatics-Data-Skills/" itemprop="url">
                  Bioinfomatics Data Skills Cheatsheets
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-10-08T14:01:47+08:00" content="2016-10-08">
              2016-10-08
            </time>
          </span>

          
            <span class="post-category" >
              &nbsp; | &nbsp;
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
              
                <span itemprop="about" itemscope itemtype="https://schema.org/Thing">
                  <a href="/categories/Cheetsheets/" itemprop="url" rel="index">
                    <span itemprop="name">Cheetsheets</span>
                  </a>
                </span>

                
                

              
            </span>
          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>Code should be readable, broken down into small contained components (modular), and reusable (so you’re not rewriting code to do the same tasks over and over again).</p>
<h3 id="Testing-Code-Strategy"><a href="#Testing-Code-Strategy" class="headerlink" title="Testing Code Strategy:"></a>Testing Code Strategy:</h3><ul>
<li>How many times is this code called by other code?</li>
<li>If this code were wrong, how detrimental to the final results would it be?</li>
<li>How noticeable would an error be if one occurred?</li>
</ul>
<p>It’s important to never assume a dataset is high quality. Rather, data’s quality should be proved through exploratory data analysis (known as EDA). EDA is not complex or time consuming, and will make your research much more robust to lurking surprises in large datasets.</p>
<p>Make <strong>Figures and Statistics</strong> the Results of Scripts</p>
<p>It’s important to always use <strong>relative paths</strong> (e.g., ../ data/stats/qual.txt) rather than absolute paths (e.g., /home/vinceb/projects/ zmays-snps/data/stats/qual.txt).</p>
<h3 id="Document-Readme-in-project’s-main-directories"><a href="#Document-Readme-in-project’s-main-directories" class="headerlink" title="Document/Readme in project’s main directories"></a>Document/Readme in project’s main directories</h3><ul>
<li>methods and workflows (command-line)</li>
<li>origin of all data</li>
<li>when you downloaded data/ data version/ how you downloaded the data</li>
<li>software version</li>
</ul>
<p>leverage directories to help stay organized.</p>
<h3 id="Shell-Expansion-Tips"><a href="#Shell-Expansion-Tips" class="headerlink" title="Shell Expansion Tips"></a>Shell Expansion Tips</h3><p><code>$ echo dog-{gone,bowl,bark}</code><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">$ ls</div><div class="line">dog-gone   dog-bowl   dog-bark</div></pre></td></tr></table></figure></p>
<p><code>$ mkdir -p zmays-snps/{data/seqs,scripts,analysis}</code></p>
<p><code>$ touch seqs/zmays{A,B,C}_R{1,2}.fastq</code></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">$ ls seqs/</div><div class="line"></div><div class="line">zmaysA_R1.fastq zmaysB_R1.fastq zmaysC_R1.fastq zmaysA_R2.fastq zmaysB_R2.fastq zmaysC_R2.fastq</div></pre></td></tr></table></figure>
<h3 id="shell-wildcards"><a href="#shell-wildcards" class="headerlink" title="shell wildcards"></a>shell wildcards</h3><p>Wildcard What it matches</p>
<ul>
<li><p>*: Zero or more characters (but ignores hidden les starting with a period).</p>
</li>
<li><p>?: One character (also ignores hidden les).</p>
</li>
<li><p>[A-Z]: Any character between the supplied alphanumeric range (in this case, any character betweenAandZ); this works for any alphanumeric    character range (e.g.,[0-9]matches any character between 0 and 9).</p>
</li>
<li><p>best to be as restrictive as possible with wildcards<br>Instead of zmaysB<em>, use `**zmaysB\</em>fastq<strong><code>or</code></strong>zmaysB_R?.fastq**` (the ? only matches a single character).</p>
</li>
</ul>
<p><code>$ ls zmays[AB]_R1.fastq</code></p>
<p>zmaysA_R1.fastq     zmaysB_R1.fastq</p>
<p><code>$ ls zmays[A-B]_R1.fastq</code></p>
<p>zmaysA_R1.fastq     zmaysB_R1.fastq</p>
<h3 id="Leading-Zeros-and-Sorting"><a href="#Leading-Zeros-and-Sorting" class="headerlink" title="Leading Zeros and Sorting"></a>Leading Zeros and Sorting</h3><p>e.g., le-0021.txt rather than le-21.txt</p>
<p><code>$ ls -l</code></p>
<p>-rw-r–r– 1 vinceb staff 0 Feb 21 21:23 genes-001.txt</p>
<p>-rw-r–r– 1 vinceb staff 0 Feb 21 21:23 genes-002.txt</p>
<p>[…]</p>
<p>-rw-r–r– 1 vinceb staff 0 Feb 21 21:23 genes-013.txt</p>
<p>-rw-r–r– 1 vinceb staff 0 Feb 21 21:23 genes-014.txt</p>
<h3 id="use-markdown-to"><a href="#use-markdown-to" class="headerlink" title="use markdown to"></a>use markdown to</h3><h3 id="Using-pipelines"><a href="#Using-pipelines" class="headerlink" title="Using pipelines"></a>Using pipelines</h3><h4 id="tee"><a href="#tee" class="headerlink" title="tee"></a>tee</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ program1 input.txt | tee intermediate-file.txt | program2 &gt; results.txt</div></pre></td></tr></table></figure>
<p>Here, program1’s standard output is both written to intermediate- le.txt and piped directly into program2’s standard input.</p>
<h3 id="Tmux"><a href="#Tmux" class="headerlink" title="Tmux"></a>Tmux</h3><p>new session with a name<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">$ tmux new-session -s maize-snps</div></pre></td></tr></table></figure></p>
<table>
<thead>
<tr>
<th>Key sequence</th>
<th>Action</th>
</tr>
</thead>
<tbody>
<tr>
<td>Control-a d</td>
<td>Detach</td>
</tr>
<tr>
<td>Control-a c</td>
<td>Create new window</td>
</tr>
<tr>
<td>tmux ls</td>
<td>list all sessions</td>
</tr>
<tr>
<td>tmux new -s new</td>
<td>creat a session named “new”</td>
</tr>
<tr>
<td>tmux att -t new</td>
<td>attach a session named “new”</td>
</tr>
<tr>
<td>tmux att -d -t new</td>
<td>attach a session named “test”, detaching it first</td>
</tr>
</tbody>
</table>
<h4 id="change-defalt-key-with-tmux-conf"><a href="#change-defalt-key-with-tmux-conf" class="headerlink" title="change defalt key with .tmux.conf"></a>change defalt key with .tmux.conf</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div><div class="line">30</div><div class="line">31</div><div class="line">32</div><div class="line">33</div></pre></td><td class="code"><pre><div class="line">set -g history-limit 10000</div><div class="line"># Automatically set window title</div><div class="line">#set-window-option -g automatic-renam</div><div class="line">#set-window-option -g xterm-keys on</div><div class="line"></div><div class="line">set-option -g set-titles on</div><div class="line"># Shift arrow to switch windows S shift M alt C ctrl</div><div class="line">unbind-key -n S-Left</div><div class="line">unbind-key -n S-Right</div><div class="line"></div><div class="line">#bind -n C-Left  previous-window</div><div class="line">#bind -n C-Right next-window</div><div class="line">bind -n F2 new-window</div><div class="line">bind -n F3 previous-window</div><div class="line">bind -n F4 next-window</div><div class="line">bind -n F7 copy-mode</div><div class="line"></div><div class="line"># kill window (prefix Ctrl+q)</div><div class="line">#bind ^q killw</div><div class="line"></div><div class="line"># display</div><div class="line">#set -g status-utf8 on</div><div class="line">set -g status-keys vi</div><div class="line">set -g status-interval 1</div><div class="line">set -g status-attr bright</div><div class="line">set -g status-fg white</div><div class="line">set -g status-bg black</div><div class="line">set -g status-left-length 20</div><div class="line">set -g status-left &apos;#[fg=green][#[fg=red]#S#[fg=green]]#[default]&apos;</div><div class="line">set -g status-justify centre</div><div class="line">set -g status-right &apos;#[fg=green][ %m/%d %H:%M:%S ]#[default]&apos;</div><div class="line">setw -g window-status-current-format &apos;#[fg=yellow](#I.#P#F#W)#[default]&apos;</div><div class="line">setw -g window-status-format &apos;#I#F#W&apos;</div></pre></td></tr></table></figure>

          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
      

  
  

  
  
  

  <article class="post post-type-normal " itemscope itemtype="http://schema.org/Article">

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
            
            
              
                
                <a class="post-title-link" href="/2016/10/08/hello-world/" itemprop="url">
                  build blog with hexo on github
                </a>
              
            
          </h1>
        

        <div class="post-meta">
          <span class="post-time">
            <span class="post-meta-item-icon">
              <i class="fa fa-calendar-o"></i>
            </span>
            <span class="post-meta-item-text">发表于</span>
            <time itemprop="dateCreated" datetime="2016-10-08T10:59:46+08:00" content="2016-10-08">
              2016-10-08
            </time>
          </span>

          

          
            
          

          

          
          

          
        </div>
      </header>
    


    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><a href="http://xiaolai.li/2016/06/22/makecs-build-a-blog-with-hexo-on-github/" target="_blank" rel="external">Referece</a></p>
<p>Creat a new repository on github with name ‘yourname.github.io’, then:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line">cd ~/Public</div><div class="line">git clone https://github.com/yourname/yourname.github.io.git</div><div class="line">hexo init yourname.github.io</div><div class="line">cd yourname.github.io</div><div class="line">npm install hexo-deployer-git --save</div></pre></td></tr></table></figure></p>
<h3 id="deployment"><a href="#deployment" class="headerlink" title="deployment"></a>deployment</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">hexo generate &amp;&amp; hexo server # hgs</div><div class="line">hexo generate &amp;&amp; hexo deploy # hgd</div></pre></td></tr></table></figure>
<h3 id="new-article"><a href="#new-article" class="headerlink" title="new article"></a>new article</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">hexo new &quot;new article name&quot;</div></pre></td></tr></table></figure>
<p>edit your article in the source/_posts/ directory</p>
<h3 id="Write-markdown-files-with-atom"><a href="#Write-markdown-files-with-atom" class="headerlink" title="Write markdown files with atom"></a>Write markdown files with atom</h3><p>用 ⇧⌃m 呼出预览</p>
<p><a href="http://note.youdao.com/iyoudao/?p=2411" target="_blank" rel="external">markdown语法</a></p>
<h3 id="html字符实体在markdown里表示特殊符号"><a href="#html字符实体在markdown里表示特殊符号" class="headerlink" title="html字符实体在markdown里表示特殊符号"></a>html字符实体在markdown里表示特殊符号</h3><p>比如 <code>|</code>符号 的表示方式有，可用在markdown插入表格中:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line">&amp;verbar;</div><div class="line">&amp;vert;</div><div class="line">&amp;VerticalLine;</div><div class="line">&amp;#x0007C;</div><div class="line">&amp;#124;</div></pre></td></tr></table></figure></p>
<p><a href="http://www.w3school.com.cn/tags/html_ref_entities.html" target="_blank" rel="external">字符实体参考手册</a></p>
<p><a href="http://www.w3school.com.cn/tags/html_ref_ascii.asp" target="_blank" rel="external">ASCII</a></p>
<p><a href="http://www.w3school.com.cn/tags/html_ref_symbols.html" target="_blank" rel="external">符号</a></p>
<p><a href="http://www.w3school.com.cn/html/html_colors.asp" target="_blank" rel="external">颜色代码</a></p>
<p><a href="http://www.w3school.com.cn/html/html_colornames.asp" target="_blank" rel="external">颜色名</a></p>
<h2 id="Google-Material-design"><a href="#Google-Material-design" class="headerlink" title="Google Material design"></a><a href="https://material.google.com/" target="_blank" rel="external">Google Material design</a></h2>
          
        
      
    </div>

    <div>
      
    </div>

    <div>
      
    </div>

    <footer class="post-footer">
      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </article>


    
  </section>

  


          </div>
          


          

        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    <div class="sidebar-inner">

      

      

      <section class="site-overview sidebar-panel  sidebar-panel-active ">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/images/avatar.gif"
               alt="Lizhong Wang" />
          <p class="site-author-name" itemprop="name">Lizhong Wang</p>
          <p class="site-description motion-element" itemprop="description">自我提升和自我完善是一种终身的修行</p>
        </div>
        <nav class="site-state motion-element">
          <div class="site-state-item site-state-posts">
            <a href="/archives">
              <span class="site-state-item-count">36</span>
              <span class="site-state-item-name">日志</span>
            </a>
          </div>

          
            <div class="site-state-item site-state-categories">
              
                <span class="site-state-item-count">4</span>
                <span class="site-state-item-name">分类</span>
              
            </div>
          

          
            <div class="site-state-item site-state-tags">
              <a href="/tags">
                <span class="site-state-item-count">38</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
            
              <span class="links-of-author-item">
                <a href="http://weibo.com/samr" target="_blank" title="微博">
                  
                    <i class="fa fa-fw fa-weibo"></i>
                  
                  微博
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://www.zhihu.com/people/wanglzh" target="_blank" title="知乎">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                  知乎
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="https://github.com/wlz0726" target="_blank" title="Github">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                  Github
                </a>
              </span>
            
              <span class="links-of-author-item">
                <a href="http://ofr9vioug.bkt.clouddn.com/QRcode-wanglizhong0726.jpg" target="_blank" title="Wechat">
                  
                    <i class="fa fa-fw fa-globe"></i>
                  
                  Wechat
                </a>
              </span>
            
          
        </div>

        
        

        
        

      </section>

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2017</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Lizhong Wang</span>
</div>

<div class="powered-by">
  由 <a class="theme-link" href="https://hexo.io">Hexo</a> 强力驱动
</div>

<div class="theme-info">
  主题 -
  <a class="theme-link" href="https://github.com/iissnan/hexo-theme-next">
    NexT.Muse
  </a>
</div>

        

        
      </div>
    </footer>

    <div class="back-to-top">
      <i class="fa fa-arrow-up"></i>
    </div>
  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  



  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.0.1"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.0.1"></script>



  
  

  

  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.0.1"></script>



  



  




  
  

  

  

  

</body>
</html>
